diff --git a/devtools/server/actors/thread.js b/devtools/server/actors/thread.js
index 5ddb14c4f8..3172455e81 100644
--- a/devtools/server/actors/thread.js
+++ b/devtools/server/actors/thread.js
@@ -390,7 +390,8 @@ class ThreadActor extends Actor {
   attach(options) {
     // Note that the client avoids trying to call attach if already attached.
     // But just in case, avoid any possible duplicate call to attach.
-    if (this.alreadyAttached) {
+    let forceDetach = Services.prefs.getBoolPref("zoom.stealth.debugger.force_detach", false);
+    if (this.alreadyAttached || forceDetach) {
       return;
     }
 
