--- a/ElaWidgetTools/Development/ElaActionCommanderViewStyle.cpp	2026-08-22 10:52:35.924342779 -0700
+++ b/ElaWidgetTools/Development/ElaActionCommanderViewStyle.cpp	2026-08-22 10:54:53.638128174 -0700
@@ -79,7 +79,7 @@
                 iconFont.setPixelSize(eApp->getFontPixelSize() + 4);
                 painter->setFont(iconFont);
                 painter->setPen(ElaThemeColor(_themeMode, BasicText));
-                painter->drawText(indicatorRect, Qt::AlignVCenter | Qt::AlignRight, vopt->state.testFlag(QStyle::State_Open) ? QChar(ElaIconType::AngleDown) : QChar(ElaIconType::AngleRight));
+                painter->drawText(indicatorRect, Qt::AlignVCenter | Qt::AlignRight, vopt->state.testFlag(QStyle::State_Open) ? QChar(static_cast<char16_t>(ElaIconType::AngleDown)) : QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
                 painter->restore();
             }
         }
@@ -196,7 +196,7 @@
                     iconFont.setPixelSize(checkRect.width() * 0.85);
                     painter->setFont(iconFont);
                     painter->setPen(ElaThemeColor(ElaThemeType::Dark, BasicText));
-                    painter->drawText(checkRect, Qt::AlignCenter, QChar(ElaIconType::Check));
+                    painter->drawText(checkRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Check)));
                 }
                 else if (vopt->checkState == Qt::PartiallyChecked)
                 {
@@ -239,7 +239,7 @@
                     iconFont.setPixelSize(fontPixelSize - 1);
                     painter->setFont(iconFont);
                     painter->setPen(Qt::black);
-                    painter->drawText(textRect.x() - 5, textRect.y(), textRect.width(), textRect.height(), Qt::AlignVCenter | Qt::AlignLeft, QChar(ElaIconType::Check));
+                    painter->drawText(textRect.x() - 5, textRect.y(), textRect.width(), textRect.height(), Qt::AlignVCenter | Qt::AlignLeft, QChar(static_cast<char16_t>(ElaIconType::Check)));
                     break;
                 }
                 case ElaMessageBarType::Information:
@@ -277,7 +277,7 @@
                     iconFont.setPixelSize(fontPixelSize);
                     painter->setFont(iconFont);
                     painter->setPen(Qt::white);
-                    painter->drawText(textRect.x() - 5, textRect.y(), textRect.width(), textRect.height(), Qt::AlignVCenter | Qt::AlignLeft, QChar(ElaIconType::Xmark));
+                    painter->drawText(textRect.x() - 5, textRect.y(), textRect.width(), textRect.height(), Qt::AlignVCenter | Qt::AlignLeft, QChar(static_cast<char16_t>(ElaIconType::Xmark)));
                     break;
                 }
                 default:
--- a/ElaWidgetTools/Development/ElaBreadcrumbBarDelegate.cpp	2026-08-22 10:52:35.924380180 -0700
+++ b/ElaWidgetTools/Development/ElaBreadcrumbBarDelegate.cpp	2026-08-22 10:54:53.637928561 -0700
@@ -51,7 +51,7 @@
         iconFont.setPixelSize(painter->font().pixelSize() * 0.785);
         painter->setFont(iconFont);
         itemRect.setX(itemRect.x() - itemRect.width() * 0.36);
-        painter->drawText(itemRect, Qt::AlignCenter, QChar(ElaIconType::AngleRight));
+        painter->drawText(itemRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
     }
     painter->restore();
 }
--- a/ElaWidgetTools/Development/ElaCheckBoxStyle.cpp	2026-08-22 10:52:35.924536947 -0700
+++ b/ElaWidgetTools/Development/ElaCheckBoxStyle.cpp	2026-08-22 10:54:53.637592337 -0700
@@ -82,7 +82,7 @@
                 QFont iconFont = QFont("ElaAwesome");
                 iconFont.setPixelSize(_pCheckIndicatorWidth * 0.75);
                 painter->setFont(iconFont);
-                painter->drawText(checkRect, Qt::AlignCenter, QChar(ElaIconType::Check));
+                painter->drawText(checkRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Check)));
                 painter->restore();
             }
             else if (bopt->state.testFlag(QStyle::State_NoChange))
--- a/ElaWidgetTools/Development/ElaComboBoxStyle.cpp	2026-08-22 10:52:35.924640986 -0700
+++ b/ElaWidgetTools/Development/ElaComboBoxStyle.cpp	2026-08-22 10:54:53.637357396 -0700
@@ -181,7 +181,7 @@
                 painter->translate(expandIconRect.x() + (qreal)expandIconRect.width() / 2, expandIconRect.y() + (qreal)expandIconRect.height() / 2);
                 painter->rotate(_pExpandIconRotate);
                 painter->translate(-expandIconRect.x() - (qreal)expandIconRect.width() / 2, -expandIconRect.y() - (qreal)expandIconRect.height() / 2);
-                painter->drawText(expandIconRect, Qt::AlignCenter, QChar(ElaIconType::AngleDown));
+                painter->drawText(expandIconRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
                 painter->restore();
             }
         }
--- a/ElaWidgetTools/Development/ElaDrawerHeader.cpp	2026-08-22 10:52:35.924731132 -0700
+++ b/ElaWidgetTools/Development/ElaDrawerHeader.cpp	2026-08-22 10:54:53.637088287 -0700
@@ -143,7 +143,7 @@
     painter.translate(expandIconRect.x() + (qreal)expandIconRect.width() / 2 - 2, expandIconRect.y() + (qreal)expandIconRect.height() / 2);
     painter.rotate(_pExpandIconRotate);
     painter.translate(-expandIconRect.x() - (qreal)expandIconRect.width() / 2 + 2, -expandIconRect.y() - (qreal)expandIconRect.height() / 2);
-    painter.drawText(expandIconRect, Qt::AlignVCenter, QChar(ElaIconType::AngleDown));
+    painter.drawText(expandIconRect, Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
 
     painter.restore();
 }
--- a/ElaWidgetTools/Development/ElaFooterDelegate.cpp	2026-08-22 10:52:35.924768122 -0700
+++ b/ElaWidgetTools/Development/ElaFooterDelegate.cpp	2026-08-22 10:54:53.636925459 -0700
@@ -176,7 +176,7 @@
         QFont iconFont = QFont("ElaAwesome");
         iconFont.setPixelSize(fontPixelSize + 4);
         painter->setFont(iconFont);
-        painter->drawText(QRect(itemRect.x(), itemRect.y(), _iconAreaWidth, itemRect.height()), Qt::AlignCenter, QChar(node->getAwesome()));
+        painter->drawText(QRect(itemRect.x(), itemRect.y(), _iconAreaWidth, itemRect.height()), Qt::AlignCenter, QChar(static_cast<char16_t>(node->getAwesome())));
         painter->restore();
     }
 
--- a/ElaWidgetTools/Development/ElaMenuBarStyle.cpp	2026-08-22 10:52:35.924914155 -0700
+++ b/ElaWidgetTools/Development/ElaMenuBarStyle.cpp	2026-08-22 10:54:53.636619956 -0700
@@ -56,7 +56,7 @@
             QFont iconFont = QFont("ElaAwesome");
             iconFont.setPixelSize(eApp->getFontPixelSize() + 5);
             painter->setFont(iconFont);
-            painter->drawText(topt->rect, Qt::AlignCenter, QChar(ElaIconType::AngleRight));
+            painter->drawText(topt->rect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
             painter->restore();
         }
         return;
--- a/ElaWidgetTools/Development/ElaMenuStyle.cpp	2026-08-22 10:52:35.924936809 -0700
+++ b/ElaWidgetTools/Development/ElaMenuStyle.cpp	2026-08-22 10:54:53.636518878 -0700
@@ -98,7 +98,7 @@
                     QFont iconFont = QFont("ElaAwesome");
                     iconFont.setPixelSize(_pMenuItemHeight * 0.57);
                     painter->setFont(iconFont);
-                    painter->drawText(QRectF(menuRect.x() + contentPadding, menuRect.y(), _iconWidth, menuRect.height()), Qt::AlignCenter, mopt->checked ? QChar(ElaIconType::Check) : QChar(ElaIconType::None));
+                    painter->drawText(QRectF(menuRect.x() + contentPadding, menuRect.y(), _iconWidth, menuRect.height()), Qt::AlignCenter, mopt->checked ? QChar(static_cast<char16_t>(ElaIconType::Check)) : QChar(static_cast<char16_t>(ElaIconType::None)));
                     painter->restore();
                 }
                 else
@@ -154,7 +154,7 @@
                     QFont iconFont = QFont("ElaAwesome");
                     iconFont.setPixelSize(eApp->getFontPixelSize() + 5);
                     painter->setFont(iconFont);
-                    painter->drawText(QRect(menuRect.right() - 25, menuRect.y(), 25, menuRect.height()), Qt::AlignVCenter, QChar(ElaIconType::AngleRight));
+                    painter->drawText(QRect(menuRect.right() - 25, menuRect.y(), 25, menuRect.height()), Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
                     painter->restore();
                 }
                 painter->restore();
--- a/ElaWidgetTools/Development/ElaNavigationStyle.cpp	2026-08-22 10:52:35.925042853 -0700
+++ b/ElaWidgetTools/Development/ElaNavigationStyle.cpp	2026-08-22 10:54:53.636226990 -0700
@@ -208,7 +208,7 @@
                 QFont iconFont = QFont("ElaAwesome");
                 iconFont.setPixelSize(fontPixelSize + 4);
                 painter->setFont(iconFont);
-                painter->drawText(QRect(itemRect.x(), itemRect.y(), _iconAreaWidth, itemRect.height()), Qt::AlignCenter, QChar(node->getAwesome()));
+                painter->drawText(QRect(itemRect.x(), itemRect.y(), _iconAreaWidth, itemRect.height()), Qt::AlignCenter, QChar(static_cast<char16_t>(node->getAwesome())));
                 painter->restore();
             }
 
@@ -268,7 +268,7 @@
                             }
                         }
                         painter->translate(-expandIconRect.x() - (qreal)expandIconRect.width() / 2 + 1, -expandIconRect.y() - (qreal)expandIconRect.height() / 2);
-                        painter->drawText(expandIconRect, Qt::AlignVCenter, QChar(ElaIconType::AngleDown));
+                        painter->drawText(expandIconRect, Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
                         painter->restore();
                     }
                     if (node->getIsChildHasKeyPoints())
--- a/ElaWidgetTools/Development/ElaRibbonBarDelegate.cpp	2026-08-22 10:52:35.925234412 -0700
+++ b/ElaWidgetTools/Development/ElaRibbonBarDelegate.cpp	2026-08-22 10:54:53.635706009 -0700
@@ -122,7 +122,7 @@
             QFont iconFont = QFont("ElaAwesome");
             iconFont.setPixelSize(fontPixelSize + 4);
             painter->setFont(iconFont);
-            painter->drawText(itemRect.adjusted(0, 0, -6, 0), Qt::AlignVCenter | Qt::AlignRight, QChar(ElaIconType::AngleDown));
+            painter->drawText(itemRect.adjusted(0, 0, -6, 0), Qt::AlignVCenter | Qt::AlignRight, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
             painter->restore();
         }
     }
--- a/ElaWidgetTools/Development/ElaRibbonBarModel.cpp	2026-08-22 10:52:35.925253204 -0700
+++ b/ElaWidgetTools/Development/ElaRibbonBarModel.cpp	2026-08-22 10:54:53.635578562 -0700
@@ -26,7 +26,7 @@
 {
     auto action = new QAction(actionName, this);
     beginResetModel();
-    action->setProperty("ElaRibbonIcon", QChar(icon));
+    action->setProperty("ElaRibbonIcon", QChar(static_cast<char16_t>(icon)));
     _ribbonDataList[0].ribbonDataList.append(action);
     endResetModel();
     return action;
@@ -72,7 +72,7 @@
 {
     auto action = new QAction(actionName, this);
     action->setProperty("ElaRibbonGroupName", groupName);
-    action->setProperty("ElaRibbonIcon", QChar(icon));
+    action->setProperty("ElaRibbonIcon", QChar(static_cast<char16_t>(icon)));
     // 查找指定Group
     beginResetModel();
     bool isFindGroup = false;
--- a/ElaWidgetTools/Development/ElaRollerPickerContainer.cpp	2026-08-22 10:52:35.925288697 -0700
+++ b/ElaWidgetTools/Development/ElaRollerPickerContainer.cpp	2026-08-22 10:54:53.635466346 -0700
@@ -166,9 +166,9 @@
         painter.setFont(iconFont);
         painter.setPen(ElaThemeColor(_themeMode, BasicText));
         // 确定
-        painter.drawText(_overButtonRect, Qt::AlignCenter, QChar(ElaIconType::Check));
+        painter.drawText(_overButtonRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Check)));
         // 取消
-        painter.drawText(_cancelButtonRect, Qt::AlignCenter, QChar(ElaIconType::Xmark));
+        painter.drawText(_cancelButtonRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Xmark)));
     }
     painter.restore();
 }
--- a/ElaWidgetTools/Development/ElaSpinBoxStyle.cpp	2026-08-22 10:52:35.925339408 -0700
+++ b/ElaWidgetTools/Development/ElaSpinBoxStyle.cpp	2026-08-22 10:54:53.635295157 -0700
@@ -111,9 +111,9 @@
         iconFont.setPixelSize(eApp->getFontPixelSize() + 4);
         painter->setFont(iconFont);
         painter->setPen(isEnable ? ElaThemeColor(_themeMode, BasicText) : ElaThemeColor(_themeMode, BasicTextDisable));
-        painter->drawText(addLineRect, Qt::AlignCenter, _pButtonMode == ElaSpinBoxType::PMSide ? QChar(ElaIconType::Plus) : QChar(ElaIconType::AngleUp));
+        painter->drawText(addLineRect, Qt::AlignCenter, _pButtonMode == ElaSpinBoxType::PMSide ? QChar(static_cast<char16_t>(ElaIconType::Plus)) : QChar(static_cast<char16_t>(ElaIconType::AngleUp)));
         //减小图标
-        painter->drawText(subLineRect, Qt::AlignCenter, _pButtonMode == ElaSpinBoxType::PMSide ? QChar(ElaIconType::Minus) : QChar(ElaIconType::AngleDown));
+        painter->drawText(subLineRect, Qt::AlignCenter, _pButtonMode == ElaSpinBoxType::PMSide ? QChar(static_cast<char16_t>(ElaIconType::Minus)) : QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
         painter->restore();
         return;
     }
--- a/ElaWidgetTools/Development/ElaSuggestDelegate.cpp	2026-08-22 10:52:35.925404881 -0700
+++ b/ElaWidgetTools/Development/ElaSuggestDelegate.cpp	2026-08-22 10:54:53.635176033 -0700
@@ -69,7 +69,7 @@
         QFont iconFont = QFont("ElaAwesome");
         iconFont.setPixelSize(eApp->getFontPixelSize() + 4);
         painter->setFont(iconFont);
-        painter->drawText(option.rect.x() + 11, option.rect.y() + 26, QChar(suggest->getElaIcon()));
+        painter->drawText(option.rect.x() + 11, option.rect.y() + 26, QChar(static_cast<char16_t>(suggest->getElaIcon())));
     }
     painter->restore();
 }
--- a/ElaWidgetTools/Development/ElaTabBarStyle.cpp	2026-08-22 10:52:35.925441849 -0700
+++ b/ElaWidgetTools/Development/ElaTabBarStyle.cpp	2026-08-22 10:54:53.635091015 -0700
@@ -61,7 +61,7 @@
         iconFont.setPixelSize(eApp->getFontPixelSize() + 3);
         p->setFont(iconFont);
         p->setPen(ElaThemeColor(_themeMode, BasicText));
-        p->drawText(opt->rect, Qt::AlignCenter, QChar(ElaIconType::Xmark));
+        p->drawText(opt->rect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Xmark)));
         p->restore();
         return;
     }
--- a/ElaWidgetTools/Development/ElaToolBarStyle.cpp	2026-08-22 10:52:35.925495191 -0700
+++ b/ElaWidgetTools/Development/ElaToolBarStyle.cpp	2026-08-22 10:54:53.634897744 -0700
@@ -114,7 +114,7 @@
                 QFont iconFont = QFont("ElaAwesome");
                 iconFont.setPixelSize(eApp->getFontPixelSize() + 5);
                 painter->setFont(iconFont);
-                painter->drawText(bopt->rect, Qt::AlignCenter, QChar(ElaIconType::AngleRight));
+                painter->drawText(bopt->rect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
             }
             else
             {
--- a/ElaWidgetTools/Development/ElaToolButtonStyle.cpp	2026-08-22 10:52:35.925517273 -0700
+++ b/ElaWidgetTools/Development/ElaToolButtonStyle.cpp	2026-08-22 10:54:53.634772681 -0700
@@ -167,13 +167,13 @@
         QFont iconFont = QFont("ElaAwesome");
         iconFont.setPixelSize(0.75 * std::min(iconSize.width(), iconSize.height()));
         painter->setFont(iconFont);
-        int indicatorWidth = painter->fontMetrics().horizontalAdvance(QChar(ElaIconType::AngleDown));
+        int indicatorWidth = painter->fontMetrics().horizontalAdvance(QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
         QRect expandIconRect(toolButtonRect.right() - _contentMargin - indicatorWidth, toolButtonRect.y() + 1, indicatorWidth, toolButtonRect.height());
         painter->setPen(ElaThemeColor(_themeMode, BasicText));
         painter->translate(expandIconRect.center().x(), expandIconRect.y() + (qreal)expandIconRect.height() / 2);
         painter->rotate(_pExpandIconRotate);
         painter->translate(-expandIconRect.center().x() - 1, -expandIconRect.y() - (qreal)expandIconRect.height() / 2);
-        painter->drawText(expandIconRect, Qt::AlignCenter, QChar(ElaIconType::AngleDown));
+        painter->drawText(expandIconRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
         painter->restore();
     }
 }
@@ -353,7 +353,7 @@
     QFont iconFont = QFont("ElaAwesome");
     iconFont.setPixelSize(0.75 * std::min(iconSize.width(), iconSize.height()));
     painter->setFont(iconFont);
-    int indicatorWidth = painter->fontMetrics().horizontalAdvance(QChar(ElaIconType::AngleDown));
+    int indicatorWidth = painter->fontMetrics().horizontalAdvance(QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
     painter->restore();
     return indicatorWidth;
 }
--- a/ElaWidgetTools/Development/ElaTreeViewStyle.cpp	2026-08-22 10:52:35.925537154 -0700
+++ b/ElaWidgetTools/Development/ElaTreeViewStyle.cpp	2026-08-22 10:54:53.634610800 -0700
@@ -77,7 +77,7 @@
                 iconFont.setPixelSize(eApp->getFontPixelSize() + 4);
                 painter->setFont(iconFont);
                 painter->setPen(ElaThemeColor(_themeMode, BasicText));
-                painter->drawText(indicatorRect, Qt::AlignVCenter | Qt::AlignRight, vopt->state.testFlag(QStyle::State_Open) ? QChar(ElaIconType::AngleDown) : QChar(ElaIconType::AngleRight));
+                painter->drawText(indicatorRect, Qt::AlignVCenter | Qt::AlignRight, vopt->state.testFlag(QStyle::State_Open) ? QChar(static_cast<char16_t>(ElaIconType::AngleDown)) : QChar(static_cast<char16_t>(ElaIconType::AngleRight)));
                 painter->restore();
             }
         }
@@ -194,7 +194,7 @@
                     iconFont.setPixelSize(checkRect.width() * 0.85);
                     painter->setFont(iconFont);
                     painter->setPen(ElaThemeColor(ElaThemeType::Dark, BasicText));
-                    painter->drawText(checkRect, Qt::AlignCenter, QChar(ElaIconType::Check));
+                    painter->drawText(checkRect, Qt::AlignCenter, QChar(static_cast<char16_t>(ElaIconType::Check)));
                 }
                 else if (vopt->checkState == Qt::PartiallyChecked)
                 {
--- a/ElaWidgetTools/ElaAcrylicUrlCard.cpp	2026-08-22 10:52:35.925593792 -0700
+++ b/ElaWidgetTools/ElaAcrylicUrlCard.cpp	2026-08-22 10:54:53.632525775 -0700
@@ -125,6 +125,6 @@
     iconFont.setPixelSize(eApp->getFontPixelSize());
     painter.setFont(iconFont);
     painter.setPen(ElaThemeColor(d->_themeMode, BasicText));
-    painter.drawText(width - 1.5 * iconFont.pixelSize(), height() - iconFont.pixelSize(), QChar(ElaIconType::UpRightFromSquare));
+    painter.drawText(width - 1.5 * iconFont.pixelSize(), height() - iconFont.pixelSize(), QChar(static_cast<char16_t>(ElaIconType::UpRightFromSquare)));
     painter.restore();
 }
--- a/ElaWidgetTools/ElaCalendarPicker.cpp	2026-08-22 10:52:35.925711547 -0700
+++ b/ElaWidgetTools/ElaCalendarPicker.cpp	2026-08-22 10:54:53.632011687 -0700
@@ -78,6 +78,6 @@
     QFont iconFont = QFont("ElaAwesome");
     iconFont.setPixelSize(eApp->getFontPixelSize() + 4);
     painter.setFont(iconFont);
-    painter.drawText(QRect(baseRect.right() - 25, 0, 15, height()), Qt::AlignVCenter | Qt::AlignRight, QChar(ElaIconType::CalendarRange));
+    painter.drawText(QRect(baseRect.right() - 25, 0, 15, height()), Qt::AlignVCenter | Qt::AlignRight, QChar(static_cast<char16_t>(ElaIconType::CalendarRange)));
     painter.restore();
 }
--- a/ElaWidgetTools/ElaIconButton.cpp	2026-08-22 10:52:35.925950266 -0700
+++ b/ElaWidgetTools/ElaIconButton.cpp	2026-08-22 10:54:53.631481329 -0700
@@ -63,7 +63,7 @@
     iconFont.setPixelSize(eApp->getFontPixelSize() + 2);
     this->setFont(iconFont);
     d->_pAwesome = awesome;
-    this->setText(QChar(awesome));
+    this->setText(QChar(static_cast<char16_t>(awesome)));
     connect(this, &ElaIconButton::pIsSelectedChanged, this, [=]() {
         update();
     });
@@ -92,7 +92,7 @@
     iconFont.setPixelSize(pixelSize);
     this->setFont(iconFont);
     d->_pAwesome = awesome;
-    this->setText(QChar(awesome));
+    this->setText(QChar(static_cast<char16_t>(awesome)));
     connect(this, &ElaIconButton::pIsSelectedChanged, this, [=]() {
         update();
     });
@@ -121,7 +121,7 @@
     iconFont.setPixelSize(pixelSize);
     this->setFont(iconFont);
     d->_pAwesome = awesome;
-    this->setText(QChar(awesome));
+    this->setText(QChar(static_cast<char16_t>(awesome)));
     this->setFixedSize(fixedWidth, fixedHeight);
     connect(this, &ElaIconButton::pIsSelectedChanged, this, [=]() {
         update();
@@ -139,7 +139,7 @@
 {
     Q_D(ElaIconButton);
     d->_pAwesome = awesome;
-    this->setText(QChar(awesome));
+    this->setText(QChar(static_cast<char16_t>(awesome)));
 }
 
 ElaIconType::IconName ElaIconButton::getAwesome() const
@@ -238,7 +238,7 @@
         painter.setPen(isEnabled() ? d->_themeMode == ElaThemeType::Light ? underMouse() ? d->_pLightHoverIconColor : d->_pLightIconColor : underMouse() ? d->_pDarkHoverIconColor
                                                                                                                                                          : d->_pDarkIconColor
                                    : ElaThemeColor(d->_themeMode, BasicTextDisable));
-        painter.drawText(rect(), Qt::AlignCenter, QChar(d->_pAwesome));
+        painter.drawText(rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(d->_pAwesome)));
     }
     painter.restore();
 }
--- a/ElaWidgetTools/ElaIcon.cpp	2026-08-22 10:52:35.925932613 -0700
+++ b/ElaWidgetTools/ElaIcon.cpp	2026-08-22 10:54:53.631548894 -0700
@@ -24,7 +24,7 @@
     // painter.setBrush(QColor("#1570A5"));
     iconFont.setPixelSize(eApp->getFontPixelSize() + 12);
     painter.setFont(iconFont);
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
@@ -41,7 +41,7 @@
     // painter.setBrush(QColor("#1570A5"));
     iconFont.setPixelSize(eApp->getFontPixelSize() + 12);
     painter.setFont(iconFont);
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
@@ -57,7 +57,7 @@
     iconFont.setPixelSize(pixelSize);
     painter.setFont(iconFont);
     // 画图形字体
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
@@ -74,7 +74,7 @@
     iconFont.setPixelSize(pixelSize);
     painter.setFont(iconFont);
     // 画图形字体
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
@@ -90,7 +90,7 @@
     iconFont.setPixelSize(pixelSize);
     painter.setFont(iconFont);
     // 画图形字体
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
@@ -107,7 +107,7 @@
     iconFont.setPixelSize(pixelSize);
     painter.setFont(iconFont);
     // 画图形字体
-    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(awesome));
+    painter.drawText(pix.rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(awesome)));
     painter.end();
     return QIcon(pix);
 }
--- a/ElaWidgetTools/ElaKeyBinder.cpp	2026-08-22 10:52:35.925998207 -0700
+++ b/ElaWidgetTools/ElaKeyBinder.cpp	2026-08-22 10:54:53.631346958 -0700
@@ -119,7 +119,7 @@
     painter.setPen(ElaThemeColor(d->_themeMode, BasicText));
     QRect iconRect = rect();
     iconRect.adjust(0, 0, -10, 0);
-    painter.drawText(iconRect, Qt::AlignVCenter | Qt::AlignRight, QChar(ElaIconType::Pencil));
+    painter.drawText(iconRect, Qt::AlignVCenter | Qt::AlignRight, QChar(static_cast<char16_t>(ElaIconType::Pencil)));
     painter.restore();
     QLabel::paintEvent(event);
 }
\ 文件末尾没有换行符
--- a/ElaWidgetTools/ElaMenuBar.cpp	2026-08-22 10:52:35.926093729 -0700
+++ b/ElaWidgetTools/ElaMenuBar.cpp	2026-08-22 10:54:53.631088181 -0700
@@ -64,7 +64,7 @@
 {
     ElaMenu* menu = new ElaMenu(title, this);
     menu->setMenuItemHeight(27);
-    menu->menuAction()->setProperty("ElaIconType", QChar(icon));
+    menu->menuAction()->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     menu->menuAction()->setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
     QMenuBar::addAction(menu->menuAction());
     return menu;
@@ -73,7 +73,7 @@
 QAction* ElaMenuBar::addElaIconAction(ElaIconType::IconName icon, const QString& text)
 {
     QAction* action = new QAction(text, this);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     action->setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
     QMenuBar::addAction(action);
     return action;
@@ -83,7 +83,7 @@
 {
     QAction* action = new QAction(text, this);
     action->setShortcut(shortcut);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     action->setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
     QMenuBar::addAction(action);
     return action;
--- a/ElaWidgetTools/ElaMenu.cpp	2026-08-22 10:52:35.926075666 -0700
+++ b/ElaWidgetTools/ElaMenu.cpp	2026-08-22 10:54:53.631150678 -0700
@@ -71,14 +71,14 @@
 {
     ElaMenu* menu = new ElaMenu(title, this);
     QMenu::addAction(menu->menuAction());
-    menu->menuAction()->setProperty("ElaIconType", QChar(icon));
+    menu->menuAction()->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     return menu;
 }
 
 QAction* ElaMenu::addElaIconAction(ElaIconType::IconName icon, const QString& text)
 {
     QAction* action = new QAction(text, this);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     QMenu::addAction(action);
     return action;
 }
@@ -87,7 +87,7 @@
 {
     QAction* action = new QAction(text, this);
     action->setShortcut(shortcut);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     QMenu::addAction(action);
     return action;
 }
--- a/ElaWidgetTools/ElaMultiSelectComboBox.cpp	2026-08-22 10:52:35.926187321 -0700
+++ b/ElaWidgetTools/ElaMultiSelectComboBox.cpp	2026-08-22 10:54:53.630929263 -0700
@@ -200,7 +200,7 @@
         painter.translate(expandIconRect.x() + (qreal)expandIconRect.width() / 2 - 2, expandIconRect.y() + (qreal)expandIconRect.height() / 2);
         painter.rotate(d->_pExpandIconRotate);
         painter.translate(-expandIconRect.x() - (qreal)expandIconRect.width() / 2 + 2, -expandIconRect.y() - (qreal)expandIconRect.height() / 2);
-        painter.drawText(expandIconRect, Qt::AlignVCenter, QChar(ElaIconType::AngleDown));
+        painter.drawText(expandIconRect, Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::AngleDown)));
         painter.restore();
     }
 }
--- a/ElaWidgetTools/ElaText.cpp	2026-08-22 10:52:35.926672733 -0700
+++ b/ElaWidgetTools/ElaText.cpp	2026-08-22 10:54:53.629906144 -0700
@@ -170,7 +170,7 @@
         iconFont.setPixelSize(this->font().pixelSize());
         painter.setFont(iconFont);
         painter.setPen(ElaThemeColor(d->_themeMode, BasicText));
-        painter.drawText(rect(), Qt::AlignCenter, QChar(d->_pElaIcon));
+        painter.drawText(rect(), Qt::AlignCenter, QChar(static_cast<char16_t>(d->_pElaIcon)));
         painter.restore();
     }
     else
--- a/ElaWidgetTools/ElaToolBar.cpp	2026-08-22 10:52:35.926745359 -0700
+++ b/ElaWidgetTools/ElaToolBar.cpp	2026-08-22 10:54:53.629763348 -0700
@@ -78,7 +78,7 @@
 QAction* ElaToolBar::addElaIconAction(ElaIconType::IconName icon, const QString& text)
 {
     QAction* action = new QAction(text, this);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     action->setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
     addAction(action);
     return action;
@@ -88,7 +88,7 @@
 {
     QAction* action = new QAction(text, this);
     action->setShortcut(shortcut);
-    action->setProperty("ElaIconType", QChar(icon));
+    action->setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     action->setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
     addAction(action);
     return action;
--- a/ElaWidgetTools/ElaToolButton.cpp	2026-08-22 10:52:35.926760560 -0700
+++ b/ElaWidgetTools/ElaToolButton.cpp	2026-08-22 10:54:53.629704520 -0700
@@ -81,7 +81,7 @@
 
 void ElaToolButton::setElaIcon(ElaIconType::IconName icon)
 {
-    setProperty("ElaIconType", QChar(icon));
+    setProperty("ElaIconType", QChar(static_cast<char16_t>(icon)));
     setIcon(ElaIcon::getInstance()->getElaIcon(ElaIconType::Broom, 1));
 }
 
--- a/ElaWidgetTools/private/ElaMessageBarPrivate.cpp	2026-08-22 10:52:35.928995199 -0700
+++ b/ElaWidgetTools/private/ElaMessageBarPrivate.cpp	2026-08-22 10:54:53.633774209 -0700
@@ -341,7 +341,7 @@
     QFont iconFont = QFont("ElaAwesome");
     iconFont.setPixelSize(eApp->getFontPixelSize() - 1);
     painter->setFont(iconFont);
-    painter->drawText(_leftPadding, 0, q->width(), q->height(), Qt::AlignVCenter, QChar(ElaIconType::Check));
+    painter->drawText(_leftPadding, 0, q->width(), q->height(), Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::Check)));
     // 时间进度条绘制
     QPainterPath clipPath;
     clipPath.addRoundedRect(foregroundRect, _borderRadius, _borderRadius);
@@ -423,7 +423,7 @@
     QFont iconFont = QFont("ElaAwesome");
     iconFont.setPixelSize(eApp->getFontPixelSize());
     painter->setFont(iconFont);
-    painter->drawText(_leftPadding + 1, 0, q->width(), q->height(), Qt::AlignVCenter, QChar(ElaIconType::Xmark));
+    painter->drawText(_leftPadding + 1, 0, q->width(), q->height(), Qt::AlignVCenter, QChar(static_cast<char16_t>(ElaIconType::Xmark)));
     // 时间进度条绘制
     QPainterPath clipPath;
     clipPath.addRoundedRect(foregroundRect, _borderRadius, _borderRadius);
