diff --git a/src/writer.py b/src/writer.py
index a57c558..86b5dd4 100644
--- a/src/writer.py
+++ b/src/writer.py
@@ -324,9 +324,7 @@ def __write_csv(self, filename):
     def __convert_csv_to_fit(filename):
         dir_path = os.path.dirname(os.path.realpath(__file__))
         subprocess.call([
-            "java",
-            "-jar",
-            f"{dir_path}/../SDK/java/FitCSVTool.jar",
+            "FitCSVTool",
             "-c",
             f"{filename}.csv",
             f"{filename}.fit"
