diff --git a/Cargo.lock b/Cargo.lock
index 723cfde9c3..2e1d414771 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -281,7 +281,7 @@
  "toml 0.8.23",
  "tracing",
  "tracing-subscriber",
- "tracing-tree 0.4.0",
+ "tracing-tree",
  "which 7.0.3",
 ]
 
@@ -954,7 +954,7 @@
  "syn",
  "tracing",
  "tracing-subscriber",
- "tracing-tree 0.4.1",
+ "tracing-tree",
 ]
 
 [[package]]
@@ -2137,18 +2137,6 @@
 ]
 
 [[package]]
-name = "tracing-tree"
-version = "0.4.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "ac87aa03b6a4d5a7e4810d1a80c19601dbe0f8a837e9177f23af721c7ba7beec"
-dependencies = [
- "nu-ansi-term",
- "tracing-core",
- "tracing-log",
- "tracing-subscriber",
-]
-
-[[package]]
 name = "tree-sitter"
 version = "0.25.10"
 source = "registry+https://github.com/rust-lang/crates.io-index"
diff --git a/kani-compiler/Cargo.toml b/kani-compiler/Cargo.toml
index ec22f80066..17e3126555 100644
--- a/kani-compiler/Cargo.toml
+++ b/kani-compiler/Cargo.toml
@@ -10,7 +10,7 @@
 
 [dependencies]
 cbmc = { path = "../cprover_bindings", package = "cprover_bindings", optional = true }
-charon = { path = "../charon/charon", optional = true, default-features = false } 
+charon = { path = "../charon/charon", optional = true, default-features = false }
 clap = { version = "4.4.11", features = ["derive", "cargo"] }
 itertools = "0.14"
 kani_metadata = { path = "../kani_metadata" }
@@ -26,7 +26,7 @@
 syn = { version = "2.0.72", features = ["parsing", "extra-traits"] }
 tracing = {version = "0.1", features = ["max_level_trace", "release_max_level_debug"]}
 tracing-subscriber = {version = "0.3.20", features = ["env-filter", "json", "fmt"]}
-tracing-tree = "0.4.0"
+tracing-tree = { git = "https://github.com/Nadrieril/tracing-tree" }
 
 # Future proofing: enable backend dependencies using feature.
 [features]
