diff --git a/src/backends/greetd/GreetdManager.cpp b/src/backends/greetd/GreetdManager.cpp
index 113e5a4..448404e 100644
--- a/src/backends/greetd/GreetdManager.cpp
+++ b/src/backends/greetd/GreetdManager.cpp
@@ -99,7 +99,7 @@ bool GreetDLogin::startSession( QString baseCmd, QString type ) {
 
 
 QString GreetDLogin::getX11Session( QString base ) {
-    QString xinit( "xinit %1 -- /etc/X11/xinit/xserverrc :%2 vt%3 -keeptty -noreset -novtswitch -auth /tmp/Xauth.%4" );
+    QString xinit( "xinit /etc/X11/xinit/Xsession %1 -- /etc/X11/xinit/xserverrc :%2 vt%3 -keeptty -noreset -novtswitch -auth /tmp/Xauth.%4" );
 
     /* Arg2: Get the display */
     int display;
