diff --git a/js/inputbar.js b/js/inputbar.js
index ceb2265..3201c4d 100644
--- a/js/inputbar.js
+++ b/js/inputbar.js
@@ -322,38 +322,6 @@ weechat.directive('inputBar', function() {
                     }
                 }
 
-                // Left Alt+[0-9] -> jump to buffer
-                if ($event.altKey && !$event.ctrlKey && (code > 47 && code < 58) && settings.enableQuickKeys) {
-                    if (code === 48) {
-                        code = 58;
-                    }
-                    bufferNumber = code - 48 - 1 ;
-
-                    // quick select filtered entries
-                    if (($scope.$parent.search.length || $scope.$parent.onlyUnread) && $scope.$parent.filteredBuffers.length) {
-                        filteredBufferNum = $scope.$parent.filteredBuffers[bufferNumber];
-                        if (filteredBufferNum !== undefined) {
-                            activeBufferId = [filteredBufferNum.number, filteredBufferNum.id];
-                        }
-                    } else {
-                        // Map the buffers to only their numbers and IDs so we don't have to
-                        // copy the entire (possibly very large) buffer object, and then sort
-                        // the buffers according to their WeeChat number
-                        sortedBuffers = _.map(models.getBuffers(), function(buffer) {
-                            return [buffer.number, buffer.id];
-                        }).sort(function(left, right) {
-                            // By default, Array.prototype.sort() sorts alphabetically.
-                            // Pass an ordering function to sort by first element.
-                            return left[0] - right[0];
-                        });
-                        activeBufferId = sortedBuffers[bufferNumber];
-                    }
-                    if (activeBufferId) {
-                        $scope.$parent.setActiveBuffer(activeBufferId[1]);
-                        $event.preventDefault();
-                    }
-                }
-
                 // Tab -> nick completion
                 if (code === 9 && !$event.altKey && !$event.ctrlKey) {
                     $event.preventDefault();
