diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-06-24 18:36:18.000829507 +0200
+++ b/CMakeLists.txt	2020-06-24 18:36:15.240828648 +0200
@@ -161,7 +161,7 @@
 include_directories("${PROJECT_SOURCE_DIR}/lib/bullet/src")
 
 # Find system ENet library or build it if missing
-if((UNIX AND NOT APPLE) AND USE_SYSTEM_ENET AND NOT USE_IPV6)
+if((UNIX AND NOT APPLE) AND USE_SYSTEM_ENET)
     pkg_check_modules(ENET libenet>=1.3.4)
 endif()
 
@@ -169,7 +169,7 @@
    add_definitions(-DENABLE_IPV6)
 endif()
 
-if(ENET_FOUND AND NOT USE_IPV6)
+if(ENET_FOUND)
     include_directories(${ENET_INCLUDE_DIRS})
 else()
     # Fallback to built-in version
@@ -592,7 +592,9 @@
     ${Angelscript_LIBRARIES}
     ${CURL_LIBRARIES}
     mcpp
-    )
+)
+
+
 
 if (USE_SQLITE3)
     target_link_libraries(supertuxkart ${SQLITE3_LIBRARY})
