diff --color -ur a/lib/irrlicht/source/Irrlicht/COSOperator.cpp b/lib/irrlicht/source/Irrlicht/COSOperator.cpp
--- a/lib/irrlicht/source/Irrlicht/COSOperator.cpp	2020-11-20 11:57:33.434090245 +0100
+++ b/lib/irrlicht/source/Irrlicht/COSOperator.cpp	2020-11-20 12:03:51.098961084 +0100
@@ -14,7 +14,7 @@
 #if !defined(_IRR_SOLARIS_PLATFORM_) && !defined(__CYGWIN__)
 #include <sys/param.h>
 #include <sys/types.h>
-#if defined(ANDROID) || (defined(__linux__) && !defined(__GLIBC__))
+#if defined(ANDROID) || defined(__linux__)
 #include <linux/sysctl.h>
 #else
 #include <sys/sysctl.h>
