diff --git a/vendor/adb/adb_utils.cpp b/vendor/adb/adb_utils.cpp
index 593448a..4506a31 100644
--- a/vendor/adb/adb_utils.cpp
+++ b/vendor/adb/adb_utils.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cstdlib>
 #define TRACE_TAG ADB
 
 #include "adb_utils.h"
@@ -308,8 +309,13 @@ std::string adb_get_homedir_path() {
 }
 
 std::string adb_get_android_dir_path() {
-    std::string user_dir = adb_get_homedir_path();
-    std::string android_dir = user_dir + OS_PATH_SEPARATOR + ".android";
+    std::string android_dir = getenv("ANDROID_USER_HOME");
+
+    if (android_dir.empty()) {
+        std::string user_dir = adb_get_homedir_path();
+        android_dir.assign(user_dir + OS_PATH_SEPARATOR + ".android");
+    }
+
     struct stat buf;
     if (stat(android_dir.c_str(), &buf) == -1) {
         if (adb_mkdir(android_dir, 0750) == -1) {
