diff --git i/tilecloud/scripts/tc_viewer.py w/tilecloud/scripts/tc_viewer.py
index 1c5e6056..8cabfd10 100755
--- i/tilecloud/scripts/tc_viewer.py
+++ w/tilecloud/scripts/tc_viewer.py
@@ -112,12 +112,12 @@ def openwebglobe():
 
 @bottle.route("/favicon.ico")
 def favicon():
-    return bottle.static_file("favicon.ico", root="static")
+    return bottle.static_file("favicon.ico", root="@site_packages@/static")
 
 
 @bottle.route("/static/<path:re:.*>")
 def static(path):
-    return bottle.static_file(path, root="static")
+    return bottle.static_file(path, root="@site_packages@/static")
 
 
 @bottle.route("/")
