diff --git a/external/upstream/libint2/CMakeLists.txt b/external/upstream/libint2/CMakeLists.txt
index 212b404da..d599165a5 100644
--- a/external/upstream/libint2/CMakeLists.txt
+++ b/external/upstream/libint2/CMakeLists.txt
@@ -73,6 +73,7 @@ else()
 
         ExternalProject_Add(libint2_external
             URL ${_url_l2_tarball}
+            URL_HASH MD5=81cf42053d29b371678a3ac52d8056f8
             CMAKE_ARGS -GNinja
                        -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
                        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
