diff --git i/maprec/map_record.py w/maprec/map_record.py
index 2a58b0f..d1ce585 100644
--- i/maprec/map_record.py
+++ w/maprec/map_record.py
@@ -161,7 +161,7 @@ class Maprecord(object):
             data['image_size'] = os.path.getsize(self.image_path)
             data['image_mtime'] = os.path.getmtime(self.image_path)
             data = json.dumps(data)
-            self._fingerprint = hashlib.sha1(data).hexdigest()
+            self._fingerprint = hashlib.sha1(str(data).encode('utf-8')).hexdigest()
         return self._fingerprint
 
 #def write(filename, image_path, srs, gcps, cutline=None, image_path_relative=True):
