diff --git a/drivers/gpu/drm/i915/display/intel_panel.c b/drivers/gpu/drm/i915/display/intel_panel.c
index d1d1b59102d693..509b11d7f0cf28 100644
--- a/drivers/gpu/drm/i915/display/intel_panel.c
+++ b/drivers/gpu/drm/i915/display/intel_panel.c
@@ -228,7 +228,7 @@ static void intel_panel_destroy_probed_modes(struct intel_connector *connector)
 void intel_panel_add_edid_fixed_modes(struct intel_connector *connector, bool has_drrs)
 {
 	intel_panel_add_edid_preferred_mode(connector);
-	if (intel_panel_preferred_fixed_mode(connector) && has_drrs)
+	if (intel_panel_preferred_fixed_mode(connector)/* && has_drrs*/)
 		intel_panel_add_edid_alt_fixed_modes(connector);
 	intel_panel_destroy_probed_modes(connector);
 }
