diff --git a/drivers/gpu/drm/i915/display/intel_panel.c b/drivers/gpu/drm/i915/display/intel_panel.c
index a0c8e43db5eb..ebb434fac481 100644
--- a/drivers/gpu/drm/i915/display/intel_panel.c
+++ b/drivers/gpu/drm/i915/display/intel_panel.c
@@ -134,31 +134,21 @@ intel_panel_edid_fixed_mode(struct intel_connector *connector)
 	struct drm_i915_private *dev_priv = to_i915(connector->base.dev);
 	const struct drm_display_mode *scan;
 	struct drm_display_mode *fixed_mode;
+	const struct drm_display_mode *scan_max = NULL;
 
 	if (list_empty(&connector->base.probed_modes))
 		return NULL;
 
 	/* prefer fixed mode from EDID if available */
 	list_for_each_entry(scan, &connector->base.probed_modes, head) {
-		if ((scan->type & DRM_MODE_TYPE_PREFERRED) == 0)
-			continue;
-
-		fixed_mode = drm_mode_duplicate(&dev_priv->drm, scan);
-		if (!fixed_mode)
-			return NULL;
-
-		drm_dbg_kms(&dev_priv->drm,
-			    "[CONNECTOR:%d:%s] using preferred mode from EDID: ",
-			    connector->base.base.id, connector->base.name);
-		drm_mode_debug_printmodeline(fixed_mode);
-
-		return fixed_mode;
+		if (!scan_max || scan->clock > scan_max->clock)
+			scan_max = scan;
 	}
 
-	scan = list_first_entry(&connector->base.probed_modes,
-				typeof(*scan), head);
+	if (!scan_max)
+		return NULL;
 
-	fixed_mode = drm_mode_duplicate(&dev_priv->drm, scan);
+	fixed_mode = drm_mode_duplicate(&dev_priv->drm, scan_max);
 	if (!fixed_mode)
 		return NULL;
 
