diff --git a/source/mqJucePlugin/PluginProcessor.cpp b/source/mqJucePlugin/PluginProcessor.cpp
index 57d15da..2a49b05 100644
--- a/source/mqJucePlugin/PluginProcessor.cpp
+++ b/source/mqJucePlugin/PluginProcessor.cpp
@@ -11,10 +11,14 @@ namespace
 	juce::PropertiesFile::Options getOptions()
 	{
 		juce::PropertiesFile::Options opts;
-		opts.applicationName = "DSP56300EmulatorVavra";
-		opts.filenameSuffix = ".settings";
-		opts.folderName = "DSP56300EmulatorVavra";
-		opts.osxLibrarySubFolder = "Application Support/DSP56300EmulatorVavra";
+		opts.applicationName = "vavra";
+		opts.filenameSuffix = "settings";
+#ifdef JUCE_LINUX 
+  	opts.folderName = ".config/The Usual Suspects/Vavra";
+#else
+  	opts.folderName = "The Usual Suspects/Vavra";
+#endif
+		opts.osxLibrarySubFolder = "Application Support";
 		return opts;
 	}
 }
diff --git a/source/nord/n2x/n2xJucePlugin/n2xPluginProcessor.cpp b/source/nord/n2x/n2xJucePlugin/n2xPluginProcessor.cpp
index 20bbb07..5f8717f 100644
--- a/source/nord/n2x/n2xJucePlugin/n2xPluginProcessor.cpp
+++ b/source/nord/n2x/n2xJucePlugin/n2xPluginProcessor.cpp
@@ -14,10 +14,14 @@ namespace
 	juce::PropertiesFile::Options getOptions()
 	{
 		juce::PropertiesFile::Options opts;
-		opts.applicationName = "DSP56300EmulatorNodalRed";
-		opts.filenameSuffix = ".settings";
-		opts.folderName = "DSP56300EmulatorNodalRed";
-		opts.osxLibrarySubFolder = "Application Support/DSP56300EmulatorNodalRed";
+		opts.applicationName = "n2x";
+		opts.filenameSuffix = "settings";
+#ifdef JUCE_LINUX 
+  	opts.folderName = ".config/The Usual Suspects/NodalRed2x";
+#else
+  	opts.folderName = "The Usual Suspects/NodalRed2x";
+#endif
+		opts.osxLibrarySubFolder = "Application Support";
 		return opts;
 	}
 }
diff --git a/source/osTIrusJucePlugin/OsTIrusProcessor.cpp b/source/osTIrusJucePlugin/OsTIrusProcessor.cpp
index 8dd4c99..bc3d431 100644
--- a/source/osTIrusJucePlugin/OsTIrusProcessor.cpp
+++ b/source/osTIrusJucePlugin/OsTIrusProcessor.cpp
@@ -9,10 +9,14 @@ namespace
 	juce::PropertiesFile::Options getConfigOptions()
 	{
 		juce::PropertiesFile::Options opts;
-		opts.applicationName = "DSP56300Emulator_OsTIrus";
-		opts.filenameSuffix = ".settings";
-		opts.folderName = "DSP56300Emulator_OsTIrus";
-		opts.osxLibrarySubFolder = "Application Support/DSP56300Emulator_OsTIrus";
+		opts.applicationName = "ostirus";
+		opts.filenameSuffix = "settings";
+#ifdef JUCE_LINUX 
+  	opts.folderName = ".config/The Usual Suspects/OsTIrus";
+#else
+  	opts.folderName = "The Usual Suspects/OsTIrus";
+#endif
+		opts.osxLibrarySubFolder = "Application Support";
 		return opts;
 	}
 }
diff --git a/source/osirusJucePlugin/OsirusProcessor.cpp b/source/osirusJucePlugin/OsirusProcessor.cpp
index d4e6ba5..f0469f3 100644
--- a/source/osirusJucePlugin/OsirusProcessor.cpp
+++ b/source/osirusJucePlugin/OsirusProcessor.cpp
@@ -9,10 +9,14 @@ namespace
 	juce::PropertiesFile::Options getConfigOptions()
 	{
 		juce::PropertiesFile::Options opts;
-		opts.applicationName = "DSP56300 Emulator";
-		opts.filenameSuffix = ".settings";
-		opts.folderName = "DSP56300 Emulator";
-		opts.osxLibrarySubFolder = "Application Support/DSP56300 Emulator";
+		opts.applicationName = "osirus";
+		opts.filenameSuffix = "settings";
+#ifdef JUCE_LINUX 
+  	opts.folderName = ".config/The Usual Suspects/Osirus";
+#else
+  	opts.folderName = "The Usual Suspects/Osirus";
+#endif
+		opts.osxLibrarySubFolder = "Application Support";
 		return opts;
 	}
 }
diff --git a/source/xtJucePlugin/PluginProcessor.cpp b/source/xtJucePlugin/PluginProcessor.cpp
index 4b2aa2b..d6389a8 100644
--- a/source/xtJucePlugin/PluginProcessor.cpp
+++ b/source/xtJucePlugin/PluginProcessor.cpp
@@ -13,10 +13,14 @@ namespace
 	juce::PropertiesFile::Options getOptions()
 	{
 		juce::PropertiesFile::Options opts;
-		opts.applicationName = "DSP56300EmulatorXenia";
-		opts.filenameSuffix = ".settings";
-		opts.folderName = "DSP56300EmulatorXenia";
-		opts.osxLibrarySubFolder = "Application Support/DSP56300EmulatorXenia";
+		opts.applicationName = "xenia";
+		opts.filenameSuffix = "settings";
+#ifdef JUCE_LINUX 
+  	opts.folderName = ".config/The Usual Suspects/Xenia";
+#else
+  	opts.folderName = "The Usual Suspects/Xenia";
+#endif
+		opts.osxLibrarySubFolder = "Application Support";
 		return opts;
 	}
 }
