diff --git a/Telegram/SourceFiles/data/data_stories.cpp b/Telegram/SourceFiles/data/data_stories.cpp
index 0d9671f53..b44176761 100644
--- a/Telegram/SourceFiles/data/data_stories.cpp
+++ b/Telegram/SourceFiles/data/data_stories.cpp
@@ -95,9 +95,9 @@ std::vector<StoryId> RespectingPinned(const StoriesIds &ids) {
 StoriesSourceInfo StoriesSource::info() const {
 	return {
 		.id = peer->id,
-		.last = ids.empty() ? 0 : ids.back().date,
-		.count = uint32(std::min(int(ids.size()), kMaxSegmentsCount)),
-		.unreadCount = uint32(std::min(unreadCount(), kMaxSegmentsCount)),
+		.last = 0,
+		.count = 0,
+		.unreadCount = 0,
 		.premium = (peer->isUser() && peer->asUser()->isPremium()) ? 1U : 0,
 	};
 }
