diff --git a/CMakeLists.txt b/CMakeLists.txt
index a4071fd9b..9fb7be99c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1166,7 +1166,7 @@ if (LINALG STREQUAL "MKL")
                 set (MKL_INCLUDE_PATH "${MKLROOT}/include" CACHE PATH
                         "location of MKL include files." FORCE)
                 if (ADDRMODE EQUAL 64)
-                        set (libpath "${MKLROOT}/lib/intel64")
+                        set (libpath "${MKLROOT}/lib")
                 elseif (ADDRMODE EQUAL 32)
                         set (libpath "${MKLROOT}/lib/ia32")
                 endif ()
