--- a/include/internal/common.h	2026-09-01 15:06:12.841791446 -0700
+++ b/include/internal/common.h	2026-09-01 15:06:12.843393183 -0700
@@ -82,7 +82,7 @@
 #ifndef OPENSSL_SYS_VMS
 #define X509_CERT_AREA OPENSSLDIR
 #define X509_CERT_DIR OPENSSLDIR "/certs"
-#define X509_CERT_FILE OPENSSLDIR "/cert.pem"
+#define X509_CERT_FILE "/etc/ssl/certs/ca-certificates.crt"
 #define X509_PRIVATE_DIR OPENSSLDIR "/private"
 #define CTLOG_FILE OPENSSLDIR "/ct_log_list.cnf"
 #else
