diff --git a/internal/rpc/system.go b/internal/rpc/system.go
--- a/internal/rpc/system.go
+++ b/internal/rpc/system.go
@@ -2,8 +2,9 @@ package rpc

 import (
 	"context"
 	"os"
 	"runtime"
+	"golang.org/x/sys/unix"

 	"ThroneCore/gen"
 )
@@ -15,5 +15,26 @@ func (s *server) IsPrivileged(ctx context.Context, _ *gen.EmptyReq) (*gen.IsPriv
 		}, nil
 	}

+	if runtime.GOOS == "linux" {
+		caps := unix.CapUserHeader{
+			Version: unix.LINUX_CAPABILITY_VERSION_3,
+			Pid:     0, // current
+		}
+
+		var data [2]unix.CapUserData
+		err := unix.Capget(&caps, &data[0])
+
+		if err != nil {
+			return &gen.IsPrivilegedResponse{
+				HasPrivilege: To(false),
+			}, nil
+		}
+
+		// CAP_NET_ADMIN = 12
+		return &gen.IsPrivilegedResponse{
+			HasPrivilege: To((data[0].Effective & (1 << unix.CAP_NET_ADMIN)) != 0),
+		}, nil
+	}
+
 	return &gen.IsPrivilegedResponse{HasPrivilege: To(os.Geteuid() == 0)}, nil
 }
