diff --git a/src/tools/ga-5.6.5/armci/src/devices/mpi-spawn/mpi2_client.c b/src/tools/ga-5.6.5/armci/src/devices/mpi-spawn/mpi2_client.c
index ff2bb6b2..57e7f688 100644
--- a/src/tools/ga-5.6.5/armci/src/devices/mpi-spawn/mpi2_client.c
+++ b/src/tools/ga-5.6.5/armci/src/devices/mpi-spawn/mpi2_client.c
@@ -167,7 +167,7 @@ void armci_mpi_strided2(int op, void *ptr, int stride_levels, int stride_arr[],
     for(i=1; i<=stride_levels; i++) 
     {
        stride *= stride_arr[i-1];
-       MPI_Check( MPI_Type_hvector(count[i], count[i-1], stride,
+       MPI_Check( MPI_Type_create_hvector(count[i], count[i-1], stride,
                                   type[i-1], &type[i]) );
     }
     MPI_Check( MPI_Type_commit(&type[stride_levels]) );
diff --git a/src/tools/ga-5.6.5/comex/src-armci/message.c b/src/tools/ga-5.6.5/comex/src-armci/message.c
index a845b9ec..b8a84dd0 100644
--- a/src/tools/ga-5.6.5/comex/src-armci/message.c
+++ b/src/tools/ga-5.6.5/comex/src-armci/message.c
@@ -262,7 +262,7 @@ void armci_msg_sel_scope(int scope, void *x, int n, char* op, int type, int cont
         disp[1] = sizeof(long long);
         block[0] = 1;
         block[1] = 1;
-        MPI_Type_struct(2, block, disp, type, &MPI_LONGLONG_INT);
+        MPI_Type_create_struct(2, block, disp, type, &MPI_LONGLONG_INT);
     }
 
     if (strncmp(op, "min", 3) == 0) {
diff --git a/src/tools/ga-5.6.5/tcgmsg/tcgmsg-mpi/misc.c b/src/tools/ga-5.6.5/tcgmsg/tcgmsg-mpi/misc.c
index cc90c8bd..7c12bbe0 100644
--- a/src/tools/ga-5.6.5/tcgmsg/tcgmsg-mpi/misc.c
+++ b/src/tools/ga-5.6.5/tcgmsg/tcgmsg-mpi/misc.c
@@ -142,7 +142,7 @@ void tcgi_alt_pbegin(int *argc, char **argv[])
 #else
         MPI_Init(argc, argv);
 #endif
-        MPI_Errhandler_set(TCGMSG_Comm, MPI_ERRORS_RETURN);
+        MPI_Comm_set_errhandler(TCGMSG_Comm, MPI_ERRORS_RETURN);
     }
 
     MPI_Comm_size(TCGMSG_Comm, &numprocs);
