diff --git a/library/log.py b/library/log.py
index bc11142..435d317 100644
--- a/library/log.py
+++ b/library/log.py
@@ -27,7 +27,6 @@ locale.setlocale(locale.LC_ALL, '')
 logging.basicConfig(  # format='%(asctime)s [%(levelname)s] %(message)s in %(pathname)s:%(lineno)d',
     format="%(asctime)s [%(levelname)s] %(message)s",
     handlers=[
-        RotatingFileHandler("log.log", maxBytes=1000000, backupCount=0),  # Log in textfile max 1MB
         logging.StreamHandler()  # Log also in console
     ],
     datefmt='%x %X')
