--- a/libtransmission/file-posix.cc	2026-04-24 23:42:47.224544795 -0700
+++ b/libtransmission/file-posix.cc	2026-04-24 23:43:01.597314700 -0700
@@ -262,6 +262,45 @@
     return std::empty(path) ? "/"sv : path;
 }
 
+bool tr_sys_path_limit_filename(std::string& subpath, std::string_view suffix, size_t extra)
+{
+    constexpr size_t kMaxExt = 9; // Includes dot. Keeps ".backup", ".original".
+
+    auto basename_pos = subpath.find_last_of('/');
+    if (basename_pos == std::string::npos) basename_pos = 0;
+    else if (basename_pos == subpath.size() - 1) return false; // subpath ends with '/', so it's not a file.
+    else ++basename_pos;
+
+    if (auto basename_len = subpath.size() - basename_pos; basename_len + extra < 256) return false;
+
+    auto ext_pos = subpath.find_last_of('.');
+    if (ext_pos != std::string::npos)
+    {
+        if (ext_pos <= basename_pos)
+        {
+            ext_pos = std::string::npos;
+        }
+        else if (auto ext_len = subpath.size() - ext_pos; ext_len > kMaxExt)
+        {
+            ext_pos = std::string::npos;
+        }
+    }
+    auto const ext = ext_pos == std::string::npos ? ""s : std::string{ subpath.cbegin() + ext_pos, subpath.cend() };
+
+    auto cut_pos = basename_pos + 255 - ext.size() - suffix.size() - extra;
+    for (auto iter = subpath.cbegin() + cut_pos; iter > subpath.cbegin() + basename_pos; --iter)
+    {
+        if ((*iter & 0xC0) == 0x80) --cut_pos;
+        else break;
+    }
+
+    subpath.erase(cut_pos);
+    subpath.append(suffix);
+    subpath.append(ext);
+
+    return true;
+}
+
 // This function is adapted from Node.js's path.posix.dirname() function,
 // which is copyrighted by Joyent, Inc. and other Node contributors
 // and is distributed under MIT (SPDX:MIT) license.
--- a/libtransmission/file-win32.cc	2026-04-24 23:42:47.224570524 -0700
+++ b/libtransmission/file-win32.cc	2026-04-24 23:43:05.491449841 -0700
@@ -559,6 +559,12 @@
 
 namespace
 {
+// Not implemented. May not need to at all. Return false to indicate subpath not changed.
+bool tr_sys_path_limit_filename(std::string& subpath, std::string_view suffix, size_t extra)
+{
+    return false;
+}
+
 [[nodiscard]] bool isWindowsDeviceRoot(char ch) noexcept
 {
     return isalpha(static_cast<int>(ch)) != 0;
--- a/libtransmission/file.h	2026-04-24 23:42:47.224609194 -0700
+++ b/libtransmission/file.h	2026-04-24 23:46:33.153306273 -0700
@@ -232,6 +232,22 @@
  */
 std::string_view tr_sys_path_basename(std::string_view path, tr_error* error = nullptr);
 
+
+/**
+* @brief Limit filename length to 255.
+*        Truncate basename at tail, append suffix, then append extension back.
+*        The resultant subpath length is limited to (255 - extra).
+*
+* @param[in,out] subpath Only the basename part may be changed.
+* @param[in] suffix Normally a file ID string to help avoid filename collision.
+* @param[in] extra Extra bytes to reserve, for ".part", or 0 if not needed.
+*
+* @return Whether subpath is changed.
+*         If subpath basename too long, then subpath is changed, return true.
+*         If basename length was less than (255 - extra), return false.
+*/
+bool tr_sys_path_limit_filename(std::string& subpath, std::string_view suffix, size_t extra);
+
 /**
  * @brief Portability wrapper for `dirname()`.
  *
--- a/libtransmission/inout.cc	2026-04-24 23:42:47.224699132 -0700
+++ b/libtransmission/inout.cc	2026-04-24 23:44:04.017759917 -0700
@@ -99,8 +99,17 @@
     auto err = ENOENT;
     if (writable)
     {
+        // When writable=true we can safely cast away const since this is called from tr_ioWrite
+        auto& mutable_tor = const_cast<tr_torrent&>(tor);
         auto const base = tor.current_dir();
         auto const suffix = session.isIncompleteFileNamingEnabled() ? tr_torrent_files::PartialFileSuffix : ""sv;
+        auto subpath = std::string{ tor.file_subpath(file_index) };
+        auto const idstr = fmt::format("_{:x}-{}", file_index, tor.info_hash_string());
+        if (tr_sys_path_limit_filename(subpath, idstr, suffix.size()))
+        {
+            mutable_tor.set_file_subpath(file_index, subpath);
+            mutable_tor.save_resume_file();
+        }
         auto const filename = tr_pathbuf{ base, '/', tor.file_subpath(file_index), suffix };
         if (auto const fd = open_files.get(tor_id, file_index, writable, filename, prealloc, file_size); fd)
         {
