diff --git a/bcml/__main__.py b/bcml/__main__.py
index 5be1a56..c30b94d 100644
--- a/bcml/__main__.py
+++ b/bcml/__main__.py
@@ -3,7 +3,6 @@ import sys
 from contextlib import redirect_stderr, redirect_stdout
 from datetime import datetime
 from multiprocessing import set_start_method, Process
-from os import chmod  # pylint: disable=ungrouped-imports
 from pathlib import Path
 from random import randint
 from subprocess import Popen, DEVNULL
@@ -62,8 +61,6 @@ def main(debug: bool = False):
 
     try:
         if SYSTEM != "Windows":
-            chmod(util.get_exec_dir() / "helpers/msyt", int("755", 8))
-            chmod(util.get_exec_dir() / "helpers/7z", int("755", 8))
             os.setpgrp()
         LOG.parent.mkdir(parents=True, exist_ok=True)
         for folder in util.get_work_dir().glob("*"):
diff --git a/bcml/util.py b/bcml/util.py
index 20d4998..a1d3cb2 100644
--- a/bcml/util.py
+++ b/bcml/util.py
@@ -1390,7 +1390,6 @@ def get_7z_path():
     if system() == "Windows":
         return str(get_exec_dir() / "helpers" / "7z.exe")
     bundle_path = get_exec_dir() / "helpers" / "7z"
-    os.chmod(bundle_path, 0o755)
     if get_settings("force_7z"):
         return str(bundle_path)
     return shutil.which("7z") or str(bundle_path)
