diff --git a/src/rvthtool/verify.cpp b/src/rvthtool/verify.cpp
index 818c7af..ac2df43 100644
--- a/src/rvthtool/verify.cpp
+++ b/src/rvthtool/verify.cpp
@@ -96,13 +96,13 @@ static bool progress_callback(const RvtH_Verify_Progress_State *state, void *use
 				case RVTH_VERIFY_ERROR_BAD_HASH:
 					printf("\n*** ERROR: Pt%u [%u,%u,%u%s]: H%u hash is invalid.\n",
 						state->pt_current, state->group_cur,
-						state->sector / 8, state->sector,
+						state->sector / 8U, state->sector,
 						s_kb, state->hash_level);
 					break;
 				case RVTH_VERIFY_ERROR_TABLE_COPY:
 					printf("\n*** ERROR: Pt%u [%u,%u,%u%s]: H%u table copy doesn't match base sector.\n",
 						state->pt_current, state->group_cur,
-						state->sector / 8, state->sector,
+						state->sector / 8U, state->sector,
 						s_kb, state->hash_level);
 					break;
 			}
