diff --git a/gui/Makefile.am b/gui/Makefile.am
index ae678815..3e5bd1f7 100644
--- a/gui/Makefile.am
+++ b/gui/Makefile.am
@@ -31,7 +31,7 @@ bin_PROGRAMS = poke-gui
 poke_gui_SOURCES = poke-gui.c
 
 poke_gui_CPPFLAGS = -I$(top_builddir)/common -I$(top_srcdir)/gl-gui \
-                    $(TCL_INCLUDE_SPEC) $(TK_INCLUDE_SPEC) \
+                    $(TCL_INCLUDES) $(TK_INCLUDES) \
                     -DPOKEGUIDIR=\"$(guifilesdir)\" \
                     -DLOCALEDIR=\"$(localedir)\"
 poke_gui_CFLAGS = -Wall
