--- a/files/setup/autodesk_fusion_installer_x86-64.sh
+++ b/files/setup/autodesk_fusion_installer_x86-64.sh
@@ -163,7 +163,7 @@
 
     # If there are missing commands, install them
     if [ ${#MISSING_COMMANDS[@]} -gt 0 ]; then
-        install_required_packages
+        echo -e "${YELLOW}Skipping install_required_packages in Nix environment!${NOCOLOR}"
     else
         echo -e "${GREEN}All required commands are available!${NOCOLOR}"
     fi
@@ -371,11 +371,11 @@
             check_gpu_driver
             check_gpu_vram
             check_disk_space
-            download_files
-            check_and_install_wine
+            echo -e "${YELLOW}Skipping download_files in Nix environment!${NOCOLOR}"
+            echo -e "${YELLOW}Skipping check_and_install_wine in Nix environment!${NOCOLOR}"
             wine_autodesk_fusion_install
             autodesk_fusion_patch_qt6webenginecore
             autodesk_fusion_patch_siappdll
             wine_autodesk_fusion_install_extensions
-            autodesk_fusion_shortcuts_load
+            echo -e "${YELLOW}Skipping autodesk_fusion_shortcuts_load in Nix environment!${NOCOLOR}"
             autodesk_fusion_safe_logfile
