diff --git a/src/hydra-eval-jobs/hydra-eval-jobs.cc b/src/hydra-eval-jobs/hydra-eval-jobs.cc
index 2be3dd80..97188d5e 100644
--- a/src/hydra-eval-jobs/hydra-eval-jobs.cc
+++ b/src/hydra-eval-jobs/hydra-eval-jobs.cc
@@ -279,9 +279,8 @@ int main(int argc, char * * argv)
                 /* FIXME: The build hook in conjunction with import-from-derivation is causing "unexpected EOF" during eval */
                 settings.builders = "";
 
-                /* Prevent access to paths outside of the Nix search path and
-                   to the environment. */
-                evalSettings.restrictEval = true;
+                /* Turn off restrictEval until I have time to figure out how to make my builds work with it */
+                evalSettings.restrictEval = false;
 
                 if (releaseExpr == "") throw UsageError("no expression specified");
 
