--- a/i3lock.c	2021-02-15 18:42:59.766411664 +0700
+++ b/i3lock.c	2021-02-15 18:44:29.258395421 +0700
@@ -535,7 +535,7 @@
     /* Get state of Caps and Num lock modifiers, to be displayed in
      * STATE_AUTH_WRONG state */
     xkb_mod_index_t idx, num_mods;
-    const char *mod_name;
+    const char *mod_name = "";
 
     num_mods = xkb_keymap_num_mods(xkb_keymap);
 
@@ -543,10 +543,6 @@
         if (!xkb_state_mod_index_is_active(xkb_state, idx, XKB_STATE_MODS_EFFECTIVE))
             continue;
 
-        mod_name = xkb_keymap_mod_get_name(xkb_keymap, idx);
-        if (mod_name == NULL)
-            continue;
-
         /* Replace certain xkb names with nicer, human-readable ones. */
         if (strcmp(mod_name, XKB_MOD_NAME_CAPS) == 0)
             mod_name = "Caps Lock";
