--- a/src/tabbar.cpp
+++ b/src/tabbar.cpp
@@ -25,6 +25,7 @@
 
 TabBar::TabBar(QWidget *parent) :
     QTabBar(parent)
+    , mp_stackedWidget(nullptr)
 {
     QTabBar::setDrawBase(false);
     setTabsClosable(true);
@@ -251,7 +252,7 @@
 
 QSize TabBar::tabSizeHint(int index) const
 {
-    QWidget *w = mp_stackedWidget->widget(index);
+    QWidget *w = mp_stackedWidget ? mp_stackedWidget->widget(index) : nullptr;
 
     if ((w && qobject_cast<ContentManagerView*>(w)) || index >= count() - 1)
         return QSize(40, 40); // the "Library" tab is only icon
