From 681d8edad0d27f7cd25aac064c173968a08f304a Mon Sep 17 00:00:00 2001
From: Daniel Bonniot <dbonniot@chemaxon.com>
Date: Tue, 28 May 2019 17:48:39 +0200
Subject: [PATCH] Add more llvm intrinsics

---
 src/main.rs | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/main.rs b/src/main.rs
index e37f98b..e420d23 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -831,8 +831,15 @@ fn run() -> Result<i32, failure::Error> {
                     // XXX unclear whether these produce library calls on some platforms or not
                     if func.starts_with("llvm.bswap.")
                         | func.starts_with("llvm.ctlz.")
+                        | func.starts_with("llvm.cttz.")
+                        | func.starts_with("llvm.sadd.with.overflow.")
+                        | func.starts_with("llvm.smul.with.overflow.")
+                        | func.starts_with("llvm.ssub.with.overflow.")
                         | func.starts_with("llvm.uadd.with.overflow.")
                         | func.starts_with("llvm.umul.with.overflow.")
+                        | func.starts_with("llvm.usub.sat.")
+                        | func.starts_with("llvm.usub.with.overflow.")
+                        | func.starts_with("llvm.x86.sse2.pmovmskb.")
                     {
                         if !llvm_seen.contains(func) {
                             llvm_seen.insert(func);
