diff --git a/src/main.rs b/src/main.rs
index f377fcc..4145a25 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -135,20 +135,11 @@ fn run() -> Result<ExitStatus> {
                 "The XARGO_RUST_SRC env variable must be set and point to the \
                  Rust source directory when working with the 'dev' channel",
             )?,
-            Channel::Nightly => if let Some(src) = rustc::Src::from_env() {
+            _ => if let Some(src) = rustc::Src::from_env() {
                 src
             } else {
                 sysroot.src()?
             },
-            Channel::Stable | Channel::Beta => {
-                writeln!(
-                    io::stderr(),
-                    "WARNING: the sysroot can't be built for the {:?} channel. \
-                     Switch to nightly.",
-                    meta.channel
-                ).ok();
-                return cargo::run(&args, verbose);
-            }
         };
 
         let cmode = if let Some(triple) = args.target() {
