diff --git a/./original/basis_ffi.c b/./new/basis_ffi.c
index 644d1c1..d394a27 100644
--- a/basis_ffi.c
+++ b/basis_ffi.c
@@ -143,7 +143,7 @@ void ffiopen_out (unsigned char *c, long clen, unsigned char *a, long alen) {
   #ifdef EVAL
   int fd = open((const char *) c, O_RDWR|O_CREAT|O_TRUNC, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
   #else
-  int fd = open((const char *) c, O_RDWR|O_CREAT|O_TRUNC);
+  int fd = open((const char *) c, O_RDWR|O_CREAT|O_TRUNC, 0);
   #endif
   if (0 <= fd){
     a[0] = 0;
