diff --git i/go.mod w/go.mod
index d04dc07..3992fab 100644
--- i/go.mod
+++ w/go.mod
@@ -1,6 +1,6 @@
 module git.sr.ht/~hedy/gelim
 
-go 1.16
+go 1.18
 
 require (
 	git.sr.ht/~adnano/go-xdg v0.1.0
@@ -12,3 +12,10 @@ require (
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/term v0.0.0-20210429154555-c04ba851c2a4
 )
+
+require (
+	github.com/mattn/go-colorable v0.1.13 // indirect
+	github.com/mattn/go-isatty v0.0.17 // indirect
+	github.com/mattn/go-runewidth v0.0.3 // indirect
+	golang.org/x/sys v0.6.0 // indirect
+)
