diff --git a/cmake/cef_variables.cmake b/cmake/cef_variables.cmake
index cc49b69..bfac1d0 100644
--- a/cef/cmake/cef_variables.cmake
+++ b/cef/cmake/cef_variables.cmake
@@ -83,7 +83,7 @@ if(OS_LINUX)
     -pipe                           # Use pipes rather than temporary files for communication between build stages
     -pthread                        # Use the pthread library
     -Wall                           # Enable all warnings
-    -Werror                         # Treat warnings as errors
+    #-Werror                         # Treat warnings as errors
     -Wno-missing-field-initializers # Don't warn about missing field initializers
     -Wno-unused-parameter           # Don't warn about unused parameters
     )
