--- /dev/null
+++ b/dependency-locks/annotationProcessor.lockfile
@@ -0,0 +1,18 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.code.findbugs:jsr305:1.3.9
+com.google.dagger:dagger-compiler:2.19
+com.google.dagger:dagger-producers:2.19
+com.google.dagger:dagger-spi:2.19
+com.google.dagger:dagger:2.19
+com.google.errorprone:error_prone_annotations:2.1.3
+com.google.errorprone:javac-shaded:9-dev-r4023-3
+com.google.googlejavaformat:google-java-format:1.5
+com.google.guava:guava:25.0-jre
+com.google.j2objc:j2objc-annotations:1.1
+com.squareup:javapoet:1.11.1
+javax.annotation:jsr250-api:1.0
+javax.inject:javax.inject:1
+org.checkerframework:checker-compat-qual:2.5.3
+org.codehaus.mojo:animal-sniffer-annotations:1.14
--- /dev/null
+++ b/dependency-locks/archives.lockfile
@@ -0,0 +1,3 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
--- /dev/null
+++ b/dependency-locks/buildscript-classpath.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+net.sf.proguard:proguard-base:6.0.3
+net.sf.proguard:proguard-gradle:6.0.3
--- /dev/null
+++ b/dependency-locks/compileClasspath.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
--- /dev/null
+++ b/dependency-locks/compile.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
--- /dev/null
+++ b/dependency-locks/compileOnly.lockfile
@@ -0,0 +1,3 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
--- /dev/null
+++ b/dependency-locks/default.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
--- /dev/null
+++ b/dependency-locks/runtimeClasspath.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
--- /dev/null
+++ b/dependency-locks/runtime.lockfile
@@ -0,0 +1,5 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
--- /dev/null
+++ b/dependency-locks/testAnnotationProcessor.lockfile
@@ -0,0 +1,3 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
--- /dev/null
+++ b/dependency-locks/testCompileClasspath.lockfile
@@ -0,0 +1,11 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
+junit:junit:4.12
+net.bytebuddy:byte-buddy-agent:1.9.0
+net.bytebuddy:byte-buddy:1.9.0
+org.hamcrest:hamcrest-core:1.3
+org.mockito:mockito-core:2.23.0
+org.objenesis:objenesis:2.6
--- /dev/null
+++ b/dependency-locks/testCompile.lockfile
@@ -0,0 +1,11 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
+junit:junit:4.12
+net.bytebuddy:byte-buddy-agent:1.9.0
+net.bytebuddy:byte-buddy:1.9.0
+org.hamcrest:hamcrest-core:1.3
+org.mockito:mockito-core:2.23.0
+org.objenesis:objenesis:2.6
--- /dev/null
+++ b/dependency-locks/testCompileOnly.lockfile
@@ -0,0 +1,3 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
--- /dev/null
+++ b/dependency-locks/testRuntimeClasspath.lockfile
@@ -0,0 +1,11 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
+junit:junit:4.12
+net.bytebuddy:byte-buddy-agent:1.9.0
+net.bytebuddy:byte-buddy:1.9.0
+org.hamcrest:hamcrest-core:1.3
+org.mockito:mockito-core:2.23.0
+org.objenesis:objenesis:2.6
--- /dev/null
+++ b/dependency-locks/testRuntime.lockfile
@@ -0,0 +1,11 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+com.google.dagger:dagger:2.19
+javax.inject:javax.inject:1
+junit:junit:4.12
+net.bytebuddy:byte-buddy-agent:1.9.0
+net.bytebuddy:byte-buddy:1.9.0
+org.hamcrest:hamcrest-core:1.3
+org.mockito:mockito-core:2.23.0
+org.objenesis:objenesis:2.6
