--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/utils/potoken/BotGuardTokenGenerator.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/utils/potoken/BotGuardTokenGenerator.kt	2026-09-24 13:04:07
@@ -7,14 +7,6 @@
 
 package moe.rukamori.archivetune.utils.potoken
 
-import android.content.Context
-import android.os.SystemClock
-import android.webkit.ConsoleMessage
-import android.webkit.JavascriptInterface
-import android.webkit.WebChromeClient
-import android.webkit.WebView
-import android.webkit.WebViewClient
-import androidx.annotation.MainThread
 import androidx.collection.ArrayMap
 import kotlinx.coroutines.CancellableContinuation
 import kotlinx.coroutines.CancellationException
