--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/BottomSheet.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/BottomSheet.kt	2026-09-24 13:03:57
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import android.view.View
 import androidx.activity.compose.BackHandler
 import androidx.compose.animation.core.Animatable
 import androidx.compose.animation.core.AnimationSpec
