--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/cast/CastScreenState.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/cast/CastScreenState.kt	2026-09-24 13:04:05
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.cast
 
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 
 sealed interface CastScreenState {
@@ -20,7 +19,7 @@
     data object Empty : CastScreenState
 
     data class Error(
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : CastScreenState
 }
 
