--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/utils/CoilBitmapLoader.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/utils/CoilBitmapLoader.kt	2026-09-24 13:02:43
@@ -7,10 +7,6 @@
 
 package moe.rukamori.archivetune.utils
 
-import android.content.Context
-import android.graphics.Bitmap
-import android.graphics.BitmapFactory
-import android.net.Uri
 import androidx.media3.common.util.BitmapLoader
 import coil3.imageLoader
 import coil3.request.ErrorResult
