--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/viewmodels/CreatePlaylistViewModel.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/viewmodels/CreatePlaylistViewModel.kt	2026-09-24 13:04:03
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.viewmodels
 
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
@@ -39,7 +38,7 @@
     @Immutable
     data class Error(
         val data: CreatePlaylistUiData,
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : CreatePlaylistScreenState
 }
 
