--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/screens/ExploreScreen.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/screens/ExploreScreen.kt	2026-09-24 13:03:59
@@ -33,8 +33,6 @@
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.hapticfeedback.HapticFeedbackType
 import androidx.compose.ui.platform.LocalHapticFeedback
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import androidx.hilt.lifecycle.viewmodel.compose.hiltViewModel
 import androidx.navigation.NavController
@@ -225,8 +223,8 @@
                     NavigationTitle(
                         title =
                             when (section.title) {
-                                "Trending" -> stringResource(R.string.trending)
-                                else -> section.title ?: stringResource(R.string.charts)
+                                "Trending" -> "trending"
+                                else -> section.title ?: "charts"
                             },
                     )
                     BoxWithConstraints(
@@ -281,7 +279,7 @@
                                             },
                                         ) {
                                             Icon(
-                                                painter = painterResource(R.drawable.more_vert),
+                                                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                                                 contentDescription = null,
                                             )
                                         }
@@ -321,7 +319,7 @@
 
                 explorePage?.newReleaseAlbums?.let { newReleaseAlbums ->
                     NavigationTitle(
-                        title = stringResource(R.string.new_release_albums),
+                        title = "new_release_albums",
                         onClick = {
                             navController.navigate("new_release")
                         },
@@ -365,7 +363,7 @@
 
                 chartsPage?.sections?.find { it.title == "Top music videos" }?.let { topVideosSection ->
                     NavigationTitle(
-                        title = stringResource(R.string.top_music_videos),
+                        title = "top_music_videos",
                     )
                     LazyRow(
                         contentPadding =
