--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/GridMenu.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/GridMenu.kt	2026-09-24 13:03:57
@@ -9,8 +9,6 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import androidx.annotation.DrawableRes
-import androidx.annotation.StringRes
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.BoxScope
@@ -37,8 +35,6 @@
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.LocalDensity
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.media3.exoplayer.offline.Download
@@ -63,16 +59,16 @@
 
 fun LazyGridScope.GridMenuItem(
     modifier: Modifier = Modifier,
-    @DrawableRes icon: Int,
+    icon: Int,
     tint: @Composable () -> Color = { LocalContentColor.current },
-    @StringRes title: Int,
+    title: Int,
     enabled: Boolean = true,
     onClick: () -> Unit,
 ) = GridMenuItem(
     modifier = modifier,
     icon = {
         Icon(
-            painter = painterResource(icon),
+            painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
             tint = tint(),
             contentDescription = null,
         )
@@ -85,7 +81,7 @@
 fun LazyGridScope.GridMenuItem(
     modifier: Modifier = Modifier,
     icon: @Composable BoxScope.() -> Unit,
-    @StringRes title: Int,
+    title: Int,
     enabled: Boolean = true,
     onClick: () -> Unit,
 ) {
@@ -188,7 +184,7 @@
                 contentAlignment = Alignment.Center,
                 content = {
                     Icon(
-                        painterResource(R.drawable.bedtime),
+                        androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                         contentDescription = null,
                         modifier = Modifier.alpha(if (enabled) 1f else 0.5f),
                     )
