--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/viewmodels/IconViewModel.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/viewmodels/IconViewModel.kt	2026-09-24 13:04:03
@@ -7,8 +7,6 @@
 
 package moe.rukamori.archivetune.viewmodels
 
-import androidx.annotation.DrawableRes
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
@@ -37,7 +35,7 @@
     data object Empty : IconScreenState
 
     data class Error(
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : IconScreenState
 }
 
@@ -67,10 +65,10 @@
 data class AppIconUiModel(
     val id: String,
     val name: String?,
-    @StringRes val nameResId: Int?,
+    val nameResId: Int?,
     val author: String?,
     val githubAuthorUrl: String?,
-    @DrawableRes val previewDrawableResId: Int,
+    val previewDrawableResId: Int,
     val isSelected: Boolean,
     val isDefault: Boolean,
 )
