--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/viewmodels/LocalPlaylistViewModel.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/viewmodels/LocalPlaylistViewModel.kt	2026-09-24 13:04:02
@@ -7,9 +7,6 @@
 
 package moe.rukamori.archivetune.viewmodels
 
-import android.content.Context
-import android.net.Uri
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
@@ -692,12 +689,12 @@
     data object Empty : PlaylistCoverState
 
     data class Error(
-        @StringRes val messageRes: Int,
+        val messageRes: Int,
     ) : PlaylistCoverState
 }
 
 sealed interface PlaylistCoverEvent {
     data class ShowMessage(
-        @StringRes val messageRes: Int,
+        val messageRes: Int,
     ) : PlaylistCoverEvent
 }
