--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/utils/LowDataMode.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/utils/LowDataMode.kt	2026-09-24 13:04:07
@@ -7,11 +7,6 @@
 
 package moe.rukamori.archivetune.utils
 
-import android.content.Context
-import android.net.ConnectivityManager
-import android.net.Network
-import android.net.NetworkCapabilities
-import android.net.NetworkRequest
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.DisposableEffect
 import androidx.compose.runtime.getValue
