--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/LyricsShareImageOptions.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/LyricsShareImageOptions.kt	2026-09-24 13:03:57
@@ -7,13 +7,12 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import moe.rukamori.archivetune.R
 import kotlin.math.roundToInt
 
 enum class LyricsShareAspectRatio(
-    @StringRes val labelRes: Int,
+    val labelRes: Int,
     val exportWidth: Int,
     val exportHeight: Int,
 ) {
