--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/lyrics/LyricsUtils.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/lyrics/LyricsUtils.kt	2026-09-24 13:03:56
@@ -7,8 +7,6 @@
 
 package moe.rukamori.archivetune.lyrics
 
-import android.icu.text.Transliterator
-import android.text.format.DateUtils
 import androidx.compose.runtime.Immutable
 import com.atilika.kuromoji.ipadic.Tokenizer
 import kotlinx.coroutines.Dispatchers
