--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/mediainfo/MediaInfoModels.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/mediainfo/MediaInfoModels.kt	2026-09-24 13:03:56
@@ -1,7 +1,5 @@
 package moe.rukamori.archivetune.mediainfo
 
-import androidx.annotation.DrawableRes
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import com.google.common.collect.ImmutableList
 import moe.rukamori.archivetune.R
