--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionUseCases.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionUseCases.kt	2026-09-24 13:04:05
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.musicrecognition
 
-import android.media.projection.MediaProjection
 import kotlinx.coroutines.CancellationException
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.distinctUntilChanged
