--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionCaptureActivity.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionCaptureActivity.kt	2026-09-24 13:04:05
@@ -7,13 +7,6 @@
 
 package moe.rukamori.archivetune.musicrecognition
 
-import android.Manifest
-import android.app.Activity
-import android.content.Context
-import android.content.pm.PackageManager
-import android.media.projection.MediaProjectionManager
-import android.os.Build
-import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.result.contract.ActivityResultContracts
 import androidx.core.content.ContextCompat
