--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionTileService.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionTileService.kt	2026-09-24 13:04:05
@@ -7,12 +7,6 @@
 
 package moe.rukamori.archivetune.musicrecognition
 
-import android.app.PendingIntent
-import android.content.Intent
-import android.graphics.drawable.Icon
-import android.os.Build
-import android.service.quicksettings.Tile
-import android.service.quicksettings.TileService
 import moe.rukamori.archivetune.R
 
 class MusicRecognitionTileService : TileService() {
