--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionRepository.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/musicrecognition/MusicRecognitionRepository.kt	2026-09-24 13:04:05
@@ -7,13 +7,6 @@
 
 package moe.rukamori.archivetune.musicrecognition
 
-import android.content.Context
-import android.media.AudioAttributes
-import android.media.AudioFormat
-import android.media.AudioPlaybackCaptureConfiguration
-import android.media.AudioRecord
-import android.media.MediaRecorder
-import android.media.projection.MediaProjection
 import androidx.datastore.preferences.core.booleanPreferencesKey
 import androidx.datastore.preferences.core.edit
 import androidx.datastore.preferences.core.stringPreferencesKey
