--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/NavigationTile.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/NavigationTile.kt	2026-09-24 13:03:57
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import androidx.annotation.DrawableRes
 import androidx.compose.foundation.background
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.focusable
@@ -24,14 +23,13 @@
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
-import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
 
 @Composable
 fun NavigationTile(
     title: String,
-    @DrawableRes icon: Int,
+    icon: Int,
     onClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
@@ -51,7 +49,7 @@
                     .clickable(onClick = onClick),
         ) {
             Icon(
-                painter = painterResource(icon),
+                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                 contentDescription = null,
             )
         }
