--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/PlayingIndicator.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/PlayingIndicator.kt	2026-09-24 11:38:29
@@ -29,7 +29,6 @@
 import androidx.compose.ui.geometry.CornerRadius
 import androidx.compose.ui.geometry.Offset
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
 import kotlinx.coroutines.delay
@@ -111,7 +110,7 @@
                 )
             } else {
                 Icon(
-                    painter = painterResource(R.drawable.play),
+                    painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                     contentDescription = null,
                     tint = color,
                 )
