--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/podcast/PodcastModels.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/podcast/PodcastModels.kt	2026-09-24 13:04:10
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.podcast
 
-import androidx.annotation.StringRes
 import androidx.compose.runtime.Immutable
 import com.google.common.collect.ImmutableList
 import moe.rukamori.archivetune.models.MediaMetadata
@@ -24,7 +23,7 @@
 
     @Immutable
     data class Error(
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : PodcastScreenState
 }
 
@@ -78,6 +77,6 @@
     ) : PodcastEvent
 
     data class ShowMessage(
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : PodcastEvent
 }
