--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/viewmodels/SearchDiscoveryViewModel.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/viewmodels/SearchDiscoveryViewModel.kt	2026-09-24 13:04:03
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.viewmodels
 
-import androidx.annotation.StringRes
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
@@ -32,7 +31,7 @@
     data object Empty : SearchDiscoveryScreenState
 
     data class Error(
-        @StringRes val messageResId: Int,
+        val messageResId: Int,
     ) : SearchDiscoveryScreenState
 }
 
