--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/SpeedDialGridItem.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/SpeedDialGridItem.kt	2026-09-24 13:03:57
@@ -37,7 +37,6 @@
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalDensity
-import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
 import coil3.compose.AsyncImage
@@ -157,7 +156,7 @@
 
                 if (item !is SongItem) {
                     Icon(
-                        painter = painterResource(R.drawable.navigate_next),
+                        painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                         contentDescription = null,
                         tint = foregroundColor,
                         modifier = Modifier.size(24.dp),
@@ -167,7 +166,7 @@
 
             if (isPinned) {
                 Icon(
-                    painter = painterResource(R.drawable.bookmark_filled),
+                    painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                     contentDescription = null,
                     tint = foregroundColor,
                     modifier =
