--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/lyrics/YouTubeSubtitleLyricsProvider.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/lyrics/YouTubeSubtitleLyricsProvider.kt	2026-09-24 13:03:56
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.lyrics
 
-import android.content.Context
 import moe.rukamori.archivetune.innertube.YouTube
 import moe.rukamori.archivetune.utils.YTPlayerUtils
 
