diff --git a/Cargo.toml b/Cargo.toml
index c94fbb9..5b99fa7 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,11 +61,11 @@ cli = ["clap", "clap_complete", "clap_mangen"]
 
 [target.'cfg(not(windows))'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "1", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
+sequoia-openpgp = { version = "1", default-features = true, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
 
 [target.'cfg(windows)'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "1", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
+sequoia-openpgp = { version = "1", default-features = true, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
 
 [package.metadata.docs.rs]
 # Enables a crypto backend for the docs.rs generation:
