From c699953d60c6d6710eb4a57f1875d2d81fafaf6d Mon Sep 17 00:00:00 2001
From: Dan Nixon <dan@dan-nixon.com>
Date: Wed, 14 Feb 2024 18:07:34 +0000
Subject: [PATCH] Add missing <cstdint> include

---
 Conf.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Conf.h b/Conf.h
index 4faf12d..aac4cf8 100644
--- a/Conf.h
+++ b/Conf.h
@@ -19,6 +19,7 @@
 #if !defined(CONF_H)
 #define	CONF_H
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
-- 
2.42.0

