diff --git i/Makefile w/Makefile
index 80654d8..bf634c3 100644
--- i/Makefile
+++ w/Makefile
@@ -30,12 +30,11 @@ install:
 
 	$(MAKE) -C po install prefix=$(DESTDIR)
 
-	cp opendict.py $(opendictdir)
-	chmod a+rx $(opendictdir)/opendict.py
+	mkdir -p $(bindir)
+	cp opendict.py $(bindir)/opendict
+	chmod a+rx $(bindir)/opendict
 	cp copying.html $(opendictdir)
 	chmod a+r $(opendictdir)/copying.html
-	mkdir -p $(bindir)
-	ln -sf $(opendictdir)/opendict.py $(bindir)/opendict
 	mkdir -p $(DESTDIR)/share/applications
 	cp misc/opendict.desktop $(DESTDIR)/share/applications
 	chmod a+r $(DESTDIR)/share/applications/opendict.desktop
