diff --git a/pyproject.toml b/pyproject.toml
index a5477a09..cb44f8b9 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,8 +31,8 @@ dependencies = [
-  "pydantic ~=1.9.1",
+  "pydantic >=1.9.1",
   "shodan ~=1.28.0",
   "detect-secrets ~=1.4.0",
-  "alive-progress ~=2.4.1",
+  "alive-progress >=2.4.1",
   "tabulate ~=0.9.0",
   "azure-identity ~=1.12.0",
   "azure-storage-blob ~=12.14.1",
   "msgraph-core ~=0.2.2",
