diff --git a/src/utils/Makefile b/src/utils/Makefile
index 9422f6c30e..1393de8bb5 100644
--- a/src/utils/Makefile
+++ b/src/utils/Makefile
@@ -41,8 +41,6 @@ all: $(CONFIGFILE)
 	$(Q)cp omnetpp $(OMNETPP_BIN_DIR)
 	$(Q)cp omnest $(OMNETPP_BIN_DIR)
 	$(Q)chmod +x $(OMNETPP_BIN_DIR)/opp_makemake $(OMNETPP_BIN_DIR)/opp_test $(OMNETPP_BIN_DIR)/opp_fingerprinttest $(OMNETPP_BIN_DIR)/splitvec $(OMNETPP_BIN_DIR)/opp_configfilepath $(OMNETPP_BIN_DIR)/opp_shlib_postprocess $(OMNETPP_BIN_DIR)/opp_runall $(OMNETPP_BIN_DIR)/omnetpp $(OMNETPP_BIN_DIR)/omnest
-	@# next line is for samples/rundemo
-	$(Q)echo $(WISH) >$(HOME)/.wishname
 
 clean:
 	$(qecho) Cleaning utils
