diff --git i/settings.gradle.kts w/settings.gradle.kts
index 2589a00db..f0c6faf86 100644
--- i/settings.gradle.kts
+++ w/settings.gradle.kts
@@ -28,10 +28,6 @@ dependencyResolutionManagement {
         maven {
             name = "GitHubPackages (Bitwarden)"
             url = uri("https://maven.pkg.github.com/bitwarden/sdk")
-            credentials {
-                username = ""
-                password = userProperties["gitHubToken"] as String? ?: System.getenv("GITHUB_TOKEN")
-            }
         }
         if ((userProperties["localSdk"] as String?).toBoolean()) {
             mavenLocal()
