--- a/internal/pcsconfig/pcsconfig.go
+++ b/internal/pcsconfig/pcsconfig.go
@@ -278,37 +278,21 @@
 		return pcsutil.ExecutablePathJoin(configDir)
 	}
 
-	// 使用旧版
-	// 如果旧版的配置文件存在, 则使用旧版
-	oldConfigDir := pcsutil.ExecutablePath()
-	_, err := os.Stat(filepath.Join(oldConfigDir, ConfigName))
-	if err == nil {
-		return oldConfigDir
-	}
-
 	switch runtime.GOOS {
 	case "windows":
 		dataPath, ok := os.LookupEnv("APPDATA")
 		if !ok {
 			pcsConfigVerbose.Warn("Environment APPDATA not set")
-			return oldConfigDir
+			return "."
 		}
 		return filepath.Join(dataPath, "BaiduPCS-Go")
 	default:
 		dataPath, ok := os.LookupEnv("HOME")
 		if !ok {
 			pcsConfigVerbose.Warn("Environment HOME not set")
-			return oldConfigDir
-		}
-		configDir = filepath.Join(dataPath, ".config", "BaiduPCS-Go")
-
-		// 检测是否可写
-		err = os.MkdirAll(configDir, 0700)
-		if err != nil {
-			pcsConfigVerbose.Warnf("check config dir error: %s\n", err)
-			return oldConfigDir
+			return "."
 		}
-		return configDir
+		return filepath.Join(dataPath, ".config", "BaiduPCS-Go")
 	}
 }
 
