diff --git a/desktop_version/CMakeLists.txt b/desktop_version/CMakeLists.txt
index 3cf9ef7..9fed238 100644
--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -108,19 +108,14 @@ if (DEFINED SDL2_INCLUDE_DIRS AND DEFINED SDL2_LIBRARIES)
 	target_include_directories(vvvvvv PUBLIC "$<BUILD_INTERFACE:${SDL2_INCLUDE_DIRS}>")
 	target_link_libraries(vvvvvv ${SDL2_LIBRARIES})
 else()
-	# Only try to autodetect if both SDL2 variables aren't explicitly set
-	find_package(SDL2 CONFIG)
-	if (TARGET SDL2::SDL2)
-		message(STATUS "Using TARGET SDL2::SDL2")
-		target_link_libraries(vvvvvv SDL2::SDL2 SDL2_mixer)
-	elseif (TARGET SDL2)
-		message(STATUS "Using TARGET SDL2")
-		target_link_libraries(vvvvvv SDL2 SDL2_mixer)
-	else()
-		message(STATUS "No TARGET SDL2::SDL2, or SDL2, using variables")
-		target_include_directories(vvvvvv PUBLIC "$<BUILD_INTERFACE:${SDL2_INCLUDE_DIRS}>")
-		target_link_libraries(vvvvvv ${SDL2_LIBRARIES} SDL2_mixer)
-	endif()
+	find_path(SDL2_INCLUDE_DIR SDL.h PATH_SUFFIXES SDL2)
+	find_library(SDL2_LIBRARY SDL2)
+
+	find_path(SDL2_MIXER_INCLUDE_DIR SDL_mixer.h PATH_SUFFIXES SDL2)
+	find_library(SDL2_MIXER_LIBRARY SDL2_mixer)
+
+	target_include_directories(vvvvvv PUBLIC ${SDL2_INCLUDE_DIR} ${SDL2_MIXER_INCLUDE_DIR})
+	target_link_libraries(vvvvvv ${SDL2_LIBRARY} ${SDL2_MIXER_LIBRARY})
 endif()
 
 # Yes, more Apple Crap
