diff --git a/CMakeLists.txt b/CMakeLists.txt
index 026ffa6..7e994e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,17 +15,6 @@ ENDIF()
 # Compiler Flags
 ADD_DEFINITIONS(-DPHYSFS_SUPPORTS_DEFAULT=0 -DPHYSFS_SUPPORTS_ZIP=1)
 
-# Executable Suffix
-IF(APPLE)
-	SET(CMAKE_EXECUTABLE_SUFFIX ".osx")
-ELSEIF(WIN32)
-	# Nothing!
-ELSEIF(CMAKE_SIZEOF_VOID_P MATCHES "8")
-	SET(CMAKE_EXECUTABLE_SUFFIX ".x86_64")
-ELSE()
-	SET(CMAKE_EXECUTABLE_SUFFIX ".x86")
-ENDIF()
-
 # Include Directories
 INCLUDE_DIRECTORIES(src tinyxml physfs lodepng)
 
