diff --git i/Makefile w/Makefile
index d15480e..5c2a0d1 100644
--- i/Makefile
+++ w/Makefile
@@ -9,9 +9,9 @@ sync:
 
 img:
 	gmt -j -v -m "SLAZAV-1" -o ${IMG_NAME} OUT/*.img /usr/share/mapsoft/slazav.typ
-	mv -f ${IMG_NAME} /home/sla/CH/data/maps/
-	sed -e "/${IMG_NAME}/s/[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}/$(date +%F)/"\
-	  -i /home/sla/CH/data/maps/index.m4i
+	#mv -f ${IMG_NAME} /home/sla/CH/data/maps/
+	#sed -e "/${IMG_NAME}/s/[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}/$(date +%F)/"\
+	#  -i /home/sla/CH/data/maps/index.m4i
 
 index: index_fan index_pam index_zab index_saj index_tsh index_kav index_nep index_put\
   index_ura index_sun index_chi
diff --git i/bin/make_in.sh w/bin/make_in.sh
index 84ab728..47c25ec 100755
--- i/bin/make_in.sh
+++ w/bin/make_in.sh
@@ -59,7 +59,7 @@ for i in $IN_DIR/*.fig $IN_DIR/*.mp; do
             $sources -o "$vmap"
 
   if [ -s "$old" ]; then
-    ~/mapsoft/vector/vmap3/vmap_fix_diff "$old" "$vmap" "$vmap"
+    vmap_fix_diff "$old" "$vmap" "$vmap"
   fi
 
   rm -f "$old"
diff --git i/bin/make_out.sh w/bin/make_out.sh
index 3d8eb2a..7e26b23 100755
--- i/bin/make_out.sh
+++ w/bin/make_out.sh
@@ -33,6 +33,7 @@ for i in $VMAP_DIR/*.vmap; do
   if [ "$png" -ot "$i" ]; then
     echo "Updating png: $name"
 
+    mkdir -p -- "$LAST_PNG_DIR"
     [ -s "$png" ] &&
       convert "$png" -scale 50% "$LAST_PNG_DIR/${name}_o.png" ||:
 
diff --git i/settings.sh w/settings.sh
index 369dcef..13a7669 100644
--- i/settings.sh
+++ w/settings.sh
@@ -5,11 +5,11 @@ export PATH=./:bin/:$PATH
 export STYLE=hr
 
 export FIG_DIR=fig
-export VMAP_DIR=/home/sla/map_hr/vmap
+export VMAP_DIR=vmap
 
 export SCALE=100000
 
-export IN_DIR=/home/sla/map_hr/IN
-export OUT_DIR=/home/sla/map_hr/OUT
+export IN_DIR=IN
+export OUT_DIR=OUT
 
 export FILTER=mmb
\ No newline at end of file
