diff --color -ur a/source/modes/drun.c b/source/modes/drun.c
--- a/source/modes/drun.c	2022-10-30 15:41:07.970678984 +0100
+++ b/source/modes/drun.c	2022-10-30 15:47:02.861931289 +0100
@@ -1016,6 +1016,19 @@
       TICK_N("Get Desktop apps (user dir)");
     }
 
+    {
+      gchar *dir;
+      // Second read nix-profile dir
+      dir = g_build_filename ( g_get_home_dir (), ".nix-profile/share/applications", NULL );
+      walk_dir ( pd, dir, dir );
+      g_free ( dir );
+
+      // Third read nix-profile dir
+      dir = g_build_filename ( "/run/current-system/sw/share/applications", NULL );
+      walk_dir ( pd, dir, dir );
+      g_free ( dir );
+    }
+
     /** Load application entires */
     p = rofi_theme_find_property(wid, P_BOOLEAN, "parse-system", TRUE);
     if (p == NULL || (p->type == P_BOOLEAN && p->value.b)) {
