Seulement dans b: ..rej
diff --color -ur a/source/dialogs/drun.c b/source/dialogs/drun.c
--- a/source/dialogs/drun.c	2019-07-01 14:29:09.000000000 +0200
+++ b/source/dialogs/drun.c	2021-03-13 18:34:56.927420856 +0100
@@ -594,6 +594,17 @@
     dir = g_build_filename ( g_get_user_data_dir (), "applications", NULL );
     walk_dir ( pd, dir, dir );
     g_free ( dir );
+
+	// Second read nix-profile dir
+	dir = g_build_filename ( g_get_home_dir (), ".nix-profile/share/applications", NULL );
+	walk_dir ( pd, dir, dir );
+	g_free ( dir );
+
+	// Third read nix-profile dir
+	dir = g_build_filename ( "/run/current-system/sw/share/applications", NULL );
+	walk_dir ( pd, dir, dir );
+	g_free ( dir );
+
     TICK_N ( "Get Desktop apps (user dir)" );
     // Then read thee system data dirs.
     const gchar * const * sys = g_get_system_data_dirs ();
Seulement dans b/source/dialogs: drun.c.orig
Seulement dans b/source/dialogs: drun.c.rej
