diff --git a/modules/juce_dsp/frequency/juce_Windowing.cpp b/modules/juce_dsp/frequency/juce_Windowing.cpp
index 13097817e..f5fcf62a8 100644
--- a/modules/juce_dsp/frequency/juce_Windowing.cpp
+++ b/modules/juce_dsp/frequency/juce_Windowing.cpp
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 namespace dsp
diff --git a/modules/juce_dsp/frequency/juce_Windowing.h b/modules/juce_dsp/frequency/juce_Windowing.h
index 2aca0ad20..5ac5c0f2a 100644
--- a/modules/juce_dsp/frequency/juce_Windowing.h
+++ b/modules/juce_dsp/frequency/juce_Windowing.h
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 namespace dsp
diff --git a/modules/juce_gui_basics/misc/juce_DropShadower.cpp b/modules/juce_gui_basics/misc/juce_DropShadower.cpp
index cbd05cc7f..f9a26693d 100644
--- a/modules/juce_gui_basics/misc/juce_DropShadower.cpp
+++ b/modules/juce_gui_basics/misc/juce_DropShadower.cpp
@@ -22,6 +22,8 @@
 
   ==============================================================================
 */
+#include <utility>
+
 
 namespace juce
 {
diff --git a/modules/juce_gui_basics/misc/juce_DropShadower.h b/modules/juce_gui_basics/misc/juce_DropShadower.h
index 38babb5a3..4c9540fec 100644
--- a/modules/juce_gui_basics/misc/juce_DropShadower.h
+++ b/modules/juce_gui_basics/misc/juce_DropShadower.h
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 
diff --git a/modules/juce_gui_basics/native/juce_linux_Windowing.cpp b/modules/juce_gui_basics/native/juce_linux_Windowing.cpp
index e81cc47dd..9faf76ffc 100644
--- a/modules/juce_gui_basics/native/juce_linux_Windowing.cpp
+++ b/modules/juce_gui_basics/native/juce_linux_Windowing.cpp
@@ -23,6 +23,8 @@
   ==============================================================================
 */
 
+#include <utility>
+
 namespace juce
 {
 
