diff --git a/src/plugin/utils.cpp b/src/plugin/utils.cpp
index 8318f698..18572edd 100644
--- a/src/plugin/utils.cpp
+++ b/src/plugin/utils.cpp
@@ -107,7 +107,7 @@ std::string PluginInfo::wine_version() const {
         access(wineloader_path.c_str(), X_OK) == 0) {
         wine_path = wineloader_path;
     } else {
-        wine_path = bp::search_path("wine").string();
+        wine_path = "@wine@/bin/wine";
     }
 
     bp::ipstream output;
@@ -436,7 +436,7 @@ Configuration load_config_for(const fs::path& yabridge_path) {
 bool send_notification(const std::string& title,
                        const std::string body,
                        bool append_origin) {
-    const fs::path notify_send_path = bp::search_path("notify-send");
+    const fs::path notify_send_path = "@libnotify@/bin/notify-send";
     if (notify_send_path.empty()) {
         return false;
     }
