diff --git a/plugin/auth_pam/CMakeLists.txt b/plugin/auth_pam/CMakeLists.txt
index e7a9d423..531c718a 100644
--- a/plugin/auth_pam/CMakeLists.txt
+++ b/plugin/auth_pam/CMakeLists.txt
@@ -15,13 +15,5 @@ IF(HAVE_PAM_APPL_H)
   IF (TARGET auth_pam)
     MYSQL_ADD_EXECUTABLE(auth_pam_tool auth_pam_tool.c DESTINATION ${INSTALL_PLUGINDIR}/auth_pam_tool_dir COMPONENT Server)
     TARGET_LINK_LIBRARIES(auth_pam_tool pam)
-    INSTALL(CODE "EXECUTE_PROCESS(
-                     COMMAND chmod u=rwx,g=,o= auth_pam_tool_dir
-                     COMMAND chmod u=rwxs,g=rx,o=rx auth_pam_tool_dir/auth_pam_tool
-                     WORKING_DIRECTORY \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${INSTALL_PLUGINDIR}/)"
-                   COMPONENT Server)
-  ENDIF()
-  IF(TARGET auth_pam OR TARGET auth_pam_v1)
-    ADD_SUBDIRECTORY(testing)
   ENDIF()
 ENDIF(HAVE_PAM_APPL_H)
