diff --git a/Makefile b/Makefile
index 51ae168..fcc91a9 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ COMPILER := gfortran
 #  COMPILER := ifort
 endif
 
-#  Use 
+#  Use
 #  make OPENGL=no
 #  to over-ride this, or change it here.
 OPENGL := yes
@@ -43,8 +43,8 @@ else
 endif
 
 #  Choose where you want to put the installed binary. The normal make
-#  leaves a a link to the binary from ${BASE}/bin anyway. 
-PREFIX  := /usr/local
+#  leaves a a link to the binary from ${BASE}/bin anyway.
+PREFIX  := ${out}
 INSTALL_DIR := ${PREFIX}/bin
 
 RECIPES = /usr/local/numerical-recipes/recipes_f/recipes/
@@ -116,7 +116,8 @@ test tests: force
 install: ${INSTALL_DIR}/orient
 
 ${INSTALL_DIR}/orient: ${BASE}/${DIR}/orient
-	cp -p ${BASE}/${DIR}/orient ${INSTALL_DIR}
+	mkdir -p ${INSTALL_DIR}
+	cp -p ${BASE}/${DIR}/orient ${INSTALL_DIR}/
 
 clean:
 	cd ${DIR}; rm *.mod *.o
@@ -168,4 +169,3 @@ histograms.f90: %.f90: %.strip
 
 recipes.f90:
 	cd ${SRC}; ${BASE}/bin/patch_recipes.pl --recipes ${RECIPES} $@
-
