diff --git i/app/build.gradle w/app/build.gradle
index a6d69f687..0533deaa6 100644
--- i/app/build.gradle
+++ w/app/build.gradle
@@ -14,12 +14,7 @@ if (project.hasProperty('strict.release')) {
 
 /* gets the version name from the latest Git tag */
 def getVersionName = { ->
-    def stdout = new ByteArrayOutputStream()
-    exec {
-        commandLine 'git', 'describe', '--tags', '--always'
-        standardOutput = stdout
-    }
-    return stdout.toString().trim()
+    return '@versionFromNix@'
 }
 
 // yes, this actually needs both quotes https://stackoverflow.com/a/41391841
