diff --git a/src/config.c b/src/config.c
index 86b95d5..74015cd 100644
--- a/src/config.c
+++ b/src/config.c
@@ -837,7 +837,13 @@ void config_free()
 void config_generate()
 {
 	struct dirent *ent;
-	DIR *dh = opendir(CONFIG_DIR);
+
+      char *config_dir = getenv("KEYD_CONFIG_DIR");
+      if (!config_dir) {
+          config_dir = CONFIG_DIR;
+      }
+
+      DIR *dh = opendir(config_dir);
 
 	if(!dh) {
 		perror("opendir");
@@ -852,7 +858,7 @@ void config_generate()
 			continue;
 
 
-		sprintf(path, "%s/%s", CONFIG_DIR, ent->d_name);
+		sprintf(path, "%s/%s", config_dir, ent->d_name);
 
 		cfg = calloc(1, sizeof(struct keyboard_config));
 
