diff --git a/src/keyd.c b/src/keyd.c
index 00f2cb4..24af05d 100644
--- a/src/keyd.c
+++ b/src/keyd.c
@@ -69,6 +69,8 @@ static struct keyboard *keyboards = NULL;
 
 int debug = 0;
 
+static char* keyd_config_dir = CONFIG_DIR;
+
 void dbg(const char *fmt, ...)
 {
 	va_list ap;
@@ -314,7 +316,7 @@ void reload_config()
 	keyboards = NULL;
 
 	free_configs();
-	read_config_dir(CONFIG_DIR);
+	read_config_dir(keyd_config_dir);
 	scan_keyboards(0);
 }
 
@@ -760,6 +762,10 @@ int main(int argc, char *argv[])
 		debug = atoi(getenv("KEYD_DEBUG"));
 
 	dbg("Debug mode enabled.");
+	char* cfg_dir = getenv("KEYD_CONFIG_DIR");
+	if (cfg_dir) {
+		keyd_config_dir = cfg_dir;
+	}
 
 	if (argc > 1) {
 		if (!strcmp(argv[1], "-v") || !strcmp(argv[1], "--version")) {
@@ -848,7 +854,7 @@ int main(int argc, char *argv[])
 	atexit(cleanup);
 
 	info("Starting keyd v%s (%s).", VERSION, GIT_COMMIT_HASH);
-	read_config_dir(CONFIG_DIR);
+	read_config_dir(keyd_config_dir);
 	vkbd = vkbd_init(VIRTUAL_KEYBOARD_NAME);
 
 	main_loop();
