diff -Naur --strip-trailing-cr -aur /nix/store/rk94vfh5na7grm2fpb9vw8m7a698wn7r-source/.envrc rhasspy-asr-kaldi/.envrc
--- /nix/store/rk94vfh5na7grm2fpb9vw8m7a698wn7r-source/.envrc	1970-01-01 01:00:00.000000000 +0100
+++ rhasspy-asr-kaldi/.envrc	2021-01-26 13:50:26.459347172 +0100
@@ -0,0 +1 @@
+use nix
diff -Naur --strip-trailing-cr -aur /nix/store/rk94vfh5na7grm2fpb9vw8m7a698wn7r-source/rhasspyasr_kaldi/transcribe.py rhasspy-asr-kaldi/rhasspyasr_kaldi/transcribe.py
--- /nix/store/rk94vfh5na7grm2fpb9vw8m7a698wn7r-source/rhasspyasr_kaldi/transcribe.py	2021-01-26 10:48:12.952961880 +0100
+++ rhasspy-asr-kaldi/rhasspyasr_kaldi/transcribe.py	2021-01-26 14:26:31.957269049 +0100
@@ -352,7 +352,7 @@
         online_conf = self.model_dir / "online" / "conf" / "online.conf"
 
         kaldi_cmd = [
-            str(self.kaldi_dir / "online2-cli-nnet3-decode-faster-confidence"),
+            str("online2-cli-nnet3-decode-faster-confidence"),
             f"--config={online_conf}",
             "--frame-subsampling-factor=3",
             "--max-active=7000",
