diff --git a/q4nx/model_converter.py b/q4nx/model_converter.py
index f12975e..456d222 100644
--- a/q4nx/model_converter.py
+++ b/q4nx/model_converter.py
@@ -96,7 +96,10 @@ class __Q4NX_Converter(ABC):
                 self.num_layers = field.contents()
 
 
-    def _load_config(self, config_file_path: str = "configs"):
+    def _load_config(self, config_file_path: str | None = None):
+        if config_file_path is None:
+            base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+            config_file_path = os.path.join(base_dir, "configs")
         config_path = os.path.join(config_file_path, ModelArchConfigs[self.model_arch])
         print(f"[INFO] Loading Q4NX config from {config_path}")
         self.q4nx_config = json.load(open(config_path))
