Binärdateien a/.git/index und b/.git/index sind verschieden.
diff -ur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2020-12-26 11:23:51.021240681 +0100
+++ b/src/CMakeLists.txt	2020-12-26 11:25:58.212454401 +0100
@@ -792,22 +792,23 @@
 	target_compile_definitions(arcan_db PRIVATE ARCAN_DB_STANDALONE)
 	list(APPEND BIN_INSTALL arcan_db)
 
+    # NIX DOESN'T LIKE INSTALLERS WHICH DO SUID
 	#
 	# Special case, the egl-dri platform requires suid- for the chain-loader
 	#
 	#
-	if (VIDEO_PLATFORM STREQUAL "egl-dri")
-		amsg("${CL_YEL}egl-dri+privsep${CL_RST}\t${CL_GRN}installing SUID${CL_RST}")
-		install(TARGETS arcan DESTINATION bin
-			PERMISSIONS
-				SETUID
-				OWNER_WRITE OWNER_READ OWNER_EXECUTE
-				GROUP_READ GROUP_EXECUTE
-				WORLD_READ WORLD_EXECUTE
-		)
-	else()
+	#if (VIDEO_PLATFORM STREQUAL "egl-dri")
+	#	amsg("${CL_YEL}egl-dri+privsep${CL_RST}\t${CL_GRN}installing SUID${CL_RST}")
+	#	install(TARGETS arcan DESTINATION bin
+	#		PERMISSIONS
+	#			SETUID
+	#			OWNER_WRITE OWNER_READ OWNER_EXECUTE
+	#			GROUP_READ GROUP_EXECUTE
+	#			WORLD_READ WORLD_EXECUTE
+	#	)
+	#else()
 	install(TARGETS arcan DESTINATION bin)
-	endif()
+	#endif()
 
 	install(TARGETS ${BIN_INSTALL} DESTINATION bin)
 
