diff --git a/mkosi/__init__.py b/mkosi/__init__.py
--- a/mkosi/__init__.py
+++ b/mkosi/__init__.py
@@ -639,7 +639,7 @@ def finalize_host_scripts(
         if context.config.find_binary(binary):
             scripts[binary] = (binary, "--root", "/buildroot")
     if ukify := context.config.find_binary("ukify"):
-        scripts["ukify"] = (python_binary(context.config), ukify)
+        scripts["ukify"] = (ukify,)
     return finalize_scripts(context.config, scripts | dict(helpers))
 
 
@@ -788,7 +788,11 @@ def script_maybe_chroot_sandbox(
 
     helpers = {
         "mkosi-chroot": [
-            finalize_interpreter(bool(context.config.tools_tree)), "-SI", "/sandbox.py",
+            *(
+                ["@MKOSI_SANDBOX@"]
+                if not bool(context.config.tools_tree)
+                else [finalize_interpreter(bool(context.config.tools_tree)), "-SI", "/sandbox.py"]
+            ),
             "--bind", "/buildroot", "/",
             "--bind", "/var/tmp", "/var/tmp",
             *apivfs_options(root=Path("/")),
@@ -1617,7 +1621,7 @@ def run_ukify(
     sign: bool = True,
     json_out: bool = False,
 ) -> dict[str, Any]:
-    ukify = context.config.find_binary("ukify", "/usr/lib/systemd/ukify")
+    ukify = context.config.find_binary("ukify", "@UKIFY@")
     if not ukify:
         die("Could not find ukify")
 
@@ -1629,7 +1633,6 @@ def run_ukify(
     (context.workspace / "cmdline").write_text(f"{' '.join(cmdline)}\x00")
 
     cmd = [
-        python_binary(context.config),
         ukify,
         "build",
         *arguments,
@@ -1724,11 +1727,11 @@ def build_uki(
     profiles: Sequence[Path],
     output: Path,
 ) -> dict[str, Any]:
-    if not (ukify := context.config.find_binary("ukify", "/usr/lib/systemd/ukify")):
+    if not (ukify := context.config.find_binary("ukify", "@UKIFY@")):
         die("Could not find ukify")
 
     json_out = False
-    ukify_version = systemd_tool_version(python_binary(context.config), ukify, sandbox=context.sandbox)
+    ukify_version = systemd_tool_version(ukify, sandbox=context.sandbox)
 
     arguments: list[PathString] = [
         "--os-release", f"@{workdir(context.root / 'usr/lib/os-release')}",
@@ -1848,7 +1851,6 @@ def build_uki(
         # new .ucode section support?
         if (
             systemd_tool_version(
-                python_binary(context.config),
                 ukify,
                 sandbox=context.sandbox,
             )
@@ -1916,7 +1918,7 @@ def want_uki(context: Context) -> bool:
         or (
             context.config.unified_kernel_images == UnifiedKernelImage.auto
             and systemd_stub_binary(context).exists()
-            and context.config.find_binary("ukify", "/usr/lib/systemd/ukify") is not None
+            and context.config.find_binary("ukify", "@UKIFY@") is not None
         )
     )
 
@@ -2856,9 +2858,9 @@ def check_ukify(
     reason: str,
     hint: Optional[str] = None,
 ) -> None:
-    ukify = check_tool(config, "ukify", "/usr/lib/systemd/ukify", reason=reason, hint=hint)
+    ukify = check_tool(config, "ukify", "@UKIFY@", reason=reason, hint=hint)
 
-    v = systemd_tool_version(python_binary(config), ukify, sandbox=config.sandbox)
+    v = systemd_tool_version(ukify, sandbox=config.sandbox)
     if v < version:
         die(
             f"Found '{ukify}' with version {v} but version {version} or newer is required to {reason}.",
diff --git a/mkosi/bootloader.py b/mkosi/bootloader.py
--- a/mkosi/bootloader.py
+++ b/mkosi/bootloader.py
@@ -315,8 +315,7 @@ def find_signed_grub_image(context: Context) -> Optional[Path]:
 def python_binary(config: Config) -> PathString:
     # If there's no tools tree, prefer the interpreter from MKOSI_INTERPRETER. If there is a tools
     # tree, just use the default python3 interpreter.
-    exe = Path(sys.executable)
-    return "python3" if config.tools_tree or not exe.is_relative_to("/usr") else exe
+    return "python3" if config.tools_tree else "@PYTHON_PEFILE@"
 
 
 def extract_pe_section(context: Context, binary: Path, section: str, output: Path) -> Path:
diff --git a/mkosi/qemu.py b/mkosi/qemu.py
--- a/mkosi/qemu.py
+++ b/mkosi/qemu.py
@@ -192,7 +192,7 @@ def find_ovmf_firmware(config: Config, firmware: Firmware) -> Optional[OvmfConfi
     if not firmware.is_uefi():
         return None
 
-    desc = list((config.tools() / "usr/share/qemu/firmware").glob("*"))
+    desc = list((config.tools() / "@QEMU_FIRMWARE@").glob("*"))
     if config.tools() == Path("/"):
         desc += list((config.tools() / "etc/qemu/firmware").glob("*"))
 
diff --git a/mkosi/run.py b/mkosi/run.py
--- a/mkosi/run.py
+++ b/mkosi/run.py
@@ -352,7 +352,11 @@ def spawn(
             module = stack.enter_context(resource_path(sys.modules[__package__ or __name__]))
             prefix = [
                 *(["strace", "--detach-on=execve", "--string-limit=256"] if ARG_DEBUG_SANDBOX.get() else []),
-                sys.executable, "-SI", os.fspath(module / "sandbox.py"),
+                *(
+                    ["@MKOSI_SANDBOX@"]
+                    if os.path.exists("@MKOSI_SANDBOX@")
+                    else [sys.executable, "-SI", os.fspath(module / "sandbox.py")]
+                ),
                 *sbx,
             ]  # fmt: skip
 
@@ -805,7 +809,7 @@ def chroot_options() -> list[PathString]:
         "--unshare-ipc",
         "--setenv", "container", "mkosi",
         "--setenv", "HOME", "/",
-        "--setenv", "PATH", "/usr/bin:/usr/sbin",
+        "--setenv", "PATH", "/usr/bin:/usr/sbin:@NIX_PATH@",
         "--setenv", "BUILDROOT", "/",
     ]  # fmt: skip
 
diff --git a/mkosi/sandbox.py b/mkosi/sandbox.py
--- a/mkosi/sandbox.py
+++ b/mkosi/sandbox.py
@@ -158,7 +158,7 @@ class cap_user_data_t(ctypes.Structure):
     ]
 
 
-libc = ctypes.CDLL(None, use_errno=True)
+libc = ctypes.CDLL("@LIBC@" if os.path.exists("@LIBC@") else None, use_errno=True)
 
 libc.syscall.restype = ctypes.c_long
 libc.unshare.argtypes = (ctypes.c_int,)
@@ -370,7 +370,7 @@ def seccomp_suppress(*, chown: bool = False, sync: bool = False) -> None:
     if not chown and not sync:
         return
 
-    libseccomp = ctypes.CDLL("libseccomp.so.2")
+    libseccomp = ctypes.CDLL("@LIBSECCOMP@" if os.path.exists("@LIBSECCOMP@") else "libseccomp.so.2")
     if libseccomp is None:
         raise FileNotFoundError("libseccomp.so.2")
 
