diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0d183ca..31301d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,18 +15,8 @@ if(BUILD_TESTS)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native -mtune=native -pipe -fno-plt")
 endif(BUILD_TESTS)
 
-execute_process(
-    COMMAND git rev-list --count HEAD
-    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-    OUTPUT_VARIABLE GIT_REV_LIST
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-execute_process(
-    COMMAND git rev-parse --short HEAD
-    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-    OUTPUT_VARIABLE GIT_REV_PARSE
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+set(GIT_REV_LIST "493")
+set(GIT_REV_PARSE "79bec5c")
 
 set(PROJECT_VERSION "r${GIT_REV_LIST}.${GIT_REV_PARSE}")
 set(PROJECT_VERSION_MAJOR "${GIT_REV_LIST}")
