diff --git a/misc/CMakeLists.txt b/misc/CMakeLists.txt
index c9b70bd5..ff6c93e1 100644
--- a/misc/CMakeLists.txt
+++ b/misc/CMakeLists.txt
@@ -164,11 +164,11 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/share/applications/linyaps.desktop
 
 # set linglong XDG_DATA_DIRS environtment
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/script/linglong.sh
-        DESTINATION /etc/profile.d)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/profile.d)
 
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/script/linglong.sh
-  DESTINATION /etc/X11/Xsession.d/
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/etc/X11/Xsession.d/
   RENAME 21linglong)
 
 # polkit actions
diff --git a/misc/share/applications/linyaps.desktop b/misc/share/applications/linyaps.desktop
index 069e3d6e..d3c47107 100644
--- a/misc/share/applications/linyaps.desktop
+++ b/misc/share/applications/linyaps.desktop
@@ -2,7 +2,7 @@
 Name=Linyaps
 Encoding=UTF-8
 Type=Application
-Exec=/usr/bin/ll-cli install %f
+Exec=ll-cli install %f
 NoDisplay=true
 Terminal=true
 Name[zh_CN]=如意玲珑
