diff --git a/include/configs/rk3328_common.h b/include/configs/rk3328_common.h
index bd2bfe2910..1123bfe899 100644
--- a/include/configs/rk3328_common.h
+++ b/include/configs/rk3328_common.h
@@ -21,8 +21,8 @@
 	"fdt_addr_r=0x01e00000\0"	\
 	"fdtoverlay_addr_r=0x01f00000\0"	\
 	"kernel_addr_r=0x02080000\0"	\
-	"ramdisk_addr_r=0x06000000\0"	\
-	"kernel_comp_addr_r=0x08000000\0"	\
+	"ramdisk_addr_r=0x08000000\0" \
+	"kernel_comp_addr_r=0x10000000\0" \
 	"kernel_comp_size=0x2000000\0"
 
 #define CFG_EXTRA_ENV_SETTINGS		\
