From 75d9edf6dc57d0ffbd29df93079a7c129f577b1c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Thalheim?= <joerg@thalheim.io>
Date: Sat, 6 Jun 2020 01:41:29 +0100
Subject: [PATCH] make mycroft precise optional
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>
---
 setup.py.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setup.py.in b/setup.py.in
index 11ede88..c81df86 100644
--- a/setup.py.in
+++ b/setup.py.in
@@ -91,6 +91,11 @@ enable_porcupine = is_yes("@ENABLE_PORCUPINE@")
 if not enable_porcupine:
     packages.pop("rhasspywake_porcupine_hermes")
 
+# mycroft precise
+enable_precise = is_yes("@ENABLE_PRECISE@")
+if not enable_precise:
+    packages.pop("rhasspywake_precise_hermes")
+
 # snips
 enable_snips = is_yes("@ENABLE_SNIPS@")
 if not enable_snips:
-- 
2.26.2

