diff --git a/ext/openssl/openssl.c b/ext/openssl/openssl.c
index 4af09b0e..3691e979 100644
--- a/ext/openssl/openssl.c
+++ b/ext/openssl/openssl.c
@@ -3524,11 +3524,7 @@ PHP_FUNCTION(openssl_csr_sign)
 		goto cleanup;
 	}
 
-#if PHP_OPENSSL_API_VERSION >= 0x10100
-	ASN1_INTEGER_set_int64(X509_get_serialNumber(new_cert), serial);
-#else
-	ASN1_INTEGER_set(X509_get_serialNumber(new_cert), serial);
-#endif
+	ASN1_INTEGER_set(X509_get_serialNumber(new_cert), (long)serial);
 
 	X509_set_subject_name(new_cert, X509_REQ_get_subject_name(csr));
 
