diff --git a/setup_config_version.sh b/setup_config_version.sh
index fde40c1..6386014 100755
--- a/setup_config_version.sh
+++ b/setup_config_version.sh
@@ -8,7 +8,7 @@ case "$1" in
                exit
 esac
 
-case "$(busybox head -n1 "$1")" in
+case "$(head -n1 "$1")" in
 	"#"*)
 		comment="#"
 		;;
@@ -23,7 +23,7 @@ case "$(busybox head -n1 "$1")" in
 		;;
 esac
 
-busybox md5sum "$1" | \
-	busybox cut -d" " -f1 | \
-	busybox xargs -I{} busybox sed -i "2i$comment configversion: {}" \
+md5sum "$1" | \
+	cut -d" " -f1 | \
+	xargs -I{} sed -i "2i$comment configversion: {}" \
 	"$1"
