diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index 194814d..beb9232 100755
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -16,16 +16,6 @@ while ! superctl status > /dev/null 2>&1; do
 	sleep 0.5
 done
 
-# Load our sound daemons
-
-if [ "$(command -v pulseaudio)" ]; then
-	superctl start pulseaudio
-elif [ "$(command -v pipewire)" ]; then
-	# pipewire-pulse will start pipewire
-	superctl start pipewire-pulse
-	superctl start wireplumber
-fi
-
 # Periodically update some status bar components
 sxmo_hook_statusbar.sh all
 sxmo_daemons.sh start statusbar_periodics sxmo_run_aligned.sh 60 \
