diff --git a/configs/default_hooks/three_button_touchscreen/sxmo_hook_lock.sh b/configs/default_hooks/three_button_touchscreen/sxmo_hook_lock.sh
index c9c4263..4c0fccf 100755
--- a/configs/default_hooks/three_button_touchscreen/sxmo_hook_lock.sh
+++ b/configs/default_hooks/three_button_touchscreen/sxmo_hook_lock.sh
@@ -37,7 +37,7 @@ sxmo_daemons.sh stop periodic_wakelock_check
 # Go to screenoff after 8 seconds of inactivity
 if ! [ -e "$XDG_CACHE_HOME/sxmo/sxmo.noidle" ]; then
 	sxmo_daemons.sh start idle_locker sxmo_idle.sh -w \
-		timeout 8 "sxmo_hook_screenoff.sh"
+		timeout "${SXMO_LOCK_IDLE_TIME:-8}" "sxmo_hook_screenoff.sh"
 fi
 
 wait
