From 4d2cb5b6b8d0aa236c91333185d3379b26daee4a Mon Sep 17 00:00:00 2001
From: Millian Poquet <millian.poquet@irit.fr>
Date: Fri, 3 Feb 2023 17:19:42 +0100
Subject: [PATCH] bs: loosen tomlkit dep version constraint

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index a71591e..0a97590 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,7 +24,7 @@ psutil = "^5.9.0"
 PyYAML = "^6.0"
 execo = "^2.6.8"
 requests = "^2.27.1"
-tomlkit = "^0.10.0"
+tomlkit = ">=0.10.0"
 
 [tool.poetry.dev-dependencies]
 pytest = "^3.0"
-- 
2.38.1

