diff --git a/SpriteSheetPacker/SpriteSheetPacker.pro b/SpriteSheetPacker/SpriteSheetPacker.pro
index cee8a6f..73b1cda 100644
--- a/SpriteSheetPacker/SpriteSheetPacker.pro
+++ b/SpriteSheetPacker/SpriteSheetPacker.pro
@@ -129,14 +129,17 @@ CONFIG(release,debug|release) {
     win32 {
         DEPLOY_COMMAND = windeployqt
         isEmpty(TARGET_EXT) TARGET_EXT = .exe
+        DEPLOY_TARGET = $$shell_quote($$shell_path($${DESTDIR}/$${TARGET}$${TARGET_EXT}))
+
+        QMAKE_POST_LINK = $${DEPLOY_COMMAND} $${DEPLOY_TARGET}
     }
 
     macx {
         DEPLOY_COMMAND = macdeployqt
         isEmpty(TARGET_EXT) TARGET_EXT = .app
-    }
+        DEPLOY_TARGET = $$shell_quote($$shell_path($${DESTDIR}/$${TARGET}$${TARGET_EXT}))
 
-    DEPLOY_TARGET = $$shell_quote($$shell_path($${DESTDIR}/$${TARGET}$${TARGET_EXT}))
+        QMAKE_POST_LINK = $${DEPLOY_COMMAND} $${DEPLOY_TARGET}
+    }
 
-    QMAKE_POST_LINK = $${DEPLOY_COMMAND} $${DEPLOY_TARGET}
 }
