diff --git a/setup.py b/setup.py
index aa29c1f..c4d44a4 100644
--- a/setup.py
+++ b/setup.py
@@ -16,7 +16,7 @@ setup(name='Pegamoid',
       url='https://gitlab.com/Jellby/Pegamoid',
       license='GPL v3.0',
       scripts=['pegamoid.py'],
-      install_requires=['numpy (>=1.9.0)', 'h5py', 'VTK (>=8.1.0)', 'qtpy', 'future (>=0.15.2);python_version<"3.0"' ],
+      install_requires=['numpy (>=1.9.0)', 'h5py', 'qtpy', 'future (>=0.15.2);python_version<"3.0"' ],
       long_description=long_description,
       long_description_content_type="text/markdown",
       classifiers=[

