From 6b07d65a87c627a510af53717af74b265b0674df Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20L=C3=B8vbr=C3=B8tte=20Olsen?= <daniel@dodsorf.as>
Date: Sat, 2 Nov 2019 10:55:50 +0100
Subject: [PATCH 1/3] disable_spotless

---
 build.gradle | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/build.gradle b/build.gradle
index 89d9f44..54ca85e 100644
--- a/build.gradle
+++ b/build.gradle
@@ -79,11 +79,11 @@ shadowJar {
 	baseName = project.name
 }
 
-spotless {
-	java {
-		eclipse().configFile 'config/format/eclipseformat.xml'
-	}
-}
+#spotless {
+#	java {
+#		eclipse().configFile 'config/format/eclipseformat.xml'
+#	}
+#t}
 
 jacocoTestReport {
 	reports {
-- 
2.19.3

