diff --git a/bin/SHARC_MOLCAS.py b/bin/SHARC_MOLCAS.py
index c49da01..5d61d1f 100755
--- a/bin/SHARC_MOLCAS.py
+++ b/bin/SHARC_MOLCAS.py
@@ -873,7 +873,7 @@ def makermatrix(a,b):
 
 # ======================================================================= #
 def getversion(out,MOLCAS):
-    allowedrange=[ (18.0,18.999), (8.29999,8.30001) ]
+    allowedrange=[ (18.0,21.02), (8.29999,8.30001) ]
     # first try to find $MOLCAS/.molcasversion
     molcasversion=os.path.join(MOLCAS,'.molcasversion')
     if os.path.isfile(molcasversion):
