diff --git a/tests/INPUT/MOLPRO_ddr/QM/MOLPRO.resources b/tests/INPUT/MOLPRO_ddr/QM/MOLPRO.resources
index ec204dc..2eb350b 100644
--- a/tests/INPUT/MOLPRO_ddr/QM/MOLPRO.resources
+++ b/tests/INPUT/MOLPRO_ddr/QM/MOLPRO.resources
@@ -1,4 +1,5 @@
 molpro $MOLPRO/
 scratchdir $SCRADIR/WORK
 savedir $SCRADIR/TRAJ/restart
-wfoverlap $SHARC/wfoverlap.x
\ No newline at end of file
+wfoverlap $SHARC/wfoverlap.x
+memory 1000
diff --git a/tests/INPUT/MOLPRO_overlap/QM/MOLPRO.resources b/tests/INPUT/MOLPRO_overlap/QM/MOLPRO.resources
index ec204dc..2eb350b 100644
--- a/tests/INPUT/MOLPRO_overlap/QM/MOLPRO.resources
+++ b/tests/INPUT/MOLPRO_overlap/QM/MOLPRO.resources
@@ -1,4 +1,5 @@
 molpro $MOLPRO/
 scratchdir $SCRADIR/WORK
 savedir $SCRADIR/TRAJ/restart
-wfoverlap $SHARC/wfoverlap.x
\ No newline at end of file
+wfoverlap $SHARC/wfoverlap.x
+memory 1000
diff --git a/tests/INPUT/MOLPRO_singlestate/QM/MOLPRO.resources b/tests/INPUT/MOLPRO_singlestate/QM/MOLPRO.resources
index ec204dc..2eb350b 100644
--- a/tests/INPUT/MOLPRO_singlestate/QM/MOLPRO.resources
+++ b/tests/INPUT/MOLPRO_singlestate/QM/MOLPRO.resources
@@ -1,4 +1,5 @@
 molpro $MOLPRO/
 scratchdir $SCRADIR/WORK
 savedir $SCRADIR/TRAJ/restart
-wfoverlap $SHARC/wfoverlap.x
\ No newline at end of file
+wfoverlap $SHARC/wfoverlap.x
+memory 1000
