diff --git a/bin/tests.py b/bin/tests.py
index e67168c..8a5783a 100755
--- a/bin/tests.py
+++ b/bin/tests.py
@@ -355,7 +355,9 @@ def run_tests(INFOS):
     if os.path.isdir(newpath):
       sys.stdout.write('%s *** OVERWRITTEN ***\n' % (newpath))
       shutil.rmtree(newpath)
-    shutil.copytree(path,newpath)
+    os.system("mkdir -p " + newpath)  
+    os.system("cp -r " + path + "/* " + newpath)  
+    os.system("chmod -R +w " + newpath)  
     os.chdir(newpath)
 
     starttime=datetime.datetime.now()
@@ -612,4 +614,4 @@ if __name__ == '__main__':
     main()
   except KeyboardInterrupt:
     sys.stdout.write('\nCtrl+C makes me a sad SHARC ;-(\n')
-    quit(0)
\ No newline at end of file
+    quit(0)
