diff --git c/mk/flags.mk i/mk/flags.mk
index 3c0887cf9a..254ab2bb77 100644
--- c/mk/flags.mk
+++ i/mk/flags.mk
@@ -27,15 +27,15 @@ LDFLAGS ?=
 
 CROSS_COMPILE ?=
 
-GCC     := $(CROSS_COMPILE)gcc
-GXX     := $(CROSS_COMPILE)g++
-AR      := $(CROSS_COMPILE)ar
-AS      := $(CROSS_COMPILE)as
-LD      := $(CROSS_COMPILE)ld
-NM      := $(CROSS_COMPILE)nm
-OBJDUMP := $(CROSS_COMPILE)objdump
-OBJCOPY := $(CROSS_COMPILE)objcopy
-SIZE    := $(CROSS_COMPILE)size
+GCC     = $(CROSS_COMPILE)gcc
+CXX     = $(CROSS_COMPILE)g++
+AR      = $(CROSS_COMPILE)ar
+AS      = $(CROSS_COMPILE)as
+LD      = $(CROSS_COMPILE)ld
+NM      = $(CROSS_COMPILE)nm
+OBJDUMP = $(CROSS_COMPILE)objdump
+OBJCOPY = $(CROSS_COMPILE)objcopy
+SIZE    = $(CROSS_COMPILE)size
 
 ifeq ($(COMPILER),clang)
 CC  := clang
@@ -44,7 +44,7 @@ else
 CC  := $(GCC)
 CXX := $(GXX)
 endif
-CPP := $(CC) -E
+CPP = $(CC) -E
 
 EMBOX_GCC := $(TOOLCHAIN_DIR)/embox-gcc
 EMBOX_GXX := $(TOOLCHAIN_DIR)/embox-g++
