diff --git c/mk/flags.mk i/mk/flags.mk
index cc977f7a07..acc4ce050a 100644
--- c/mk/flags.mk
+++ i/mk/flags.mk
@@ -23,14 +23,14 @@ MODULE_CPPFLAGS ?=
 COMPILER ?= gcc
 CROSS_COMPILE ?=
 
-CXX     ?= $(CROSS_COMPILE)g++
-AR      ?= $(CROSS_COMPILE)ar
-AS      ?= $(CROSS_COMPILE)as
-LD      ?= $(CROSS_COMPILE)ld
-NM      ?= $(CROSS_COMPILE)nm
-OBJDUMP ?= $(CROSS_COMPILE)objdump
-OBJCOPY ?= $(CROSS_COMPILE)objcopy
-SIZE    ?= $(CROSS_COMPILE)size
+CXX     = $(CROSS_COMPILE)g++
+AR      = $(CROSS_COMPILE)ar
+AS      = $(CROSS_COMPILE)as
+LD      = $(CROSS_COMPILE)ld
+NM      = $(CROSS_COMPILE)nm
+OBJDUMP = $(CROSS_COMPILE)objdump
+OBJCOPY = $(CROSS_COMPILE)objcopy
+SIZE    = $(CROSS_COMPILE)size
 
 ifeq ($(COMPILER),clang)
 CC      := clang
@@ -39,7 +39,7 @@ else
 CC      := $(CROSS_COMPILE)gcc
 LIBGCC_FINDER=$(CC) $(CFLAGS)
 endif
-CPP     ?= $(CC) -E
+CPP     = $(CC) -E
 
 CPPFLAGS += -D__EMBOX_VERSION__=\"$(EMBOX_VERSION)\"
 
