diff --git a/samples/landlock/sandboxer.c b/samples/landlock/sandboxer.c
index e8223c3e781a..e4583b8447a1 100644
--- a/samples/landlock/sandboxer.c
+++ b/samples/landlock/sandboxer.c
@@ -326,10 +326,3 @@ int main(const int argc, char *const argv[], char *const *const envp)
-
-		/* Must be printed for any ABI < LANDLOCK_ABI_LAST. */
-		fprintf(stderr,
-			"Hint: You should update the running kernel "
-			"to leverage Landlock features "
-			"provided by ABI version %d (instead of %d).\n",
-			LANDLOCK_ABI_LAST, abi);
 		__attribute__((fallthrough));
 	case LANDLOCK_ABI_LAST:
 		break;
