diff --git i/build/build.ts w/build/build.ts
index 1ae3f65..c1109ab 100644
--- i/build/build.ts
+++ w/build/build.ts
@@ -139,21 +139,7 @@ function generateSymbol(id: string, svgStr: string): string {
 }
 
 async function getBuildMetadata(): Promise<string> {
-  const date = new Date().toISOString().slice(2, 10).replaceAll("-", "");
-
-  const [commit, diff, newFiles] = await Promise.all([
-    execAsync("git rev-parse HEAD"),
-    execAsync("git diff HEAD"),
-    execAsync("git ls-files --others --exclude-standard"),
-  ]).then((res) => res.map((r) => r.stdout.trim()));
-
-  if (!diff && !newFiles) return date + commit.slice(0, 4);
-
-  const hash = createHash("md5");
-  hash.update(commit).update(diff).update(newFiles);
-  for (const file of newFiles.split("\n").filter((f) => f))
-    hash.update(await fsAsync.readFile(file));
-  return date + hash.digest("hex").slice(0, 4);
+  return "nix";
 }
 
 async function init(): Promise<void> {
