diff --git a/CMakeLists.txt b/CMakeLists.txt
index ab6b8eb..75be387 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ add_definitions(-DQT_NO_CAST_FROM_ASCII)
 # Set PROJECT_VERSION_PATCH and PROJECT_VERSION_TWEAK to 0 if not present, needed by add_project_meta
 fix_project_version()
 
-find_package(Qt6 REQUIRED COMPONENTS Gui Qml Quick OpenGL Network LinguistTools)
+find_package(Qt6 REQUIRED COMPONENTS GuiPrivate  Gui Qml Quick OpenGL Network LinguistTools)
 find_package(Qt6Gui REQUIRED)
 find_package(MPV REQUIRED)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 307b695..75f7c4e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -152,6 +152,7 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
     Qt6::Quick
     Qt6::Gui
     Qt6::Network
+    Qt6::GuiPrivate
     Qt6::OpenGL
     ${MPV_LIBRARY}
     Danmaku2AssCPP)
