From 190ca55f0ae23fc8e12b7d7eda809af026f0e680 Mon Sep 17 00:00:00 2001
From: Bart Oostveen <bart@bartoostveen.nl>
Date: Sat, 9 May 2026 22:25:07 +0200
Subject: [PATCH] fix: proper color handing according to spec

---
 pkg/bot/formatting.go | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/pkg/bot/formatting.go b/pkg/bot/formatting.go
index c2fd0b1..3f57d89 100644
--- a/pkg/bot/formatting.go
+++ b/pkg/bot/formatting.go
@@ -18,7 +18,7 @@ import (
 //nolint:lll // long templates
 const (
 	DefaultTextTemplate = "{{ range .Alerts }}{{.StatusString|icon}} {{.StatusString|upper}} {{.AlertName}}: {{.Summary}}{{if ne .Fingerprint ``}} ({{.Fingerprint}}){{end}}{{if $.ShowLabels}}, labels: {{.LabelString}}{{end}}\n{{ end -}}"
-	DefaultHTMLTemplate = `{{ range .Alerts }}<font color="{{.StatusString|color}}">{{.StatusString|icon}} <b>{{.StatusString|upper}}</b> {{.AlertName}}:</font> {{.Summary}}{{if ne .Fingerprint ""}} ({{.Fingerprint}}){{end}}{{if $.ShowLabels}}<br/><b>Labels:</b> <code>{{.LabelString}}</code>{{end}}<br/>{{- end -}}`
+	DefaultHTMLTemplate = `{{ range .Alerts }}<p data-mx-color="{{.StatusString|color}}">{{.StatusString|icon}} <b>{{.StatusString|upper}}</b> {{.AlertName}}:</p> {{.Summary}}{{if ne .Fingerprint ""}} ({{.Fingerprint}}){{end}}{{if $.ShowLabels}}<br/><b>Labels:</b> <code>{{.LabelString}}</code>{{end}}<br/>{{- end -}}`
 )
 
 //go:embed templates/silence.md.tmpl
@@ -27,14 +27,14 @@ var silenceTemplate string
 // Default color and icon values.
 var (
 	DefaultColors = map[string]string{ //nolint:gochecknoglobals // used as constant
-		"alert":       "black",
-		"information": "blue",
-		"info":        "blue",
-		"warning":     "orange",
-		"critical":    "red",
-		"error":       "red",
-		"resolved":    "green",
-		"silenced":    "gray",
+		"alert":       "#000000",
+		"information": "#0000ff",
+		"info":        "#0000ff",
+		"warning":     "#ffa500",
+		"critical":    "#ff0000",
+		"error":       "#ff0000",
+		"resolved":    "#00ff00",
+		"silenced":    "#808080",
 	}
 
 	DefaultIcons = map[string]string{ //nolint:gochecknoglobals // used as constant
-- 
2.54.0

