From 9c9708bcabe07bbf06af585f66444ecff96a1293 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Wed, 23 Mar 2022 14:21:34 +0100
Subject: [PATCH 347/463] usb: typec: typec-extcon: Enable debugging for now

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/usb/typec/typec-extcon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/typec/typec-extcon.c b/drivers/usb/typec/typec-extcon.c
index e09cd7a28c24e..f617e90b71a09 100644
--- a/drivers/usb/typec/typec-extcon.c
+++ b/drivers/usb/typec/typec-extcon.c
@@ -5,6 +5,7 @@
  * This driver bridges standard type-c interfaces to drivers that
  * expect extcon interface.
  */
+#define DEBUG
 
 #include <linux/delay.h>
 #include <linux/kernel.h>
-- 
2.46.0

