diff --git a/prefsCleaner.sh b/prefsCleaner.sh
index 6586609..bc2b27e 100755
--- a/prefsCleaner.sh
+++ b/prefsCleaner.sh
@@ -8,7 +8,7 @@
 
 ## DON'T GO HIGHER THAN VERSION x.9 !! ( because of ASCII comparison in update_prefsCleaner() )
 
-readonly CURRDIR=$(pwd)
+readonly CURRDIR=${1:-$PWD}
 
 ## get the full path of this script (readlink for Linux, greadlink for Mac with coreutils installed)
 SCRIPT_FILE=$(readlink -f "${BASH_SOURCE[0]}" 2>/dev/null || greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null)
@@ -33,7 +33,7 @@ fi
 
 fQuit() {
 	## change directory back to the original working directory
-	cd "${CURRDIR}"
+	# cd "${CURRDIR}"
 	[ "$1" -eq 0 ] && echo -e "\n$2" || echo -e "\n$2" >&2
 	exit $1
 }
@@ -130,7 +130,7 @@ while getopts "sd" opt; do
 done
 
 ## change directory to the Firefox profile directory
-cd "$(dirname "${SCRIPT_FILE}")"
+# cd "$(dirname "${SCRIPT_FILE}")"
 
 # Check if running as root and if any files have the owner/group as root/wheel.
 if [ "${EUID:-"$(id -u)"}" -eq 0 ]; then
@@ -142,7 +142,7 @@ you will need to change ownership of the following files to your user:\n'
 	fQuit 1
 fi
 
-[ "$AUTOUPDATE" = true ] && update_prefsCleaner "$@"
+# [ "$AUTOUPDATE" = true ] && update_prefsCleaner "$@"
 
 echo -e "\n\n"
 echo "                   ╔══════════════════════════╗"
