diff --git i/config.def.h w/config.def.h
index 01ebd48..1725f9e 100644
--- i/config.def.h
+++ w/config.def.h
@@ -38,6 +38,8 @@ static const char *mupdf[]        = { "mupdf", "-I" };
 static const char *libreoffice[]  = { "libreoffice" };
 static const char *gimp[]         = { "gimp" };
 static const char *r2[]           = { "r2", "-c", "vv" };
+static const char *gpxsee[]       = { "gpxsee" };
+static const char *oomapper[]     = { "Mapper" };
 
 /* extensions*/
 static const char *images[]    = { "bmp", "jpg", "jpeg", "png", "gif", "xpm" };
@@ -50,6 +52,9 @@ static const char *videos[]    = { "avi", "flv", "wav", "webm", "wma", "wmv",
 				   "mp4", "mpeg", "mpg" };
 static const char *documents[] = { "odt", "doc", "docx", "xls", "xlsx", "odp",
 				   "ods", "pptx", "odg" };
+static const char *gpsdata[]   = { "gpx", "tcx", "kml", "loc", "slf", "fit", "igc", "nmea", "plt", "rte", "wpt", "geojson", "cup", "gpi", "sml", "ov2", "itn" };
+static const char *maps[]      = { "img", "jnx", "xml", "kap", "map", "mbtiles", "rmap", "rtmap", "tba", "kmz", "aqm", "sqlitedb" };
+static const char *omaps[]     = { "omap", "xmap", "ocd" };
 
 static Rule rules[] = {
 	{videos,    LEN(videos),    mpv,         LEN(mpv)         },
@@ -59,6 +64,9 @@ static Rule rules[] = {
 	{documents, LEN(documents), libreoffice, LEN(libreoffice) },
 	{arts,      LEN(arts),      gimp,        LEN(gimp)        },
 	{obj,       LEN(obj),       r2,          LEN(r2)          },
+	{gpsdata,   LEN(gpsdata),   gpxsee,      LEN(gpxsee)      },
+	{maps,      LEN(maps),      gpxsee,      LEN(gpxsee)      },
+	{omaps,     LEN(omaps),     oomapper,    LEN(oomapper)    },
 };
 
 /* normal keys */
