diff --git a/setup.py b/setup.py
index 1fc5c84..951903d 100644
--- a/setup.py
+++ b/setup.py
@@ -1,14 +1,12 @@
 import fastentrypoints
 import setuptools
-import versioneer
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 setuptools.setup(
     name="rstb",
-    version=versioneer.get_version(),
-    cmdclass=versioneer.get_cmdclass(),
+    version="@version@",
     author="leoetlino",
     author_email="leo@leolam.fr",
     description="Breath of the Wild RSTB parser and editing tool",
