From 37113ec4c1768f4b7488a9f4f41b650c1017b194 Mon Sep 17 00:00:00 2001
From: connordennyomalley <connordennyomalley@gmail.com>
Date: Mon, 30 Mar 2026 12:00:47 +0100
Subject: [PATCH] Make file does not update git submodules

---
 FIRE6/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/FIRE6/Makefile.in b/FIRE6/Makefile.in
index d8cc5d5..1202789 100644
--- a/FIRE6/Makefile.in
+++ b/FIRE6/Makefile.in
@@ -121,7 +121,7 @@ endif
 submodule:
 	cd .. && git submodule update --init --recursive
 
-fuel: submodule
+fuel:
 	cd extra/fuel && ./configure $(SYMBOLICA_OPTION) $(FLINT_OPTION) $(PREBUILT_MPFR_OPTION) $(PREBUILT_GMP_OPTION) --cpp=$(CCPLUS) --cc=$(CC)
 	$(MAKE) -C extra/fuel dep
 	$(MAKE) -C extra/fuel library
-- 
2.39.5 (Apple Git-154)

