diff -ur a/config/config.c b/config/config.c
--- a/config/config.c	2019-07-01 14:29:09.000000000 +0200
+++ b/config/config.c	2020-07-04 22:32:05.084307968 +0200
@@ -52,7 +52,7 @@
     .menu_font              = "mono 12",
 
     /** Whether to load and show icons */
-    .show_icons             = FALSE,
+    .show_icons             = TRUE,
 
     /** Terminal to use. (for ssh and open in terminal) */
     .terminal_emulator = "rofi-sensible-terminal",
diff -ur a/source/rofi-icon-fetcher.c b/source/rofi-icon-fetcher.c
--- a/source/rofi-icon-fetcher.c	2019-07-01 14:29:09.000000000 +0200
+++ b/source/rofi-icon-fetcher.c	2020-07-04 22:35:54.098924340 +0200
@@ -103,6 +103,7 @@
     static const gchar * const icon_fallback_themes[] = {
         "Adwaita",
         "gnome",
+		"hicolor",
         NULL
     };
     const char                 *themes[2] = { config.icon_theme, NULL };
diff -ur a/subprojects/libnkutils/src/xdg-theme.c b/subprojects/libnkutils/src/xdg-theme.c
--- a/subprojects/libnkutils/src/xdg-theme.c	2018-03-20 13:15:02.000000000 +0100
+++ b/subprojects/libnkutils/src/xdg-theme.c	2020-07-04 22:38:06.518424208 +0200
@@ -316,6 +316,19 @@
 
     try_dir(g_get_user_data_dir());
 
+	 {    
+		const gchar* nix_profiles = g_getenv("NIX_PROFILES");
+		gchar** nix_profiles_list = g_strsplit(nix_profiles, " ", -1); 
+		int i = 0; 
+		gchar* current_profile_dir = nix_profiles_list[0];
+		for (i = 0; current_profile_dir != NULL; current_profile_dir = nix_profiles_list[++i]) {
+			gchar* dir = g_build_filename(current_profile_dir, "share", NULL);
+			try_dir(dir);
+			g_free ( dir );
+		}    
+		g_strfreev(nix_profiles_list);
+	}
+
     switch ( self->type )
     {
     case TYPE_ICON:
