diff -ur a/source/dialogs/drun.c b/source/dialogs/drun.c
--- a/source/dialogs/drun.c	2019-07-01 14:29:09.000000000 +0200
+++ b/source/dialogs/drun.c	2020-07-04 16:09:56.143707599 +0200
@@ -589,6 +589,19 @@
 {
     TICK_N ( "Get Desktop apps (start)" );
 
+	{
+		const gchar* nix_profiles = g_getenv("NIX_PROFILES");
+		gchar** nix_profiles_list = g_strsplit(nix_profiles, " ", -1);
+		int i = 0;
+		gchar* current_profile_dir = nix_profiles_list[0];
+		for (i = 0; current_profile_dir != NULL; current_profile_dir = nix_profiles_list[++i]) {
+			gchar* dir = g_build_filename(current_profile_dir, "share", "applications", NULL);
+			walk_dir ( pd, dir, dir );
+			g_free ( dir );
+		}
+		g_strfreev(nix_profiles_list);
+	}
+
     gchar *dir;
     // First read the user directory.
     dir = g_build_filename ( g_get_user_data_dir (), "applications", NULL );
