--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/about/AboutAttributionRepository.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/about/AboutAttributionRepository.kt	2026-09-24 13:04:07
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.about
 
-import android.content.Context
 import androidx.compose.runtime.Immutable
 import androidx.datastore.preferences.core.edit
 import com.mikepenz.aboutlibraries.Libs
