--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/musicrecognition/BackgroundMusicRecognitionService.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/musicrecognition/BackgroundMusicRecognitionService.kt	2026-09-24 13:04:05
@@ -7,19 +7,6 @@
 
 package moe.rukamori.archivetune.musicrecognition
 
-import android.app.Activity
-import android.app.Service
-import android.content.ComponentName
-import android.content.Context
-import android.content.Intent
-import android.content.pm.ServiceInfo
-import android.media.projection.MediaProjection
-import android.media.projection.MediaProjectionManager
-import android.os.Build
-import android.os.Handler
-import android.os.IBinder
-import android.os.Looper
-import android.service.quicksettings.TileService
 import androidx.core.app.ServiceCompat
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.CancellationException
@@ -30,7 +17,6 @@
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.launch
-import android.util.Base64
 import kotlinx.serialization.json.Json
 import moe.rukamori.archivetune.MainActivity
 import timber.log.Timber
