--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/extensions/ContextExt.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/extensions/ContextExt.kt	2026-09-24 13:04:08
@@ -7,9 +7,6 @@
 
 package moe.rukamori.archivetune.extensions
 
-import android.content.Context
-import android.net.ConnectivityManager
-import android.net.NetworkCapabilities
 import moe.rukamori.archivetune.constants.InnerTubeCookieKey
 import moe.rukamori.archivetune.constants.YtmSyncKey
 import moe.rukamori.archivetune.innertube.utils.hasYouTubeLoginCookie
