--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/discord/DiscordOAuthCallbackActivity.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/discord/DiscordOAuthCallbackActivity.kt	2026-09-24 13:04:04
@@ -7,9 +7,6 @@
 
 package moe.rukamori.archivetune.discord
 
-import android.app.Activity
-import android.content.Intent
-import android.os.Bundle
 
 class DiscordOAuthCallbackActivity : Activity() {
     override fun onCreate(savedInstanceState: Bundle?) {
