--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/discord/DiscordOAuthRepository.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/discord/DiscordOAuthRepository.kt	2026-09-24 13:04:04
@@ -7,8 +7,6 @@
 
 package moe.rukamori.archivetune.discord
 
-import android.content.Context
-import android.net.Uri
 import androidx.datastore.preferences.core.edit
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableSharedFlow
