--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/Library.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/Library.kt	2026-09-24 13:03:57
@@ -18,7 +18,6 @@
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Shape
-import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
 import kotlinx.coroutines.CoroutineScope
@@ -55,7 +54,7 @@
             },
         ) {
             Icon(
-                painter = painterResource(R.drawable.more_vert),
+                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                 contentDescription = null,
             )
         }
@@ -123,7 +122,7 @@
             },
         ) {
             Icon(
-                painter = painterResource(R.drawable.more_vert),
+                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                 contentDescription = null,
             )
         }
@@ -189,7 +188,7 @@
                 modifier = dragHandleModifier,
             ) {
                 Icon(
-                    painter = painterResource(R.drawable.drag_handle),
+                    painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                     contentDescription = null,
                 )
             }
@@ -239,7 +238,7 @@
             },
         ) {
             Icon(
-                painter = painterResource(R.drawable.more_vert),
+                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                 contentDescription = null,
             )
         }
