--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/screens/LoginScreen.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/screens/LoginScreen.kt	2026-09-24 13:03:59
@@ -7,15 +7,6 @@
 
 package moe.rukamori.archivetune.ui.screens
 
-import android.annotation.SuppressLint
-import android.content.Intent
-import android.net.Uri
-import android.webkit.CookieManager
-import android.webkit.JavascriptInterface
-import android.webkit.WebResourceRequest
-import android.webkit.WebView
-import android.webkit.WebViewClient
-import android.widget.Toast
 import androidx.activity.compose.BackHandler
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.windowInsetsPadding
@@ -28,8 +19,6 @@
 import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.viewinterop.AndroidView
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
@@ -66,7 +55,6 @@
         "https://youtube.com",
     )
 
-@SuppressLint("SetJavaScriptEnabled")
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun LoginScreen(
@@ -79,7 +67,7 @@
     val context = LocalContext.current
 
     val screenState by viewModel.screenState.collectAsStateWithLifecycle()
-    val loginSuccessMessage = stringResource(R.string.login_success)
+    val loginSuccessMessage = "login_success"
     var webView: WebView? = null
 
     LaunchedEffect(screenState, loginSuccessMessage) {
@@ -150,7 +138,7 @@
     )
 
     TopAppBar(
-        title = { Text(stringResource(R.string.login)) },
+        title = { Text("login") },
         navigationIcon = {
             IconButton(
                 onClick = {
@@ -169,7 +157,7 @@
                 },
             ) {
                 Icon(
-                    painterResource(R.drawable.arrow_back),
+                    androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                     contentDescription = null,
                 )
             }
