--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/LyricsImageCard.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/LyricsImageCard.kt	2026-09-24 13:03:58
@@ -7,7 +7,6 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import android.os.Build
 import androidx.compose.foundation.Image
 import androidx.compose.foundation.background
 import androidx.compose.foundation.border
@@ -44,8 +43,6 @@
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalDensity
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.AnnotatedString
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontWeight
@@ -366,7 +363,7 @@
                             contentAlignment = Alignment.Center,
                         ) {
                             Image(
-                                painter = painterResource(id = R.drawable.small_icon),
+                                painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
                                 contentDescription = null,
                                 modifier = Modifier.size(15.dp),
                                 colorFilter =
@@ -379,7 +376,7 @@
                         Spacer(modifier = Modifier.size(8.dp))
 
                         androidx.compose.material3.Text(
-                            text = stringResource(R.string.app_name),
+                            text = "app_name",
                             color = colors.secondary,
                             fontSize = 13.sp,
                             fontWeight = FontWeight.SemiBold,
