--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/component/MarkdownText.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/component/MarkdownText.kt	2026-09-24 13:03:58
@@ -7,9 +7,6 @@
 
 package moe.rukamori.archivetune.ui.component
 
-import android.text.method.LinkMovementMethod
-import android.util.TypedValue
-import android.widget.TextView
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
