--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/theme/Theme.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/theme/Theme.kt	2026-09-24 13:04:01
@@ -7,12 +7,6 @@
 
 package moe.rukamori.archivetune.ui.theme
 
-import android.app.WallpaperManager
-import android.content.Context
-import android.graphics.Bitmap
-import android.graphics.drawable.BitmapDrawable
-import android.os.Build
-import android.util.Base64
 import androidx.compose.animation.animateColorAsState
 import androidx.compose.animation.core.FiniteAnimationSpec
 import androidx.compose.animation.core.snap
