--- app/src/old_android_main/kotlin/moe/rukamori/archivetune/ui/player/Thumbnail.kt	2026-09-24 02:17:47
+++ app/src/main/kotlin/moe/rukamori/archivetune/ui/player/Thumbnail.kt	2026-09-24 13:04:02
@@ -7,8 +7,6 @@
 
 package moe.rukamori.archivetune.ui.player
 
-import android.graphics.Bitmap
-import android.os.Build
 import androidx.compose.animation.AnimatedVisibility
 import androidx.compose.animation.fadeIn
 import androidx.compose.animation.fadeOut
@@ -65,8 +63,6 @@
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.LocalLayoutDirection
 import androidx.compose.ui.platform.LocalView
-import androidx.compose.ui.res.painterResource
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.LayoutDirection
@@ -325,7 +321,7 @@
                 modifier = Modifier.padding(horizontal = 32.dp, vertical = 16.dp),
             ) {
                 Text(
-                    text = stringResource(R.string.now_playing),
+                    text = "now_playing",
                     style = MaterialTheme.typography.titleMedium,
                     color = textBackgroundColor,
                 )
@@ -439,8 +435,8 @@
                                         contentAlignment = Alignment.Center,
                                     ) {
                                         Icon(
-                                            painter = painterResource(R.drawable.about_splash),
-                                            contentDescription = stringResource(R.string.hide_player_thumbnail),
+                                            painter = androidx.compose.ui.graphics.painter.ColorPainter(androidx.compose.ui.graphics.Color.Transparent),
+                                            contentDescription = "hide_player_thumbnail",
                                             tint = textBackgroundColor.copy(alpha = 0.7f),
                                             modifier = Modifier.size(120.dp),
                                         )
