diff --git a/LibCurlImpersonate/CurlHttp.cs b/LibCurlImpersonate/CurlHttp.cs
index 950495c..8595637 100644
--- a/LibCurlImpersonate/CurlHttp.cs
+++ b/LibCurlImpersonate/CurlHttp.cs
@@ -11,7 +11,7 @@ public class CurlHttp : IDisposable
         Path.GetDirectoryName(AppContext.BaseDirectory)
     );
 
-    private static readonly string PathToCacert = Path.Join(CurDir, "cacert.pem");
+    private static readonly string PathToCacert = "@cacert@";
 
     public static IEnumerable<string> FetchLines(
         string url,
diff --git a/Stalker.Gamma/Models/StalkerGammaSettings.cs b/Stalker.Gamma/Models/StalkerGammaSettings.cs
index 37aeed2..938bf21 100644
--- a/Stalker.Gamma/Models/StalkerGammaSettings.cs
+++ b/Stalker.Gamma/Models/StalkerGammaSettings.cs
@@ -19,7 +19,7 @@ public class StalkerGammaSettings
     public string ModOrganizer244Md5 { get; set; } = "e2bb7233cdab78f56912ebf4a0091768";
     public string ModOrganizer252Md5 { get; set; } = "b223ce1297107adbabb3fbaa3769eb2b";
     public int DownloadThreads { get; set; } = 2;
-    public string PathToUnzip = "unzip";
-    public string PathTo7Z = OperatingSystem.IsWindows() ? "7zz.exe" : "7zz";
-    public string PathToTar = "tar";
+    public string PathToUnzip = "@unzip@";
+    public string PathTo7Z = "@_7zz@";
+    public string PathToTar = "@tar@";
 }
diff --git a/stalker-gamma-cli/Commands/Anomaly.cs b/stalker-gamma-cli/Commands/Anomaly.cs
index 5ecea5f..a407240 100644
--- a/stalker-gamma-cli/Commands/Anomaly.cs
+++ b/stalker-gamma-cli/Commands/Anomaly.cs
@@ -60,11 +60,7 @@ public class AnomalyInstallCmd(
         ValidateActiveProfile.Validate(_logger, _cliSettings.ActiveProfile);
         var anomaly = _cliSettings.ActiveProfile!.Anomaly;
         var cache = _cliSettings.ActiveProfile!.Cache;
-        var resourcesPath = Path.Join(Path.GetDirectoryName(AppContext.BaseDirectory), "resources");
-        stalkerGammaSettings.PathTo7Z = Path.Join(
-            resourcesPath,
-            OperatingSystem.IsWindows() ? "7zz.exe" : "7zz"
-        );
+        stalkerGammaSettings.PathTo7Z = "@_7zz@";
 
         var anomalyInstaller = (AnomalyInstaller)
             downloadableRecordFactory.CreateAnomalyRecord(cache, anomaly);
diff --git a/stalker-gamma-cli/Services/SetupUtilitiesService.cs b/stalker-gamma-cli/Services/SetupUtilitiesService.cs
index debe352..82120e8 100644
--- a/stalker-gamma-cli/Services/SetupUtilitiesService.cs
+++ b/stalker-gamma-cli/Services/SetupUtilitiesService.cs
@@ -6,10 +6,7 @@ public class SetupUtilitiesService(StalkerGammaSettings settings)
 {
     public void Setup()
     {
-        settings.PathTo7Z = Path.Join(
-            ResourcesPath,
-            OperatingSystem.IsWindows() ? "7zz.exe" : "7zz"
-        );
+        settings.PathTo7Z = "@_7zz@";
     }
 
     private static readonly string ResourcesPath = Path.Join(
