--- ij-src/platform/build-scripts/src/org/jetbrains/intellij/build/impl/JpsCompilationRunner.kt
+++ ij-src/platform/build-scripts/src/org/jetbrains/intellij/build/impl/JpsCompilationRunner.kt
@@ -51,23 +51,23 @@
   }
 
   suspend fun buildModules(modules: List<JpsModule>, canceledStatus: CanceledStatus = CanceledStatus.NULL) {
-    throw NotImplementedError("JPS compilation was broken on purpose, for details see MRI-3677")
+    runBuild()
   }
 
   suspend fun resolveProjectDependencies() {
     runBuild()
   }
 
   suspend fun buildModuleTests(module: JpsModule, canceledStatus: CanceledStatus = CanceledStatus.NULL) {
-    throw NotImplementedError("JPS compilation was broken on purpose, for details see MRI-3677")
+    runBuild()
   }
 
   suspend fun buildAll(canceledStatus: CanceledStatus = CanceledStatus.NULL) {
-    throw NotImplementedError("JPS compilation was broken on purpose, for details see MRI-3677")
+    runBuild()
   }
 
   suspend fun buildProduction(canceledStatus: CanceledStatus = CanceledStatus.NULL) {
-    throw NotImplementedError("JPS compilation was broken on purpose, for details see MRI-3677")
+    runBuild()
   }
 
   private suspend fun runBuild() = context.withCompilationLock {
@@ -88,6 +88,8 @@
           .build()
       )
     }
+    scopes.add(TargetTypeBuildScope.newBuilder().setTypeId("java-production").setForceBuild(false).setAllTargets(true).build())
+    scopes.add(TargetTypeBuildScope.newBuilder().setTypeId("java-test").setForceBuild(false).setAllTargets(true).build())
 
     val compilationStart = System.nanoTime()
     spanBuilder("compilation")
