diff --git a/config/other.go b/config/other.go
index 2c648f1..d2210e1 100644
--- a/config/other.go
+++ b/config/other.go
@@ -155,8 +155,6 @@ func setDefaults(conf *Config, bus awlevent.Bus) {
 	if isEmptyConfig {
 		conf.HttpListenOnAdminHost = true
 	}
-	// TODO: remove in next release
-	conf.HttpListenOnAdminHost = true
 
 	if conf.VPNConfig.IPNet == "" {
 		conf.VPNConfig.IPNet = defaultNetworkSubnet
