diff --git a/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp b/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
index 9b0dc1d65..808c0a3de 100644
--- a/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
@@ -366,7 +366,7 @@ void Manager::Private::init(Manager *manager) {
 			auto capabilityText = QString::fromUtf8(static_cast<const char*>(capability->data));
 			_capabilities.push_back(capabilityText);
 		}
-		Libs::g_list_free_full(capabilities, g_free);
+		Libs::g_list_free_full(capabilities, Libs::g_free);
 
 		LOG(("LibNotify capabilities: %1").arg(_capabilities.join(qstr(", "))));
 		if (_capabilities.contains(qsl("actions"))) {
