--- include/asterisk/format_cache.h	(Asterisk 13.13.1)
+++ include/asterisk/format_cache.h	(working copy)
@@ -234,2 +234,7 @@
 /*!
+ * \brief Built-in cached 3GPP EVS format.
+ */
+extern struct ast_format *ast_format_evs;
+
+/*!
  * \brief Initialize format cache support within the core.
--- main/codec_builtin.c	(Asterisk 13.13.1)
+++ main/codec_builtin.c	(working copy)
@@ -855,2 +855,13 @@

+
+static struct ast_codec evs = {
+	.name = "evs",
+	.description = "3GPP EVS",
+	.type = AST_MEDIA_TYPE_AUDIO,
+	.sample_rate = 16000,
+	.minimum_ms = 20,
+	.maximum_ms = 300,
+	.default_ms = 20,
+};
+
 #define CODEC_REGISTER_AND_CACHE(codec) \
@@ -887,2 +898,4 @@
 
+	res |= CODEC_REGISTER_AND_CACHE(evs);
+
 	res |= CODEC_REGISTER_AND_CACHE(g723);
--- main/format_cache.c	(Asterisk 13.13.1)
+++ main/format_cache.c	(working copy)
@@ -220,2 +220,7 @@
 /*!
+ * \brief Built-in cached 3GPP EVS format.
+ */
+struct ast_format *ast_format_evs;
+
+/*!
  * \brief Built-in cached t140 format.
@@ -302,2 +307,4 @@
 
+	ao2_replace(ast_format_evs, NULL);
+
 	ao2_replace(ast_format_g723, NULL);
@@ -416,2 +423,4 @@
 		ao2_replace(ast_format_g719, format);
+	} else if (!strcmp(name, "evs")) {
+		ao2_replace(ast_format_evs, format);
 	} else if (!strcmp(name, "opus")) {
--- main/rtp_engine.c	(Asterisk 13.13.1)
+++ main/rtp_engine.c	(working copy)
@@ -3720,2 +3720,4 @@
 	add_static_payload(107, ast_format_opus, 0);
+
+	ast_rtp_engine_load_format(ast_format_evs);
 
