diff --git a/pjsip/src/pjsip/sip_config.c b/pjsip/src/pjsip/sip_config.c
index bf5f8b347..2b98ffb14 100644
--- a/pjsip/src/pjsip/sip_config.c
+++ b/pjsip/src/pjsip/sip_config.c
@@ -36,7 +36,7 @@ pjsip_cfg_t pjsip_sip_cfg_var =
        PJSIP_FOLLOW_EARLY_MEDIA_FORK,
        PJSIP_REQ_HAS_VIA_ALIAS,
        PJSIP_RESOLVE_HOSTNAME_TO_GET_INTERFACE,
-       0,
+       1,
        PJSIP_ENCODE_SHORT_HNAME,
        PJSIP_ACCEPT_MULTIPLE_SDP_ANSWERS,
        0
