diff --git a/portal.go b/portal.go
index 25e0dc2..e305942 100644
--- a/portal.go
+++ b/portal.go
@@ -108,15 +108,6 @@ func (br *DiscordBridge) loadPortal(dbPortal *database.Portal, key *database.Por
 	if portal.GuildID != "" {
 		portal.Guild = portal.bridge.GetGuildByID(portal.GuildID, true)
 	}
-	if portal.ParentID != "" {
-		parentKey := database.NewPortalKey(portal.ParentID, "")
-		var ok bool
-		portal.Parent, ok = br.portalsByID[parentKey]
-		if !ok {
-			portal.Parent = br.loadPortal(br.DB.Portal.GetByID(parentKey), nil, -1)
-		}
-	}
-
 	return portal
 }
 
@@ -351,7 +342,7 @@ func (portal *Portal) CreateMatrixRoom(user *User, channel *discordgo.Channel) e
 
 	creationContent := make(map[string]interface{})
 	if portal.Type == discordgo.ChannelTypeGuildCategory {
-		creationContent["type"] = event.RoomTypeSpace
+		return nil
 	}
 	if !portal.bridge.Config.Bridge.FederateRooms {
 		creationContent["m.federate"] = false
@@ -1777,18 +1768,12 @@ func (portal *Portal) UpdateParent(parentID string) bool {
 	}
 	portal.log.Debugfln("Updating parent ID %q -> %q", portal.ParentID, parentID)
 	portal.ParentID = parentID
-	if portal.ParentID != "" {
-		portal.Parent = portal.bridge.GetPortalByID(database.NewPortalKey(parentID, ""), discordgo.ChannelTypeGuildCategory)
-	} else {
-		portal.Parent = nil
-	}
+	portal.Parent = nil
 	return true
 }
 
 func (portal *Portal) ExpectedSpaceID() id.RoomID {
-	if portal.Parent != nil {
-		return portal.Parent.MXID
-	} else if portal.Guild != nil {
+	if portal.Guild != nil {
 		return portal.Guild.MXID
 	}
 	return ""
@@ -1798,9 +1783,7 @@ func (portal *Portal) updateSpace() bool {
 	if portal.MXID == "" {
 		return false
 	}
-	if portal.Parent != nil {
-		return portal.addToSpace(portal.Parent.MXID)
-	} else if portal.Guild != nil {
+	if portal.Guild != nil {
 		return portal.addToSpace(portal.Guild.MXID)
 	}
 	return false
