diff --git a/thefuck/output_readers/rerun.py b/thefuck/output_readers/rerun.py
index b7ffe24..f938a0a 100644
--- a/thefuck/output_readers/rerun.py
+++ b/thefuck/output_readers/rerun.py
@@ -53,6 +53,9 @@ def get_output(script, expanded):
     """
     env = dict(os.environ)
     env.update(settings.env)
+    path_env = env['PATH']
+    if 'thefuck' in path_env:
+        env['PATH'] = ':'.join(path_env.split(':')[2:])
 
     if six.PY2:
         expanded = expanded.encode('utf-8')
