commit 6ffbb91036558eed103bb3a426b4c230358f6f7d
Author: CrazedProgrammer <crazedprogrammer@gmail.com>
Date:   Tue Jun 12 16:33:17 2018 +0200

    Better space formatting

diff --git a/src/ls.c b/src/ls.c
index ed6726e..f29d85f 100644
--- a/src/ls.c
+++ b/src/ls.c
@@ -177,39 +177,35 @@ static int display_loc_info( const char * path, const char * loc, size_t flags,
 	display( ( stats.st.st_mode & S_IWOTH ) ? "{c}w" : "{0}-" );
 	display( ( stats.st.st_mode & S_IXOTH ) ? "{r}x" : "{0}-" );
 
-	// number of links
-	if( flags & OPT_I ) display( "\t" );
-	display( "\t{0}%d\t", stats.st.st_nlink );
-
 	// user and group
 	struct passwd * usr = getpwuid( stats.st.st_uid );
 	struct group * grp = getgrgid( stats.st.st_gid );
 
-	if( usr == NULL ) display_padded( 14, "{g}%d", stats.st.st_uid );
-	else display_padded( 15, "{g}%s", usr->pw_name );
+	if( usr == NULL ) display_padded( 15, "  {g}%d", stats.st.st_uid );
+	else display_padded( 15, "  {g}%s", usr->pw_name );
 
-	if( grp == NULL ) display( "\t{g}%d", stats.st.st_gid );
-	else display( "\t{g}%s", grp->gr_name );
+	if( grp == NULL ) display_padded( 14, "  {g}%d", stats.st.st_gid );
+	else display_padded( 14, "  {g}%s", grp->gr_name );
 
 	// file size
 	if( flags & OPT_H ) {
 		char size[ 30 ];
 		format_file_size( stats.st.st_size, size );
-		display( "\t{y}%s", size );
+		display_padded( 15, "  {y}%s", size );
 	}
 	else {
-		display( "\t{y}%lluB", stats.st.st_size );
+		display_padded( 15, "  {y}%lluB", stats.st.st_size );
 	}
 
 	// last modified time
 	char mtime[ 30 ];
 	strftime( mtime, 30, "%h %e %H:%M", gmtime( & stats.st.st_mtime ) );
-	display( "\t{w}%s", mtime );
+	display_padded( 14, "  {w}%s", mtime );
 
 	// file/folder name
-	if( S_ISDIR( stats.st.st_mode ) ) display( "\t{b}%s  %s", icon, loc );
-	else if( S_ISLNK( stats.st.st_mode ) ) display( "\t{y}%s  %s{0}" , icon, loc );
-	else display( "\t{g}%s  %s", icon, loc );
+	if( S_ISDIR( stats.st.st_mode ) ) display( "\t{b}%s %s", icon, loc );
+	else if( S_ISLNK( stats.st.st_mode ) ) display( "\t{y}%s %s{0}" , icon, loc );
+	else display( "\t{g}%s %s", icon, loc );
 
 	// link info for links
 	if( S_ISLNK( stats.st.st_mode ) ) {
