--- hadoop-2.6.5-src/hadoop-tools/hadoop-pipes/src/CMakeLists.txt	2016-09-20 07:38:41.000000000 +0200
+++ hadoop-2.6.5-src-new/hadoop-tools/hadoop-pipes/src/CMakeLists.txt	2021-04-14 16:10:44.982999857 +0200
@@ -18,6 +18,8 @@
 
 cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
 find_package(OpenSSL REQUIRED)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(TIRPC REQUIRED libtirpc)
 
 set(CMAKE_BUILD_TYPE, Release)
 
@@ -42,6 +44,7 @@
     main/native/pipes/api
     ${CMAKE_CURRENT_SOURCE_DIR}
     ${OPENSSL_INCLUDE_DIR}
+    ${TIRPC_INCLUDEDIR}/tirpc
 )
 
 # Example programs
@@ -71,5 +74,6 @@
 )
 target_link_libraries(hadooppipes
     ${OPENSSL_LIBRARIES}
+    ${TIRPC_LIBRARIES}
     ${LIB_DL}
     ${LIB_NET}
 )
