diff --git a/nullfs.c b/nullfs.c
index 509f1a2..78a9418 100644
--- a/nullfs.c
+++ b/nullfs.c
@@ -536,9 +536,9 @@ int nullfs_statfs(struct dentry *dentry, struct kstatfs *buf)
      **/
     buf->f_type  = dentry->d_sb->s_magic;
     buf->f_bsize = dentry->d_sb->s_blocksize;
-    buf->f_blocks = 100000000;
-    buf->f_bfree =  90000000;
-    buf->f_bavail = 90000000;
+    buf->f_blocks = 1LU << 32;
+    buf->f_bfree =  1LU << 32;
+    buf->f_bavail = 1LU << 32;
     buf->f_namelen = NAME_MAX;
     return 0;
 }
