diff --git c/tile_provider.go i/tile_provider.go
index 727363e..da297b4 100644
--- c/tile_provider.go
+++ i/tile_provider.go
@@ -163,6 +163,28 @@ func NewTileProviderNone() *TileProvider {
 	return t
 }
 
+// NewTileProviderOpenTopoMapRu creates a TileProvider struct for opentopomap's tile service
+func NewTileProviderOpenTopoMapRu() *TileProvider {
+	t := new(TileProvider)
+	t.Name = "opentopomapru"
+	t.Attribution = "Maps (c) OpenTopoMap.ru [CC-BY-SA]; Data (c) OSM and contributors [ODbL]; Data (c) SRTM"
+	t.TileSize = 256
+	t.URLPattern = "https://tile-%[1]s.opentopomap.ru/%[2]d/%[3]d/%[4]d.png"
+	t.Shards = []string{"a", "b", "c"}
+	return t
+}
+
+// NewTileProviderInkatlas creates a TileProvider struct for inkatlas's tile service
+func NewTileProviderInkatlas() *TileProvider {
+	t := new(TileProvider)
+	t.Name = "inkatlas"
+	t.Attribution = "Maps (c) Inkatlas OÜ; Data (c) OSM and contributors [ODbL]; Data (c) SRTM"
+	t.TileSize = 256
+	t.URLPattern = "https://tile.inkatlas.com/inkatlas/outdoor/%[2]d/%[3]d/%[4]d.png"
+	t.Shards = []string{}
+	return t
+}
+
 // GetTileProviders returns a map of all available TileProviders
 func GetTileProviders(thunderforestApiKey string) map[string]*TileProvider {
 	m := make(map[string]*TileProvider)
@@ -181,6 +203,8 @@ func GetTileProviders(thunderforestApiKey string) map[string]*TileProvider {
 		NewTileProviderArcgisWorldImagery(),
 		NewTileProviderWikimedia(),
 		NewTileProviderNone(),
+		NewTileProviderOpenTopoMapRu(),
+		NewTileProviderInkatlas(),
 	}
 
 	for _, tp := range list {
