From 0055e9a9bf024dbd2f1142be4fdf6c5b9b4e99a0 Mon Sep 17 00:00:00 2001
From: Sofi <sofi+git@mailbox.org>
Date: Sun, 17 Jul 2022 15:55:17 +0200
Subject: [PATCH] Remove `std::process:ExitCode`

---
 Cargo.toml  | 3 +--
 src/main.rs | 8 +++-----
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index ff3abcf..2e4c104 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,8 +2,7 @@
 name = "ferium"
 version = "4.1.6"
 edition = "2021"
-# Because of `std::process::ExitCode`
-rust-version = "1.61"
+rust-version = "1.60"
 authors = [
     "Ilesh Thiada (theRookieCoder) <ileshkt@gmail.com>",
     "薛詠謙 (KyleUltimate)",
diff --git a/src/main.rs b/src/main.rs
index d4df0e1..767b55f 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -19,7 +19,6 @@ use octocrab::OctocrabBuilder;
 use online::check;
 use std::{
     env::{var, var_os},
-    process::ExitCode,
     sync::Arc,
 };
 use tokio::{runtime, spawn};
@@ -39,7 +38,7 @@ lazy_static! {
         .progress_chars("#>-");
 }
 
-fn main() -> ExitCode {
+fn main() {
     let cli = Ferium::parse();
     let mut builder = runtime::Builder::new_multi_thread();
     builder.enable_all();
@@ -51,9 +50,8 @@ fn main() -> ExitCode {
     let runtime = builder.build().expect("Could not initialise Tokio runtime");
     if let Err(err) = runtime.block_on(actual_main(cli)) {
         eprintln!("{}", err.to_string().red().bold());
-        ExitCode::FAILURE
-    } else {
-        ExitCode::SUCCESS
+        runtime.shutdown_background();
+        std::process::exit(1);
     }
 }
 
-- 
2.36.1

