diff --git i/CMakeLists.txt w/CMakeLists.txt
index c03bca2..8146cfb 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -40,7 +40,6 @@ target_include_directories(subtitles_generator
         ${UTFCPP_DIR}/source
         ${MAGICENUM_DIR}/include/magic_enum
 )
-target_link_options(subtitles_generator PRIVATE -static)
 target_link_libraries(subtitles_generator
         PRIVATE
         CLI11::CLI11
