diff --git i/CMakeLists.txt w/CMakeLists.txt
index d6a3201..6f20aea 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -55,16 +55,9 @@ if (BUILD_GUI)
     # Find GUI dependencies
     find_package(OpenGL REQUIRED)
     find_package(GLEW REQUIRED)
+    find_package(PkgConfig REQUIRED)
 
-    # GLFW configuration
-    set(GLFW_DIR "${CMAKE_SOURCE_DIR}/submodules/glfw")
-    option(GLFW_BUILD_EXAMPLES "Build the GLFW example programs" OFF)
-    option(GLFW_BUILD_TESTS "Build the GLFW test programs" OFF)
-    option(GLFW_BUILD_DOCS "Build the GLFW documentation" OFF)
-    option(GLFW_INSTALL "Generate installation target" OFF)
-    option(GLFW_DOCUMENT_INTERNALS "Include internals in documentation" OFF)
-    add_subdirectory(${GLFW_DIR} ${CMAKE_BINARY_DIR}/glfw EXCLUDE_FROM_ALL)
-    include_directories(${GLFW_DIR}/include ${GLFW_DIR}/deps)
+    pkg_check_modules(GLFW REQUIRED IMPORTED_TARGET GLOBAL glfw3)
 
     # Dear ImGui configuration
     set(IMGUI_DIR "${CMAKE_SOURCE_DIR}/submodules/imgui")
@@ -91,5 +84,6 @@ if (BUILD_GUI)
             glfw
             OpenGL::GL
             GLEW::GLEW
+            PkgConfig::GLFW
     )
 endif ()
