diff --git a/src/rsbuild.h.in b/src/rsbuild.h.in
index 70d9dc2..0809aae 100644
--- a/src/rsbuild.h.in
+++ b/src/rsbuild.h.in
@@ -20,7 +20,7 @@
 #ifndef RS_BUILD_H
 #define RS_BUILD_H
 
-#define RS_BUILD_GLOBAL_CONFIG "@CMAKE_INSTALL_PREFIX@/etc/replay-sorcery.conf"
+#define RS_BUILD_GLOBAL_CONFIG "/etc/replay-sorcery.conf"
 #define RS_BUILD_LOCAL_CONFIG "%s/.config/replay-sorcery.conf"
 
 #cmakedefine RS_BUILD_UNISTD_FOUND
