diff --git a/flasher.go b/flasher.go
index 523e4fd..e9eee28 100644
--- a/flasher.go
+++ b/flasher.go
@@ -184,7 +184,7 @@ func getPlatformTools() error {
 		fmt.Println(platformToolsZip + " checksum verification failed")
 		return err
 	}
-	platformToolsPath := cwd + string(os.PathSeparator) + "platform-tools" + string(os.PathSeparator)
+	platformToolsPath := ""
 	pathEnvironmentVariable := func() string {
 		if OS == "windows" {
 			return "Path"
