diff --git a/game_server/__init__.py b/game_server/__init__.py
index bb97f36..99035b7 100644
--- a/game_server/__init__.py
+++ b/game_server/__init__.py
@@ -12,10 +12,6 @@ from loguru import logger
 import traceback
 from os import path
 
-basepath = path.dirname(__file__)
-logsdir = path.abspath(path.join(basepath, "..", ".\\logs"))
-logfile = open(f"{logsdir}\\latest.log", "w")
-logfile.truncate(0)
 
 class Connection: 
     game_server: GameServer
@@ -33,14 +29,12 @@ class Connection:
         packet = Packet()
         packet.parse(data)
 
-        logfile.write(f'Receive: {packet.body}\n')
         logger.opt(colors=True).debug(f'<yellow>{self.peer.address}</yellow> Receive: <cyan>{packet.body}</cyan>')
         if handler := self.game_server.router.get(packet.cmdid):
             handler(self, packet.body)
 
     def send(self, msg: betterproto.Message):
         packet = bytes(Packet(body=msg))
-        logfile.write(f'Send: {msg}\n')
         # logger.opt(colors=True).debug(f'<yellow>{self.peer.address}</yellow> Send: <cyan>{msg}</cyan>')
         self.send_raw(bytes(packet))
 
@@ -101,4 +95,4 @@ class GameServer:
     def start(self):
         b = threading.Thread(name='GameServer', target=self.loop)
         b.daemon=True
-        b.start()
\ No newline at end of file
+        b.start()
