diff --git a/CMakeModules/QtConfiguration.cmake b/CMakeModules/QtConfiguration.cmake
index d74a484..fb678ad 100644
--- a/CMakeModules/QtConfiguration.cmake
+++ b/CMakeModules/QtConfiguration.cmake
@@ -53,8 +53,7 @@ foreach(COMP ${components})
   find_package(Qt6 REQUIRED COMPONENTS Gui)
   find_package(Qt6 REQUIRED COMPONENTS Quick)
   find_package(Qt6 REQUIRED COMPONENTS Widgets)
-  find_package(Qt6 REQUIRED COMPONENTS WebEngineQuick)
-  find_package(Qt6 REQUIRED COMPONENTS WebEngineCore)
+  find_package(Qt6 REQUIRED COMPONENTS WebEngine)
   find_package(Qt6 REQUIRED COMPONENTS OpenGL)
   find_package(Qt6 REQUIRED COMPONENTS DBus)
 
