diff --git a/modules/gallery.py b/modules/gallery.py
index f516f92..31c2911 100644
--- a/modules/gallery.py
+++ b/modules/gallery.py
@@ -38,7 +38,9 @@ class GalleryManager:
         img_dir = self._get_img_dir()
         # Use a generator to find image files, avoiding the creation of a full list
         def image_files_gen(directory):
-            for file in os.listdir(directory):
+            filelist = list(map(lambda file: os.path.join(directory, file), os.listdir(directory)))
+            filelist.sort(key=os.path.getctime, reverse=True)
+            for file in filelist:
                 if file.endswith(('.jpg', '.png')):
                     yield file
 
@@ -50,8 +52,7 @@ class GalleryManager:
         imgs = []
         for i, file_name in enumerate(image_files_gen(img_dir)):
             if start_index <= i < end_index:
-                image_path = os.path.join(img_dir, file_name)
-                image = Image.open(image_path)
+                image = Image.open(file_name)
                 imgs.append(image)
             elif i >= end_index:
                 break
@@ -145,7 +146,7 @@ class GalleryManager:
                     yield file_path
 
         # Sort files only when necessary and only the files we need
-        file_paths = sorted(image_file_gen(img_dir), key=os.path.getctime)
+        file_paths = sorted(image_file_gen(img_dir), key=os.path.getctime, reverse=True)
 
         # Handle index out of range errors
         try:
@@ -215,7 +216,7 @@ class GalleryManager:
                           for file in files
                           if os.path.isfile(os.path.join(img_dir, file)) and
                           file.lower().endswith(('.png', '.jpg'))]
-            file_paths.sort(key=os.path.getctime)
+            file_paths.sort(key=os.path.getctime, reverse=True)
             if total_imgs == 0:
                 self.sel_img = None
             if self.img_index == total_imgs:
