diff --git a/src/plugin/utils.cpp b/src/plugin/utils.cpp
index 7735f1c..d178848 100644
--- a/src/plugin/utils.cpp
+++ b/src/plugin/utils.cpp
@@ -350,7 +350,7 @@ std::string get_wine_version() {
     if (access(wineloader_path.c_str(), X_OK) == 0) {
         wine_path = wineloader_path;
     } else {
-        wine_path = bp::search_path("wine");
+        wine_path = "@wine@/bin/wine";
     }
 
     bp::ipstream output;
