From d42c22f8e2e15481a554b36ae2e837d13f300df4 Mon Sep 17 00:00:00 2001
From: David Guibert <david.guibert@gmail.com>
Date: Fri, 10 Mar 2017 09:35:34 +0100
Subject: [PATCH 7/9] xpra as float

---
 config.def.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config.def.h b/config.def.h
index bc5be7f..ec2b5b9 100644
--- a/config.def.h
+++ b/config.def.h
@@ -33,6 +33,7 @@ static const Rule rules[] = {
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
+	{ "Xpra",     NULL,       NULL,       0,            1,           -1 },
 	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
 };
 
-- 
2.19.2

