diff --git a/build.gradle.kts b/build.gradle.kts
index 5341c25fb..7450bb8f8 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -4,7 +4,6 @@ import org.gradle.api.tasks.testing.logging.TestLogEvent
 plugins {
     java
     `maven-publish`
-    id("com.github.johnrengelman.shadow") version "7.1.2" apply false
     id("io.papermc.paperweight.core") version "1.3.8"
 }
 
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 18cf95a32..91037a630 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -7,6 +7,18 @@ pluginManagement {
     }
 }
 
+buildscript {
+  repositories {
+    gradlePluginPortal()
+    maven {
+      url = uri("https://plugins.gradle.org/m2/")
+    }
+  }
+  dependencies {
+    classpath("gradle.plugin.com.github.johnrengelman:shadow:7.1.2")
+  }
+}
+
 rootProject.name = "paper"
 
 for (name in listOf("Paper-API", "Paper-Server", "Paper-MojangAPI")) {
