--- a/src/program/main.cpp
+++ b/src/program/main.cpp
@@ -25,7 +25,6 @@
 #include "utils.h" // create_dir
 
 #include <limits.h> // PATH_MAX
-#include <libgen.h> // dirname
 #include <signal.h> // kill
 #include <xcb/xcb.h>
 #define explicit _explicit
@@ -188,8 +187,7 @@ int main(int argc, char **argv)
     ssize_t count = readlink( "/proc/self/exe", buf, PATH_MAX );
     std::string binpath = std::string( buf, (count > 0) ? count : 0 );
     char* binpathptr = const_cast<char*>(binpath.c_str());
-    context.libtaspath = dirname(binpathptr);
-    context.libtaspath += "/libtas.so";
+    context.libtaspath = "@out@/lib/libtas.so";
 
     /* Create the working directories */
     char *path = getenv("XDG_CONFIG_HOME");
