diff --git a/configure.ac b/configure.ac
index 5c2b90d..7d650c3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,7 @@ AS_IF([test "x$enable_i386_lib" != "xyes"], [
 
     AC_SEARCH_LIBS([pthread_create], [pthread], [], [AC_MSG_ERROR(The pthread library is required!)])
 
-    PKG_CHECK_MODULES([LIBLUA], [lua53])
+    PKG_CHECK_MODULES([LIBLUA], [lua5.3])
     AC_SUBST([LIBLUA_CFLAGS])
     AC_SUBST([LIBLUA_LIBS])

