diff --git a/alvr/server_openvr/build.rs b/alvr/server_openvr/build.rs
index 99ae124..0cbf682 100644
--- a/alvr/server_openvr/build.rs
+++ b/alvr/server_openvr/build.rs
@@ -1,13 +1,11 @@
 use std::{env, path::PathBuf};
 
 fn get_vulkan_headers_path(platform_name: &str) -> PathBuf {
-    alvr_filesystem::deps_dir()
-        .join(if platform_name == "linux" {
-            "linux"
-        } else {
-            "windows"
-        })
-        .join("vulkan-headers/include")
+    if platform_name == "linux" {
+        PathBuf::from("@vulkanHeaders@/include")
+    } else {
+        alvr_filesystem::deps_dir().join("windows/vulkan-headers/include")
+    }
 }
 
 fn get_ffmpeg_path(platform_name: &str) -> PathBuf {
@@ -20,7 +18,7 @@ fn get_ffmpeg_path(platform_name: &str) -> PathBuf {
         .join("ffmpeg");
 
     if platform_name == "linux" {
-        ffmpeg_path.join("alvr_build")
+        PathBuf::from("@ffmpeg@")
     } else {
         ffmpeg_path
     }
@@ -28,7 +26,7 @@ fn get_ffmpeg_path(platform_name: &str) -> PathBuf {
 
 #[cfg(all(target_os = "linux", feature = "gpl"))]
 fn get_linux_x264_path() -> PathBuf {
-    alvr_filesystem::deps_dir().join("linux/x264/alvr_build")
+    PathBuf::from("@x264@")
 }
 
 fn main() {
