diff --git a/app/build.gradle.kts b/app/build.gradle.kts
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -83,6 +83,9 @@ android {
             enableAndroidTestCoverage = project.hasProperty("coverage")
         }
         release {
+            configure<CrashlyticsExtension> {
+                mappingFileUploadEnabled = false
+            }
             val tasks_mapbox_key: String? by project
             val tasks_google_key: String? by project
             val tasks_posthog_key: String? by project
diff --git a/wear-datalayer/build.gradle.kts b/wear-datalayer/build.gradle.kts
--- a/wear-datalayer/build.gradle.kts
+++ b/wear-datalayer/build.gradle.kts
@@ -42,11 +42,11 @@ else
 
 protobuf {
     protoc {
-        artifact = libs.protobuf.protoc.stnd.get().toString() + archSuffix
+        path = System.getenv("PROTOC_BIN")
     }
     plugins {
         create("javalite") {
-            artifact = libs.protobuf.protoc.gen.javalite.get().toString() + archSuffix
+            path = System.getenv("PROTOC_GEN_JAVALITE_BIN")
         }
         create("grpc") {
             artifact = libs.protobuf.protoc.gen.grpc.java.get().toString()
