diff --git a/app/src/main/java/org/tasks/ui/TaskListViewModel.kt b/app/src/main/java/org/tasks/ui/TaskListViewModel.kt
--- a/app/src/main/java/org/tasks/ui/TaskListViewModel.kt
+++ b/app/src/main/java/org/tasks/ui/TaskListViewModel.kt
@@ -128,8 +128,6 @@ class TaskListViewModel @Inject constructor(
                     Banner.NotificationsDisabled
                 preferences.warnAlarmsDisabled && !applicationContext.canScheduleExactAlarms() ->
                     Banner.AlarmsDisabled
-                inventory.begForMoney && !firebase.subscribeCooldown ->
-                    Banner.BegForMoney
                 accounts.any { it.isMicrosoft } && preferences.warnMicrosoft ->
                     Banner.WarnMicrosoft
                 accounts.any { it.isGoogleTasks } && preferences.warnGoogleTasks ->
