Description: increase test timeout
Author: Xavier Guimard <yadd@debian.org>
Bug: https://github.com/cyrusimap/cyrus-imapd/issues/3040
Bug-Debian: https://bugs.debian.org/960658
Forwarded: not-needed
Last-Update: 2020-05-20

--- a/cunit/unit.c
+++ b/cunit/unit.c
@@ -104,7 +104,7 @@
 }
 
 /* Each test gets a maximum of 20 seconds. */
-#define TEST_TIMEOUT_MS (20*1000)
+#define TEST_TIMEOUT_MS (240*1000)
 
 static jmp_buf jbuf;
 static const char *code;
