diff --git a/src/hint_drw.c b/src/hint_drw.c
index f2382db..fe3d0e6 100644
--- a/src/hint_drw.c
+++ b/src/hint_drw.c
@@ -218,7 +218,7 @@ static Pixmap create_backing_store(Display *dpy,
 {
 	size_t i;
 	uint8_t r, g, b;
-	XftFont *font = get_font("Monospace", hints[0].h);
+	XftFont *font = get_font("Hermit", hints[0].h);
 
 	XWindowAttributes info;
 	XGetWindowAttributes(dpy, DefaultRootWindow(dpy), &info);
