--- a/art/dex2oat/dex2oat.cc.orig	2026-10-09 20:11:17.515379755 +1100
+++ a/art/dex2oat/dex2oat.cc	2026-10-09 20:11:17.518379759 +1100
@@ -920,7 +920,7 @@
       if(android_root_.empty()) {
         Dl_info libart_so_dl_info;
         // art_sigsegv_fault chosen arbitrarily, what matters is that it's a symbol exported by by libart.so
-        dladdr(art_sigsegv_fault, &libart_so_dl_info);
+        dladdr((const void*)art_sigsegv_fault, &libart_so_dl_info);
         // make sure we didn't get NULL
         if(libart_so_dl_info.dli_fname) {
           // it's simpler if we can modify the string, so strdup it
