--- a/plugin/forward/forward.go	2026-05-29 22:07:59.185638107 -0700
+++ b/plugin/forward/forward.go	2026-05-29 22:08:55.468691985 -0700
@@ -161,13 +161,14 @@
 		})
 
 		var (
-			ret *dns.Msg
-			err error
+			ret     *dns.Msg
+			records []dns.RR
+			err     error
 		)
 		opts := f.opts
 
 		for {
-			ret, err = proxy.Connect(ctx, state, opts)
+			ret, records, err = proxy.Connect(ctx, state, opts)
 
 			if err == proxyPkg.ErrCachedClosed { // Remote side closed conn, can only happen with TCP.
 				continue
@@ -211,6 +212,25 @@
 			break
 		}
 
+		if records != nil {
+			ch := make(chan *dns.Envelope)
+			defer close(ch)
+			tr := new(dns.Transfer)
+			go tr.Out(w, r, ch)
+			j, l := 0, 0
+			for i, r := range records {
+				l += dns.Len(r)
+				if l > 65000 {
+					ch <- &dns.Envelope{RR: records[j:i]}
+					l = 0
+					j = i
+				}
+			}
+			if j < len(records) {
+				ch <- &dns.Envelope{RR: records[j:]}
+			}
+			return 0, nil
+		}
 		// Check if the reply is correct; if not return FormErr.
 		if !state.Match(ret) {
 			debug.Hexdumpf(ret, "Wrong reply for id: %d, %s %d", ret.Id, state.QName(), state.QType())
--- a/plugin/pkg/proxy/connect.go	2026-05-29 22:07:59.207777554 -0700
+++ b/plugin/pkg/proxy/connect.go	2026-05-29 22:08:24.646622293 -0700
@@ -103,7 +103,7 @@
 }
 
 // Connect selects an upstream, sends the request and waits for a response.
-func (p *Proxy) Connect(_ctx context.Context, state request.Request, opts Options) (*dns.Msg, error) {
+func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options) (*dns.Msg, []dns.RR, error) {
 	start := time.Now()
 
 	var proto string
@@ -118,12 +118,61 @@
 
 	pc, cached, err := p.transport.Dial(proto)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	// Set buffer size correctly for this client.
 	pc.c.UDPSize = max(uint16(state.Size()), 512) // #nosec G115 -- UDP size fits in uint16
 
+	var retRRs []dns.RR
+	var ret *dns.Msg
+
+	if state.QType() == dns.TypeAXFR || state.QType() == dns.TypeIXFR {
+		pc.c.SetWriteDeadline(time.Now().Add(maxTimeout))
+		if err := pc.c.WriteMsg(state.Req); err != nil {
+			pc.c.Close() // not giving it back
+			if err == io.EOF && cached {
+				return nil, nil, ErrCachedClosed
+			}
+			return nil, nil, err
+		}
+		first := true
+		for {
+			pc.c.SetReadDeadline(time.Now().Add(p.readTimeout))
+			in, err := pc.c.ReadMsg()
+			if err != nil {
+				pc.c.Close() // not giving it back
+				if err == io.EOF && cached {
+					return nil, nil, ErrCachedClosed
+				}
+				return ret, nil, err
+			}
+			if state.Req.Id != in.Id {
+				// out-of-order response. unexpected.
+				continue
+			}
+			if first {
+				if len(in.Answer) == 0 || in.Answer[0].Header().Rrtype != dns.TypeSOA {
+					pc.c.Close()
+					return nil, nil, dns.ErrSoa
+				}
+				first = !first
+				if len(in.Answer) == 1 {
+					retRRs = append(retRRs, in.Answer[0])
+					continue
+				}
+			}
+			for _, rr := range in.Answer {
+				retRRs = append(retRRs, rr)
+			}
+			if len(in.Answer) >= 0 && in.Answer[len(in.Answer)-1].Header().Rrtype == dns.TypeSOA {
+				break
+			}
+		}
+		p.transport.Yield(pc)
+		return nil, retRRs, nil
+	}
+
 	pc.c.SetWriteDeadline(time.Now().Add(maxTimeout))
 	// records the origin Id before upstream.
 	originId := state.Req.Id
@@ -135,12 +184,10 @@
 	if err := pc.c.WriteMsg(state.Req); err != nil {
 		pc.c.Close() // not giving it back
 		if err == io.EOF && cached {
-			return nil, ErrCachedClosed
+			return nil, nil, ErrCachedClosed
 		}
-		return nil, err
+		return nil, nil, err
 	}
-
-	var ret *dns.Msg
 	pc.c.SetReadDeadline(time.Now().Add(p.readTimeout))
 	for {
 		ret, err = pc.c.ReadMsg()
@@ -159,13 +206,13 @@
 
 			pc.c.Close() // not giving it back
 			if err == io.EOF && cached {
-				return nil, ErrCachedClosed
+				return nil, nil, ErrCachedClosed
 			}
 			// recovery the origin Id after upstream.
 			if ret != nil {
 				ret.Id = originId
 			}
-			return ret, err
+			return ret, nil, err
 		}
 		// drop out-of-order responses
 		if state.Req.Id == ret.Id {
@@ -184,7 +231,7 @@
 
 	requestDuration.WithLabelValues(p.proxyName, p.addr, rc).Observe(time.Since(start).Seconds())
 
-	return ret, nil
+	return ret, nil, nil
 }
 
 const cumulativeAvgWeight = 4
--- a/plugin/pkg/proxy/proxy_test.go	2026-05-29 22:07:59.207763123 -0700
+++ b/plugin/pkg/proxy/proxy_test.go	2026-05-29 22:08:24.646661828 -0700
@@ -36,7 +36,7 @@
 	rec := dnstest.NewRecorder(&test.ResponseWriter{})
 	req := request.Request{Req: m, W: rec}
 
-	resp, err := p.Connect(context.Background(), req, Options{PreferUDP: true})
+	resp, _, err := p.Connect(context.Background(), req, Options{PreferUDP: true})
 	if err != nil {
 		t.Errorf("Failed to connect to testdnsserver: %s", err)
 	}
@@ -67,7 +67,7 @@
 	rec := dnstest.NewRecorder(&test.ResponseWriter{})
 	req := request.Request{Req: m, W: rec}
 
-	_, err := p.Connect(context.Background(), req, Options{})
+	_, _, err := p.Connect(context.Background(), req, Options{})
 	if err == nil {
 		t.Fatal("Expected *not* to receive reply, but got one")
 	}
@@ -121,7 +121,7 @@
 				Req: m,
 			}
 
-			resp, err := p.Connect(context.Background(), req, tc.opts)
+			resp, _, err := p.Connect(context.Background(), req, tc.opts)
 			if err != nil {
 				t.Fatalf("Connect failed: %v", err)
 			}
@@ -214,7 +214,7 @@
 		recorder := dnstest.NewRecorder(&test.ResponseWriter{})
 		request := request.Request{Req: queryMsg, W: recorder}
 
-		response, err := p.Connect(context.Background(), request, options)
+		_, response, err := p.Connect(context.Background(), request, options)
 		if err != nil {
 			t.Errorf("Failed to connect to testdnsserver: %s", err)
 		}
