--- spotx.sh.orig	2025-07-20 20:27:39.529695104 -0400
+++ spotx.sh	2025-07-20 20:44:15.926225047 -0400
@@ -398,14 +398,7 @@
   for path in "${paths[@]}"; do
     local path="${path}"
     [[ ! -w "${path}" ]] && {
-      sudo -n true 2>/dev/null || {
-        echo -e "${yellow}Warning:${clr} SpotX-Bash does not have write permission in client directory.\nRequesting sudo permission..." >&2
-        sudo -v || {
-          echo -e "\n${red}Error:${clr} SpotX-Bash was not given sudo permission. Exiting...\n" >&2
-          exit 1
-        }
-      }
-      sudo chmod -R a+wr "${appPath}"
+      chmod -R a+wr "${appPath}"
     }
   done
 }
@@ -477,39 +470,24 @@
   }
 }
 
-sudo_check() {
-  command -v sudo &> /dev/null || { 
-    echo -e "\n${red}Error:${clr} sudo command not found. Install sudo or run this script as root.\n" >&2
-    exit 1
-  }
-  sudo -n true &> /dev/null || {
-    echo -e "This script requires sudo permission to install the client.\nPlease enter your sudo password..."
-    sudo -v || { 
-      echo -e "\n${red}Error:${clr} Failed to obtain sudo permission. Exiting...\n" >&2
-      exit 1
-    }
-  }
-}
-
 linux_working_dir() { [[ -d "/tmp" ]] && workDir="/tmp" || workDir="$HOME"; }
 
 linux_deb_install() {
-  sudo_check
   linux_working_dir
   lc01=$(echo "=kjQ59EeBNEZwhGWad2cq1Ub0QUSpRzRYtmVHJGcG1mWnF1VZZHetJ2M5ckWnFlbixGbHJGRCNlZ5hnMZdjUp9Ue502Y5ZVVmtmVtN2NSlmYjp0QJxWMDlkdoJTWsJUeld2dIZ2ZJNlTpZUbj5mUpl0Z3dkYxUjMMJjVHpldBlnY0FUejRXQTNFdBlmW0F0UjRXQDJWeWNTW" | rev | base64 --decode | base64 --decode)
   lc02=$(echo "9ADSJdTRElEMsdUZsJUePlXWpB1ZJlmYjJ1VaNHbXlVbCNkWolzRiVHZzI2aCNEZ1Z1VhNnTFlUOKhkYqRHSKZTSzIWeKhlU5I1ValHdIpUd4xWSnV1VMdGOHFmaWdUS3I0QmhjQplUMJdVW5R2RKlWQplUOKhVWXZ1RiBnWyU2a4MlZ5x2RSJnSzI2M0hkSpFUeiRXQT50Zr52YwYVbjRHMDlEdBlXU0FUaaRXQpNGaKdFT65EWalHZyIWeChFT0F0UjRXQDJWeWNTW" | rev | base64 --decode | base64 --decode)
   eval "${lc01}"; eval "${lc02}"
   printf "\xE2\x9C\x94\x20\x44\x6F\x77\x6E\x6C\x6F\x61\x64\x65\x64\x20\x61\x6E\x64\x20\x69\x6E\x73\x74\x61\x6C\x6C\x69\x6E\x67\x20\x53\x70\x6F\x74\x69\x66\x79\n"
-  [[ -f "${appBak}" ]] && sudo rm "${appBak}" 2>/dev/null
-  [[ -f "${xpuiBak}" ]] && sudo rm "${xpuiBak}" 2>/dev/null
-  [[ -d "${xpuiDir}" ]] && sudo rm -rf "${xpuiDir}" 2>/dev/null
-  sudo dpkg -i "${workDir}/${fileVar}" &>/dev/null || {
-    sudo apt-get -f install -y &>/dev/null || {
+  [[ -f "${appBak}" ]] && rm "${appBak}" 2>/dev/null
+  [[ -f "${xpuiBak}" ]] && rm "${xpuiBak}" 2>/dev/null
+  [[ -d "${xpuiDir}" ]] && rm -rf "${xpuiDir}" 2>/dev/null
+  dpkg -i "${workDir}/${fileVar}" &>/dev/null || {
+    apt-get -f install -y &>/dev/null || {
       rm "${workDir}/${fileVar}" 2>/dev/null
       echo -e "\n${red}Error:${clr} Failed to install missing dependencies. Exiting...\n" >&2
       exit 1
     }
-  } && sudo dpkg -i "${workDir}/${fileVar}" &>/dev/null || {
+  } && dpkg -i "${workDir}/${fileVar}" &>/dev/null || {
     rm "${workDir}/${fileVar}" 2>/dev/null
     echo -e "\n${red}Error:${clr} Client install failed. Exiting...\n" >&2
     exit 1
