diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe91d8b..8791d67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,25 +9,32 @@ include(CTest)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-set(CADICAL_DIR ${CMAKE_BINARY_DIR}/solvers/src/cadical_external/build)
-set(CADICAL_LIB ${CADICAL_DIR}/libcadical.a)
-if (EXISTS ${CADICAL_LIB})
-    message(STATUS "Cadical already built")
-    add_library(cadical STATIC IMPORTED)
-    set_target_properties(cadical PROPERTIES IMPORTED_LOCATION "${CADICAL_LIB}")
+find_library(CADICAL_LIB libcadical.a)
+
+if (${CADICAL_LIB} MATCHES "CADICAL_LIB-NOTFOUND")
+    set(CADICAL_DIR ${CMAKE_BINARY_DIR}/solvers/src/cadical_external/build)
+    set(CADICAL_LIB ${CADICAL_DIR}/libcadical.a)
+    if (EXISTS ${CADICAL_LIB})
+        message(STATUS "Cadical already built")
+        add_library(cadical STATIC IMPORTED)
+        set_target_properties(cadical PROPERTIES IMPORTED_LOCATION "${CADICAL_LIB}")
+    else()
+        message(STATUS "Building Cadical")
+        include(ExternalProject)
+        ExternalProject_Add(cadical_external
+            GIT_REPOSITORY https://github.com/arminbiere/cadical.git
+            GIT_TAG master
+            PREFIX solvers
+            CONFIGURE_COMMAND ./configure "CXXFLAGS=-fPIC"
+            BUILD_COMMAND make -j
+            INSTALL_COMMAND ""
+            BUILD_BYPRODUCTS "${CADICAL_LIB}"
+            BUILD_IN_SOURCE 1
+        )
+    endif()
 else()
-    message(STATUS "Building Cadical")
-    include(ExternalProject)
-    ExternalProject_Add(cadical_external
-        GIT_REPOSITORY https://github.com/arminbiere/cadical.git
-        GIT_TAG master
-        PREFIX solvers
-        CONFIGURE_COMMAND ./configure "CXXFLAGS=-fPIC"
-        BUILD_COMMAND make -j
-        INSTALL_COMMAND ""
-        BUILD_BYPRODUCTS "${CADICAL_LIB}"
-        BUILD_IN_SOURCE 1
-    )
+    message("Found system-wide CaDiCaL version at ${CADICAL_LIB}")
+    add_library(cadical STATIC IMPORTED)
 endif()
     
 add_library(solver STATIC IMPORTED)
@@ -65,4 +72,6 @@ install(TARGETS gbdc DESTINATION .)
 add_test(NAME Test_StreamBuffer COMMAND "test/tests_streambuffer")
 # add_test(NAME Test_Feature_Extraction COMMAND "test/tests_feature_extraction")
 add_test(NAME Test_StreamCompressor COMMAND "test/tests_streamcompressor")
-add_test(NAME Test_GBDLib COMMAND "test/tests_gbdlib")
\ No newline at end of file
+add_test(NAME Test_GBDLib COMMAND "test/tests_gbdlib")
+
+install(TARGETS gbdctool)
