diff --git a/src/index.ts b/src/index.ts
index af41948..8fe7cd5 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -17,25 +17,7 @@ export async function activate(context: ExtensionContext): Promise<void> {
 
   const bin = ctx.resolveBin();
   if (!bin) {
-    let msg = 'Rust Analyzer is not found, download from GitHub release?';
-    let ret = 0;
-    if (ctx.config.prompt) {
-      ret = await window.showQuickpick(['Yes', 'Cancel'], msg);
-    }
-    if (ret === 0) {
-      try {
-        const latest = await getLatestRelease(ctx.config.channel);
-        if (!latest) throw new Error('Failed to get latest release');
-        await downloadServer(context, latest);
-      } catch (e) {
-        console.error(e);
-        msg = 'Download rust-analyzer failed, you can get it from https://github.com/rust-analyzer/rust-analyzer';
-        window.showMessage(msg, 'error');
-        return;
-      }
-    } else {
-      return;
-    }
+    return;
   }
 
   // internal commands that are invoked by server
