--- a/setup.py	2020-11-28 13:42:51.165550768 +0700
+++ b/setup.py	2020-11-28 13:44:09.986701051 +0700
@@ -6,7 +6,7 @@
     version=version,
     scripts=['electrumx_server', 'electrumx_rpc', 'electrumx_compact_history'],
     python_requires='>=3.7',
-    install_requires=['aiorpcX[ws]>=0.18.3,<0.19', 'attrs',
+    install_requires=['aiorpcX>=0.18.3,<0.19', 'attrs',
                       'plyvel', 'pylru', 'aiohttp>=3.3'],
     extras_require={
         'rapidjson': ['python-rapidjson>=0.4.1,<1.0.0'],
