diff --git a/src/mpid/ch3/channels/common/src/util/error_handling.c b/src/mpid/ch3/channels/common/src/util/error_handling.c
index b6c67c4..e8b43ad 100644
--- a/src/mpid/ch3/channels/common/src/util/error_handling.c
+++ b/src/mpid/ch3/channels/common/src/util/error_handling.c
@@ -69,7 +69,7 @@ int show_backtrace = 0;
 // Signal handler for errors
 void error_sighandler(int sig, siginfo_t *info, void *secret) {
     // Always print error
-    PRINT_ERROR( "Caught error: %s (signal %d)\n", sys_siglist[sig], sig );
+    PRINT_ERROR( "Caught error: %s (signal %d)\n", strsignal(sig), sig );
     // Show backtrace if required
     if (show_backtrace) print_backtrace();
     // Raise the signal again with default handler
diff --git a/src/pm/mpirun/mpispawn.c b/src/pm/mpirun/mpispawn.c
index ed8d2c3..5f1842e 100644
--- a/src/pm/mpirun/mpispawn.c
+++ b/src/pm/mpirun/mpispawn.c
@@ -800,7 +800,7 @@ void child_handler(int signal)
     gethostname(my_host_name, MAX_HOST_LEN);
 
     rank = mt_id;
-    PRINT_DEBUG(DEBUG_Fork_verbose, "mpispawn child_handler: got signal %d: %s\n", signal, sys_siglist[signal]);
+    PRINT_DEBUG(DEBUG_Fork_verbose, "mpispawn child_handler: got signal %d: %s\n", signal, strsignal(signal));
     while (1) {
         do {
             pid = waitpid(-1, &status, WNOHANG);
