From 160b90e135e0dad21b4dd1a09c61c982e4dbabeb Mon Sep 17 00:00:00 2001
From: "P. R. d. O" <d.ol.rod@protonmail.com>
Date: Tue, 16 Nov 2021 08:10:37 -0600
Subject: [PATCH] Adding brave and vivaldi user installation

---
 app/native-autoinstall.js | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/app/native-autoinstall.js b/app/native-autoinstall.js
index 118c019..cfb7da3 100644
--- a/app/native-autoinstall.js
+++ b/app/native-autoinstall.js
@@ -172,13 +172,20 @@ function LinuxInstall() {
 	var manifests;
 	if(mode=="user") 
 		manifests = [{
-			file: process.env.HOME+"/.mozilla/native-messaging-hosts/"+config.id+".json",			manifest: JSON.stringify(firefoxManifest,null,4),
+			file: process.env.HOME+"/.mozilla/native-messaging-hosts/"+config.id+".json",
+			manifest: JSON.stringify(firefoxManifest,null,4),
 		},{
 			file: process.env.HOME+"/.config/google-chrome/NativeMessagingHosts/"+config.id+".json",
 			manifest: JSON.stringify(chromeManifest,null,4),
 		},{
 			file: process.env.HOME+"/.config/chromium/NativeMessagingHosts/"+config.id+".json",
 			manifest: JSON.stringify(chromeManifest,null,4),
+		},{
+			file: process.env.HOME+"/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts/"+config.id+".json",
+			manifest: JSON.stringify(chromeManifest,null,4),
+		},{
+			file: process.env.HOME+"/.config/vivaldi/NativeMessagingHosts/"+config.id+".json",
+			manifest: JSON.stringify(chromeManifest,null,4),
 		}];
 	else {
 		manifests = [{
@@ -221,7 +228,9 @@ function LinuxUninstall() {
 		manifests = [
 			process.env.HOME+"/.mozilla/native-messaging-hosts/"+config.id+".json",
 			process.env.HOME+"/.config/google-chrome/NativeMessagingHosts/"+config.id+".json",
-			process.env.HOME+"/.config/chromium/NativeMessagingHosts/"+config.id+".json"
+			process.env.HOME+"/.config/chromium/NativeMessagingHosts/"+config.id+".json",
+			process.env.HOME+"/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts/"+config.id+".json",
+			process.env.HOME+"/.config/vivaldi/NativeMessagingHosts/"+config.id+".json"
 		];
 	else
 		manifests = [
-- 
2.33.0

