diff --git a/CMakeLists.txt b/CMakeLists.txt
index da14c80..91cf6f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,9 +146,8 @@ if(NOT BUILD_SHARED_LIBS)
 endif()
 
 string(CONFIGURE [[
-prefix=${pcfiledir}/@plutovg_pc_prefix_relative@
-includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: PlutoVG
 Description: Tiny 2D vector graphics library in C
