--- Makefile	1970-01-01 01:00:01.000000000 +0100
+++ Makefile.new	2021-06-09 14:25:33.219942771 +0200
@@ -25,7 +25,7 @@
 ALL_SRCS:=$(wildcard *.cpp *.c)
 ALL_OBJS=$(patsubst %.cpp,%.o,$(patsubst %.c,%.o,$(ALL_SRCS)))
 
-PROGSDIR=../data
+PROGSDIR=./.
 ALL=pb readpb readdiv tracecomp bpcomp readcv cvrep sumcv ppred tree2ps bf stopafter subsample ancestral 
 #ALL=makeconcat makerandomconcat taxasubsample pb stoppb readpb readdiv tracecomp bpcomp readcv cvrep sumcv ppred tree2ps bf stopafter subsample ancestral
 PROGS=$(addprefix $(PROGSDIR)/, $(ALL))
