diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,11 +37,10 @@
 
 if (UNIX)
 	find_package(PkgConfig REQUIRED)
-	pkg_check_modules(GRAPHVIZ libgvc>=2.38.0)
+	pkg_check_modules(GRAPHVIZ IMPORTED_TARGET libgvc>=2.38.0)
 	if(GRAPHVIZ_FOUND)
-		include_directories(${GRAPHVIZ_INCLUDE_DIRS})
-		link_directories(${GRAPHVIZ_LIBRARY_DIRS})
-		add_definitions(-DENABLE_GRAPH)
+		add_definitions("-DENABLE_GRAPH")
+		link_libraries(PkgConfig::GRAPHVIZ)
 	endif()
 endif()
 
@@ -96,13 +95,12 @@
 # FIXME: This is also useful on Windows, so it should be made usable there too.
 if(UNIX)
 	if(TARGET_ARCH_FAMILY_X86)
-		pkg_check_modules(GDTOA gdtoa-desktop)
+		pkg_check_modules(GDTOA IMPORTED_TARGET gdtoa-desktop)
 		if(NOT GDTOA_FOUND)
 			message(WARNING "gdtoa-desktop package wasn't found. 80-bit floating-point values will be shown with max_digits10 digits of precision instead of shortest representation.")
 		else()
 			add_definitions("-DHAVE_GDTOA")
-			include_directories(${GDTOA_INCLUDE_DIRS})
-			link_directories(${GDTOA_LIBRARY_DIRS})
+			link_libraries(PkgConfig::GDTOA)
 		endif()
 	endif()
 endif()
