diff --git a/src/controllers/uploads.cr b/src/controllers/uploads.cr
index 529bbfbe..3114c7a8 100644
--- a/src/controllers/uploads.cr
+++ b/src/controllers/uploads.cr
@@ -16,8 +16,9 @@ class UploadsController
           extension = File.extname(extension)
           filename = "#{filename}#{extension}"
         end
-        Dir.mkdir_p(File.join(Kemal.config.public_folder, filepath))
-        File.open(File.join(Kemal.config.public_folder, filepath, filename), "w") do |file|
+        state_dir = ENV["STATE_DIRECTORY"]? || Kemal.config.public_folder
+        Dir.mkdir_p(File.join(state_dir, filepath))
+        File.open(File.join(state_dir, filepath, filename), "w") do |file|
           IO.copy(part.body, file)
         end
       end
