From 15023857d89bcb82f7bd466e7b116d151fd46c25 Mon Sep 17 00:00:00 2001
From: Pim Snel <post@pimsnel.com>
Date: Wed, 5 Jan 2022 08:53:43 +0100
Subject: [PATCH] add chscale with cyclefonts

---
 x.c | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/x.c b/x.c
index 89786b8..706bdfb 100644
--- a/x.c
+++ b/x.c
@@ -59,6 +59,7 @@ static void zoom(const Arg *);
 static void zoomabs(const Arg *);
 static void zoomreset(const Arg *);
 static void ttysend(const Arg *);
+static void cyclefonts(const Arg *);
 
 /* config.h for applying patches and the configuration. */
 #include "config.h"
@@ -951,6 +952,19 @@ xloadfont(Font *f, FcPattern *pattern)
 	return 0;
 }
 
+void
+cyclefonts(const Arg *dummy)
+{
+	fonts_current++;
+	if (fonts_current > (sizeof fonts / sizeof fonts[0]) - 1) {
+		fonts_current = 0;
+	}
+	usedfont = fonts[fonts_current];
+	chscale = chscales[fonts_current];
+	xloadfonts(fonts[fonts_current], 0);
+	redraw();
+}
+
 void
 xloadfonts(const char *fontstr, double fontsize)
 {
@@ -1115,8 +1129,9 @@ xinit(int cols, int rows)
 	if (!FcInit())
 		die("could not init fontconfig.\n");
 
-	usedfont = (opt_font == NULL)? font : opt_font;
-	xloadfonts(usedfont, 0);
+	usedfont = fonts[fonts_current];
+	chscale = chscales[fonts_current];
+	xloadfonts(fonts[fonts_current], 0);
 
 	/* colors */
 	xw.cmap = XDefaultColormap(xw.dpy, xw.scr);
-- 
2.31.1

