From 857c71671c915330a8846c668f0e19b26b215d71 Mon Sep 17 00:00:00 2001
From: John Murray <5672686+JRMurr@users.noreply.github.com>
Date: Sun, 14 Jan 2024 15:32:42 -0500
Subject: [PATCH] fix lint

---
 client/src/cache.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/src/cache.rs b/client/src/cache.rs
index f743d83..66b380c 100644
--- a/client/src/cache.rs
+++ b/client/src/cache.rs
@@ -14,7 +14,7 @@ use std::str::FromStr;
 use anyhow::{anyhow, Result};
 use serde::{Deserialize, Serialize};
 
-pub use attic::cache::{CacheName, CacheNamePattern};
+pub use attic::cache::{CacheName};
 
 /// A reference to a cache.
 #[derive(Debug, Clone)]
