diff -ru a/Telegram/SourceFiles/data/data_stories.cpp b/Telegram/SourceFiles/data/data_stories.cpp
--- a/Telegram/SourceFiles/data/data_stories.cpp	2026-09-05 12:12:09.890231721 -0700
+++ b/Telegram/SourceFiles/data/data_stories.cpp	2026-09-05 12:12:10.553536361 -0700
@@ -7,6 +7,7 @@
 */
 #include "data/data_stories.h"
 
+#include "base/options.h"
 #include "base/unixtime.h"
 #include "apiwrap.h"
 #include "core/application.h"
@@ -144,6 +145,15 @@
 
 } // namespace
 
+const char kOptionDisableStories[] = "disable-stories";
+
+base::options::toggle DisableStories({
+	.id = kOptionDisableStories,
+	.name = "Disable stories",
+	.description = "",
+	.defaultValue = false,
+});
+
 std::vector<StoryId> RespectingPinned(const StoriesIds &ids) {
 	if (ids.pinnedToTop.empty()) {
 		return ids.list;
@@ -433,6 +443,9 @@
 void Stories::parseAndApply(
 		const MTPPeerStories &stories,
 		ParseSource source) {
+	if (DisableStories.value()) {
+		return;
+	}
 	const auto &data = stories.data();
 	const auto peerId = peerFromMTP(data.vpeer());
 	const auto already = _readTill.find(peerId);
@@ -718,6 +731,9 @@
 }
 
 void Stories::loadMore(StorySourcesList list) {
+	if (DisableStories.value()) {
+		return;
+	}
 	const auto index = static_cast<int>(list);
 	if (_loadMoreRequestId[index] || _sourcesLoaded[index]) {
 		return;
只在 b/Telegram/SourceFiles/data 中存在：data_stories.cpp.orig
diff -ru a/Telegram/SourceFiles/data/data_stories.h b/Telegram/SourceFiles/data/data_stories.h
--- a/Telegram/SourceFiles/data/data_stories.h	2026-09-05 12:12:09.890268189 -0700
+++ b/Telegram/SourceFiles/data/data_stories.h	2026-09-05 12:12:10.553834831 -0700
@@ -38,6 +38,8 @@
 struct StoryAlbum;
 class StoryPreload;
 
+extern const char kOptionDisableStories[];
+
 struct StoriesIds {
 	std::vector<StoryId> list; // flat_set<int, greater> for saved/archive.
 	std::vector<StoryId> pinnedToTop;
diff -ru a/Telegram/SourceFiles/settings/settings_experimental.cpp b/Telegram/SourceFiles/settings/settings_experimental.cpp
--- a/Telegram/SourceFiles/settings/settings_experimental.cpp	2026-09-05 12:12:09.928880220 -0700
+++ b/Telegram/SourceFiles/settings/settings_experimental.cpp	2026-09-05 12:12:12.886030051 -0700
@@ -33,6 +33,7 @@
 #include "core/launcher.h"
 #include "core/sandbox.h"
 #include "chat_helpers/tabbed_panel.h"
+#include "data/data_stories.h"
 #include "dialogs/dialogs_entry.h"
 #include "dialogs/dialogs_widget.h"
 #include "dialogs/ui/dialogs_layout.h"
@@ -336,6 +337,7 @@
 			registerHighlight);
 	};
 
+	addToggle(Data::kOptionDisableStories);
 	addToggle(ChatHelpers::kOptionTabbedPanelShowOnClick);
 	addToggle(Dialogs::kOptionForumHideChatsList);
 	addToggle(Dialogs::kOptionDialogsUnreadOnTop);
只在 b/Telegram/SourceFiles/settings 中存在：settings_experimental.cpp.orig
