diff --git a/src/tubefeed/provider/youtube/yt_api/YTPlaylist.py b/src/tubefeed/provider/youtube/yt_api/YTPlaylist.py
index 23a2bcb..1489148 100644
--- a/src/tubefeed/provider/youtube/yt_api/YTPlaylist.py
+++ b/src/tubefeed/provider/youtube/yt_api/YTPlaylist.py
@@ -18,10 +18,7 @@ class YTPlaylist:
         self.description: str = item['snippet']['description']
         self.thumbnails: list[YTThumbnail] = [YTThumbnail(n, t) for n, t in item['snippet']['thumbnails'].items()]
 
-        try:
-            self.published: datetime = datetime.strptime(item['snippet']['publishedAt'], "%Y-%m-%dT%H:%M:%S.%fZ")
-        except ValueError:
-            self.published: datetime = datetime.strptime(item['snippet']['publishedAt'], "%Y-%m-%dT%H:%M:%SZ")
+        self.published: datetime = datetime.fromisoformat(item['snippet']['publishedAt'])
 
     @staticmethod
     def from_id(yt: YouTube, id: str) -> 'YTPlaylist':
