diff --git a/xdis/magics.py b/xdis/magics.py
index c6f4d465..c632ff19 100755
--- a/xdis/magics.py
+++ b/xdis/magics.py
@@ -573,12 +573,12 @@ add_canonic_versions(
 )
 
 add_canonic_versions(
-    "3.12 3.12.0 3.12.1 3.12.2 3.12.3 3.12.4 3.12.5 3.12.6 3.12.7 3.12.8 3.12.9 3.12.10",
+    "3.12 3.12.0 3.12.1 3.12.2 3.12.3 3.12.4 3.12.5 3.12.6 3.12.7 3.12.8 3.12.9 3.12.10 3.12.11",
     "3.12.0rc2",
 )
 
 add_canonic_versions(
-    "3.13 3.13.0 3.13.1 3.13.2 3.13.3",
+    "3.13 3.13.0 3.13.1 3.13.2 3.13.3 3.13.4",
     "3.13.0rc3",
 )
 
diff --git a/xdis/op_imports.py b/xdis/op_imports.py
index 550cc10e..0bbd6b7e 100644
--- a/xdis/op_imports.py
+++ b/xdis/op_imports.py
@@ -146,7 +146,25 @@ op_imports = {
     "3.11a7e": opcode_311,
     "3.12.0rc2": opcode_312,
     "3.12.0": opcode_312,
+    "3.12.1": opcode_312,
+    "3.12.10": opcode_312,
+    "3.12.11": opcode_312,
+    "3.12.12": opcode_312,
+    "3.12.13": opcode_312,
+    "3.12.2": opcode_312,
+    "3.12.3": opcode_312,
+    "3.12.4": opcode_312,
+    "3.12.5": opcode_312,
+    "3.12.6": opcode_312,
+    "3.12.7": opcode_312,
+    "3.12.8": opcode_312,
+    "3.12.9": opcode_312,
     "3.13.0rc3": opcode_313,
+    "3.13.0": opcode_313,
+    "3.13.1": opcode_313,
+    "3.13.2": opcode_313,
+    "3.13.3": opcode_313,
+    "3.13.4": opcode_313,
     "2.6pypy": opcode_26pypy,
     "2.7pypy": opcode_27pypy,
     "2.7.12pypy": opcode_27pypy,
