diff --git a/coq/consts.py b/coq/consts.py
index 06f0725b..956b622b 100644
--- a/coq/consts.py
+++ b/coq/consts.py
@@ -9,7 +9,7 @@ TOP_LEVEL = Path(__file__).resolve(strict=True).parent.parent
 REQUIREMENTS = TOP_LEVEL / "requirements.txt"
 
 
-VARS = TOP_LEVEL / ".vars"
+VARS = Path("~/.coq_nvim").expanduser()
 
 RT_DIR = VARS / "runtime"
 RT_PY = RT_DIR / "Scripts" / "python.exe" if IS_WIN else RT_DIR / "bin" / "python3"
