From 46affb032e9a5dba21166f43fbb398160d61c1ce Mon Sep 17 00:00:00 2001
From: Bart Oostveen <bart@bartoostveen.nl>
Date: Fri, 26 Jun 2026 20:37:05 +0200
Subject: [PATCH 2/5] fix(build): do not run git clean

---
 packages/icons/lib/generate-library.cjs | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/packages/icons/lib/generate-library.cjs b/packages/icons/lib/generate-library.cjs
index fd1a5083886..e750d875633 100644
--- a/packages/icons/lib/generate-library.cjs
+++ b/packages/icons/lib/generate-library.cjs
@@ -107,7 +107,6 @@ const SVG_ATTRIBUTE_WITH_DASHES = [
 // `src/library/index.ts`.
 async function main() {
 	await ensureSvgFilesTracked();
-	await cleanup();
 	await generateTsxFiles();
 	await generateIndex();
 	await validateCollection();
@@ -157,15 +156,6 @@ async function ensureSvgFilesTracked() {
 	}
 }
 
-async function cleanup() {
-	await execFileAsync( 'git', [ 'clean', '-Xfq', ICON_LIBRARY_DIR ], {
-		env: {
-			...process.env,
-			GIT_WORK_TREE: undefined,
-		},
-	} );
-}
-
 // Generate src/library/*.tsx based on the available SVG files.
 async function generateTsxFiles( svgFiles ) {
 	if ( svgFiles ) {
-- 
2.55.0

