From 8e6d9707be249a5aa35fbffa99e95221a08b1e76 Mon Sep 17 00:00:00 2001
From: Bart Oostveen <bart@bartoostveen.nl>
Date: Sun, 5 Jul 2026 21:27:35 +0200
Subject: [PATCH 4/5] fix(build): do not run npm ci

---
 bin/build-plugin-zip.sh | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/bin/build-plugin-zip.sh b/bin/build-plugin-zip.sh
index 7eacc823530..761ed2010b5 100755
--- a/bin/build-plugin-zip.sh
+++ b/bin/build-plugin-zip.sh
@@ -73,10 +73,6 @@ if [ -z "$NO_CHECKS" ]; then
 	fi
 fi
 
-# Run the build.
-status "Installing dependencies... 📦"
-npm cache verify
-npm ci
 status "Generating build... 👷‍♀️"
 npm run build
 
-- 
2.55.0

