From 0ac564d53ee8f713d460f5cc7dd51164c769828d Mon Sep 17 00:00:00 2001
From: Bart Oostveen <bart@bartoostveen.nl>
Date: Sun, 5 Jul 2026 21:27:35 +0200
Subject: [PATCH 4/5] fix(build): do not run npm ci

---
 tools/release/build-plugin-zip.mts | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/tools/release/build-plugin-zip.mts b/tools/release/build-plugin-zip.mts
index 91c72dcde6a..cd9f7d82bad 100644
--- a/tools/release/build-plugin-zip.mts
+++ b/tools/release/build-plugin-zip.mts
@@ -292,10 +292,6 @@ async function main(): Promise< void > {
 		await resetRepository();
 	}
 
-	status( 'Installing dependencies... 📦' );
-	run( 'npm', [ 'cache', 'verify' ] );
-	run( 'npm', [ 'ci' ] );
-
 	status( 'Generating build... 👷‍♀️' );
 	run( 'npm', [ 'run', 'build' ] );
 
-- 
2.55.0

