diff --git a/build.rs b/build.rs
index cc8ed00..0941fa4 100644
--- a/build.rs
+++ b/build.rs
@@ -4,8 +4,7 @@ use std::path::Path;

 fn compile_blueprint<T: ToString>(path: T) -> Result<String, String> {
     // python blueprint-compiler/blueprint-compiler.py compile ui/main.blp
-    let output = Command::new("python3")
-        .arg("blueprint-compiler/blueprint-compiler.py")
+    let output = Command::new("blueprint-compiler")
         .arg("compile")
         .arg(path.to_string())
         .stdout(Stdio::piped())
