diff --git a/bcml/__main__.py b/bcml/__main__.py
index f781840..2370b8a 100644
--- a/bcml/__main__.py
+++ b/bcml/__main__.py
@@ -2,7 +2,6 @@ import os
 import sys
 from contextlib import redirect_stderr, redirect_stdout
 from multiprocessing import set_start_method, Process
-from os import chmod  # pylint: disable=ungrouped-imports
 from pathlib import Path
 from subprocess import Popen, DEVNULL
 from shutil import rmtree
@@ -60,8 +59,6 @@ def main(debug: bool = False):
 
     try:
         if SYSTEM != "Windows":
-            chmod(util.get_exec_dir() / "helpers/msyt", int("755", 8))
-            chmod(util.get_exec_dir() / "helpers/7z", int("755", 8))
             os.setpgrp()
         LOG.parent.mkdir(parents=True, exist_ok=True)
         for folder in util.get_work_dir().glob("*"):
diff --git a/bcml/util.py b/bcml/util.py
index 5db4308..71127b3 100644
--- a/bcml/util.py
+++ b/bcml/util.py
@@ -1376,7 +1376,6 @@ def get_7z_path():
     if system() == "Windows":
         return str(get_exec_dir() / "helpers" / "7z.exe")
     bundle_path = get_exec_dir() / "helpers" / "7z"
-    os.chmod(bundle_path, 0o755)
     if get_settings("force_7z"):
         return str(bundle_path)
     return shutil.which("7z") or str(bundle_path)
