diff --git a/bcml/_api.py b/bcml/_api.py
index 1a1d8ed..5291e94 100644
--- a/bcml/_api.py
+++ b/bcml/_api.py
@@ -60,11 +60,7 @@ class Api:
         self.tmp_files = []
 
     def get_ver(self, params=None):
-        return {
-            "version": USER_VERSION,
-            "update": util.get_latest_bcml() > VERSION
-            and not util.get_settings("suppress_update"),
-        }
+        return {"version": USER_VERSION}
 
     @win_or_lose
     def sanity_check(self, kwargs=None):
@@ -722,52 +718,6 @@ class Api:
         help_thread = Thread(target=help_window, args=(self.host,))
         help_thread.start()
 
-    @win_or_lose
-    def update_bcml(self):
-        if util.get_latest_bcml() <= VERSION:
-            return
-        exe = util.get_python_exe().replace("pythonw", "python")
-        args = [
-            exe,
-            "-m",
-            "pip",
-            "install",
-            "--disable-pip-version-check",
-            "--no-warn-script-location",
-            "--upgrade",
-            "bcml",
-        ]
-        if DEBUG:
-            args.insert(-2, "--pre")
-        parent = util.get_exec_dir().parent
-        if parent.name == "pkgs":
-            try:
-                (parent / "bcml").rename(parent / "bcml.bak")
-            except (OSError, PermissionError, FileExistsError, FileNotFoundError):
-                pass
-        if SYSTEM == "Windows":
-            result = run(
-                args,
-                creationflags=util.CREATE_NO_WINDOW,
-                capture_output=True,
-                check=False,
-                text=True,
-            )
-        else:
-            result = run(
-                args,
-                capture_output=True,
-                check=False,
-                text=True,
-            )
-        if result.stderr:
-            if parent.name == "pkgs":
-                try:
-                    (parent / "bcml.bak").rename(parent / "bcml")
-                except (OSError, PermissionError, FileExistsError, FileNotFoundError):
-                    pass
-            raise RuntimeError(result.stderr)
-
     def restart(self):
         opener = Thread(target=start_new_instance)
         opener.start()
diff --git a/bcml/assets/src/js/App.jsx b/bcml/assets/src/js/App.jsx
index 2d0d965..1979059 100644
--- a/bcml/assets/src/js/App.jsx
+++ b/bcml/assets/src/js/App.jsx
@@ -35,7 +35,6 @@ class App extends React.Component {
             confirmText: "",
             confirmCallback: () => {},
             showAbout: false,
-            update: false,
             version: "3.0"
         };
         this.selects = null;
@@ -275,33 +274,6 @@ class App extends React.Component {
             .catch(this.props.onError);
     };
 
-    updateBcml = () => {
-        this.setState(
-            {
-                progressTitle: "Upgrading BCML",
-                progressStatus: "Please wait while BCML upgrades...",
-                showProgress: true
-            },
-            () => {
-                pywebview.api
-                    .update_bcml()
-                    .then(res => {
-                        if (res.error) {
-                            throw res.error;
-                        }
-                        this.setState({
-                            showProgress: false,
-                            showConfirm: true,
-                            confirmText:
-                                "BCML has been updated successfully. You must restart the program for changes to take effect. Restart now?",
-                            confirmCallback: () => pywebview.api.restart()
-                        });
-                    })
-                    .catch(this.showError);
-            }
-        );
-    };
-
     setProgress = (title, msg) => {
         this.setState({
             progressTitle: title,
@@ -328,9 +300,6 @@ class App extends React.Component {
                         <Dropdown.Item onClick={() => pywebview.api.save_mod_list()}>
                             Save Mod List
                         </Dropdown.Item>
-                        <Dropdown.Item onClick={this.updateBcml}>
-                            Update BCML
-                        </Dropdown.Item>
                         <Dropdown.Item as="a" href="/index.html?firstrun">
                             Run Setup Wizard
                         </Dropdown.Item>
@@ -441,14 +410,6 @@ class App extends React.Component {
                     message={this.state.confirmText}
                     onClose={this.state.confirmCallback.bind(this)}
                 />
-                <UpdateDialog
-                    show={this.state.update}
-                    onClose={confirmed =>
-                        this.setState({ update: false }, () =>
-                            confirmed ? this.updateBcml() : null
-                        )
-                    }
-                />
                 <BackupModal
                     show={this.state.showBackups}
                     busy={this.state.showProgress}
@@ -569,23 +530,4 @@ const ConfirmDialog = props => {
     );
 };
 
-const UpdateDialog = props => {
-    return (
-        <Modal show={props.show}>
-            <Modal.Header>
-                <Modal.Title>Update Available</Modal.Title>
-            </Modal.Header>
-            <Modal.Body>
-                There is a new update available for BCML. Would you like to install it?
-            </Modal.Body>
-            <Modal.Footer>
-                <Button onClick={() => props.onClose(true)}>OK</Button>
-                <Button variant="secondary" onClick={() => props.onClose(false)}>
-                    Close
-                </Button>
-            </Modal.Footer>
-        </Modal>
-    );
-};
-
 export default App;
diff --git a/bcml/assets/src/js/Settings.jsx b/bcml/assets/src/js/Settings.jsx
index e442b9b..eccfbdf 100644
--- a/bcml/assets/src/js/Settings.jsx
+++ b/bcml/assets/src/js/Settings.jsx
@@ -22,7 +22,6 @@ class Settings extends React.Component {
             wiiu: true,
             no_hardlinks: false,
             force_7z: false,
-            suppress_update: false,
             valid: false,
             loaded: false
         };
@@ -460,24 +459,6 @@ class Settings extends React.Component {
                                 />
                             </OverlayTrigger>
                         </Form.Group>
-                        <Form.Group controlId="suppress_update">
-                            <OverlayTrigger
-                                overlay={
-                                    <Tooltip>
-                                        By default, BCML will notify you when it detects
-                                        an updated version is available. Check this to
-                                        turn that off.
-                                    </Tooltip>
-                                }
-                                placement={"left"}>
-                                <Form.Check
-                                    type="checkbox"
-                                    label="Disable BCML update notification"
-                                    checked={this.state.suppress_update}
-                                    onChange={this.handleChange}
-                                />
-                            </OverlayTrigger>
-                        </Form.Group>
                         {window.navigator.platform.includes("inux") && (
                             <Form.Group controlId="force_7z">
                                 <OverlayTrigger
diff --git a/bcml/util.py b/bcml/util.py
index d418530..30c7d6d 100644
--- a/bcml/util.py
+++ b/bcml/util.py
@@ -565,7 +565,6 @@ DEFAULT_SETTINGS = {
     "wiiu": True,
     "no_hardlinks": False,
     "force_7z": False,
-    "suppress_update": False,
 }
 
 
@@ -1255,25 +1254,6 @@ def get_open_port():
     return port
 
 
-@lru_cache(1)
-def get_latest_bcml() -> str:
-    try:
-        res = requests.get("https://pypi.org/rss/project/bcml/releases.xml")
-        doc = minidom.parseString(res.text)
-        versions = sorted(
-            (
-                item.getElementsByTagName("title")[0].childNodes[0].data
-                for item in doc.getElementsByTagName("item")
-            ),
-            reverse=True,
-        )
-        if DEBUG:
-            return versions[0]
-        return next(v for v in versions if "a" not in v and "b" not in v)
-    except:  # pylint: disable=bare-except
-        return "0.0.0"
-
-
 class RulesParser(ConfigParser):
     # pylint: disable=arguments-differ,too-many-ancestors
     def __init__(self):
