diff --git i/docs/man/ms2render.htm w/docs/man/ms2render.htm
index 0664e9f..6cdbfa0 100644
--- i/docs/man/ms2render.htm
+++ w/docs/man/ms2render.htm
@@ -493,7 +493,7 @@
 <dt id="t---types-arg"><b>-t, --types &lt;arg&gt;</b></dt>
 <dd>
 
-<p>File with type information. Default: /usr/share/mapsoft2/types.cfg</p>
+<p>File with type information. Default: @out@/share/mapsoft2/types.cfg</p>
 
 </dd>
 <dt id="q---quite-arg"><b>-q, --quite &lt;arg&gt;</b></dt>
@@ -553,7 +553,7 @@
 <dt id="c---config-arg"><b>-c, --config &lt;arg&gt;</b></dt>
 <dd>
 
-<p>Configuration file for vector map rendering. Default: /usr/share/mapsoft2/render.cfg</p>
+<p>Configuration file for vector map rendering. Default: @out@/share/mapsoft2/render.cfg</p>
 
 </dd>
 <dt id="obj_scale-arg"><b>--obj_scale &lt;arg&gt;</b></dt>
diff --git i/docs/man/ms2view.htm w/docs/man/ms2view.htm
index 243de0a..7138440 100644
--- i/docs/man/ms2view.htm
+++ w/docs/man/ms2view.htm
@@ -475,7 +475,7 @@
 
 <h1 id="CUSTOMIZING-VIEWER-INTERFACE-WITH-CSS-STYLES">CUSTOMIZING VIEWER INTERFACE WITH CSS STYLES</h1>
 
-<p>Structure of <b>ms2view</b> window is simple: a viewer, a main menu, a statusbar, tables with waypoints, tracks, and maps, a few dialogs, etc. Appearence of these elements is controlled by GTK3 theme. One can customize it using CSS styles. Styles are read from two places: <b>/usr/share/mapsoft2/mapsoft2.css</b> -- system-wide file. By default it contains a few definitions, mostly for making interface elements smaller. Some tips about viewer structure can be also find there. <b>$HOME/.mapsoft2/mapsoft2.css</b> -- user-specific file.</p>
+<p>Structure of <b>ms2view</b> window is simple: a viewer, a main menu, a statusbar, tables with waypoints, tracks, and maps, a few dialogs, etc. Appearence of these elements is controlled by GTK3 theme. One can customize it using CSS styles. Styles are read from two places: <b>@out@/share/mapsoft2/mapsoft2.css</b> -- system-wide file. By default it contains a few definitions, mostly for making interface elements smaller. Some tips about viewer structure can be also find there. <b>$HOME/.mapsoft2/mapsoft2.css</b> -- user-specific file.</p>
 
 <p>Additional information about writing css styles can be found in GTK3 documentation or GTK theme sources:</p>
 
diff --git i/docs/man/ms2view.txt w/docs/man/ms2view.txt
index 5469cac..c165c1a 100644
--- i/docs/man/ms2view.txt
+++ w/docs/man/ms2view.txt
@@ -4,7 +4,7 @@ Structure of B<ms2view> window is simple: a viewer, a main menu, a
 statusbar, tables with waypoints, tracks, and maps, a few dialogs, etc.
 Appearence of these elements is controlled by GTK3 theme. One can
 customize it using CSS styles. Styles are read from two places:
-B</usr/share/mapsoft2/mapsoft2.css> -- system-wide file. By default it
+B<@out@/share/mapsoft2/mapsoft2.css> -- system-wide file. By default it
 contains a few definitions, mostly for making interface elements smaller.
 Some tips about viewer structure can be also find there.
 B<$HOME/.mapsoft2/mapsoft2.css> -- user-specific file.
diff --git i/docs/man/ms2vmap.htm w/docs/man/ms2vmap.htm
index 69fbda3..1fdfd43 100644
--- i/docs/man/ms2vmap.htm
+++ w/docs/man/ms2vmap.htm
@@ -64,7 +64,7 @@ ms2vmap &lt;file&gt; ... -o &lt;out file&gt; [&lt;options&gt;]</code></pre>
 <dt id="t---types-arg"><b>-t, --types &lt;arg&gt;</b></dt>
 <dd>
 
-<p>File with type information. Default: /usr/share/mapsoft2/types.cfg</p>
+<p>File with type information. Default: @out@/share/mapsoft2/types.cfg</p>
 
 </dd>
 <dt id="D---define-arg"><b>-D, --define &lt;arg&gt;</b></dt>
diff --git i/docs/man/ms2vmapdb.htm w/docs/man/ms2vmapdb.htm
index 8bdb701..c05835a 100644
--- i/docs/man/ms2vmapdb.htm
+++ w/docs/man/ms2vmapdb.htm
@@ -230,7 +230,7 @@ ms2vmapdb export -- export vector map to a different format</code></pre>
 <dt id="t---types-arg"><b>-t, --types &lt;arg&gt;</b></dt>
 <dd>
 
-<p>File with type information. Default: /usr/share/mapsoft2/types.cfg</p>
+<p>File with type information. Default: @out@/share/mapsoft2/types.cfg</p>
 
 </dd>
 <dt id="D---define-arg"><b>-D, --define &lt;arg&gt;</b></dt>
@@ -294,7 +294,7 @@ ms2vmapdb export -- export vector map to a different format</code></pre>
 <dt id="t---types-arg1"><b>-t, --types &lt;arg&gt;</b></dt>
 <dd>
 
-<p>File with type information. Default: /usr/share/mapsoft2/types.cfg</p>
+<p>File with type information. Default: @out@/share/mapsoft2/types.cfg</p>
 
 </dd>
 <dt id="D---define-arg1"><b>-D, --define &lt;arg&gt;</b></dt>
@@ -418,7 +418,7 @@ ms2vmapdb export -- export vector map to a different format</code></pre>
 <dt id="t---types-arg2"><b>-t, --types &lt;arg&gt;</b></dt>
 <dd>
 
-<p>File with type information. Default: /usr/share/mapsoft2/types.cfg</p>
+<p>File with type information. Default: @out@/share/mapsoft2/types.cfg</p>
 
 </dd>
 <dt id="D---define-arg2"><b>-D, --define &lt;arg&gt;</b></dt>
Submodule modules contains modified content
diff --git i/modules/mapview/action_manager.cpp w/modules/mapview/action_manager.cpp
index ab16dae..272b454 100644
--- i/modules/mapview/action_manager.cpp
+++ w/modules/mapview/action_manager.cpp
@@ -75,7 +75,7 @@ ActionManager::ActionManager (Mapview * mapview_):
     ADD_ACT(AMTrkDel,        "Edit")
     ADD_ACT(AMEditData,      "Edit")
 
-    AddMaps("Maps", std::string("/usr/share/") + DATADIR + "/" + MAPS_MENU_FILE);
+    AddMaps("Maps", std::string("@out@/share/") + DATADIR + "/" + MAPS_MENU_FILE);
 
     if (getenv("HOME")) {
       AddSep("Maps");
diff --git i/modules/mapview/mapview.cpp w/modules/mapview/mapview.cpp
index 997c127..43928bf 100644
--- i/modules/mapview/mapview.cpp
+++ w/modules/mapview/mapview.cpp
@@ -367,7 +367,7 @@ Mapview::load_css(){
   if (!style_context) throw Err() << "Mapview: can't get Gtk::StyleContext";
 
   try{
-    std::string css_glo = std::string("/usr/share/") + DATADIR + "/" + CSS_FILE;
+    std::string css_glo = std::string("@out@/share/") + DATADIR + "/" + CSS_FILE;
     if (file_exists(css_glo) &&
       css_provider->load_from_path(css_glo)){
       auto screen = get_screen();
diff --git i/modules/vmap2/vmap2gobj.cpp w/modules/vmap2/vmap2gobj.cpp
index 6583699..5e6a6af 100644
--- i/modules/vmap2/vmap2gobj.cpp
+++ w/modules/vmap2/vmap2gobj.cpp
@@ -22,7 +22,7 @@ using namespace std;
 void
 ms2opt_add_vmap2_render(GetOptSet & opts){
   const char *g = "VMAP2_RENDER";
-  opts.add("config", 1,'c',g, "Configuration file for vector map rendering. Default: /usr/share/mapsoft2/render.cfg");
+  opts.add("config", 1,'c',g, "Configuration file for vector map rendering. Default: @out@/share/mapsoft2/render.cfg");
   opts.add("define",      1,0,g, "Definitions for vector map rendering (json object)");
   opts.add("obj_scale",   1,0,g, "Rescaling factor for all objects, default 1.0.");
   opts.add("vmap_minsc", 1,0,g, "Minimum map scale (calculated from the 'natural' "
@@ -110,7 +110,7 @@ GObjVMap2::GObjVMap2(VMap2 & map, const Opt &o): GObjMulti(false), map(map) {
   read_words_defs defs(o.get("define", Opt()));
 
   int depth = 0;
-  std::string cfg = opt.get<string>("config", "/usr/share/mapsoft2/render.cfg");
+  std::string cfg = opt.get<string>("config", "@out@/share/mapsoft2/render.cfg");
   load_conf(cfg, defs, depth);
 }
 
diff --git i/modules/vmap2/vmap2types.cpp w/modules/vmap2/vmap2types.cpp
index 3a7ee9b..c904529 100644
--- i/modules/vmap2/vmap2types.cpp
+++ w/modules/vmap2/vmap2types.cpp
@@ -8,12 +8,12 @@
 void
 ms2opt_add_vmap2t(GetOptSet & opts){
   const char *g = "VMAP2";
-  opts.add("types",  1, 't', g, "File with type information. Default: /usr/share/mapsoft2/types.cfg");
+  opts.add("types",  1, 't', g, "File with type information. Default: @out@/share/mapsoft2/types.cfg");
   opts.add("define", 1, 'D', g, "Define variables for type information file.");
 }
 
 VMap2types::VMap2types(const Opt & o){
-  std::string file = o.get("types", "/usr/share/mapsoft2/types.cfg");
+  std::string file = o.get("types", "@out@/share/mapsoft2/types.cfg");
   if (file == "") return;
   read_words_defs defs(o.get("define", Opt()));
   load(file, defs);
diff --git i/programs/Makefile w/programs/Makefile
index c81d62e..f7d5c7d 100644
--- i/programs/Makefile
+++ w/programs/Makefile
@@ -18,7 +18,7 @@ bindir  ?= $(prefix)/bin
 datadir ?= $(prefix)/share
 
 all:
-	sh -e -c 'for i in $(PROGRAMS); do make -C $$i; done'
+	sh -e -c 'for i in $(PROGRAMS); do make -C $$i notests; done'
 
 install:
 	sh -e -c 'for i in $(PROGRAMS); do install -D -m755 $$i/$$i $(bindir)/$$i; done'
diff --git i/vmap_data/scripts/vmaps.sh w/vmap_data/scripts/vmaps.sh
index 97ebf17..d0ca927 100644
--- i/vmap_data/scripts/vmaps.sh
+++ w/vmap_data/scripts/vmaps.sh
@@ -16,8 +16,8 @@ MAP_ATTR="mapsoft"
 VMAP_DIR=vmap;   # Source folder
 VMAP_EXT=vmap;   # Extension/format of vector maps (could be vmap2, vmap, or even mp)
 
-REND_CFG=/usr/share/mapsoft2/render.cfg;  # Render configuration
-TYPEINFO=/usr/share/mapsoft2/types.cfg    # Type information
+REND_CFG=@out@/share/mapsoft2/render.cfg;  # Render configuration
+TYPEINFO=@out@/share/mapsoft2/types.cfg    # Type information
 
 ############################################################
 ## Image rendering
@@ -46,7 +46,7 @@ TLIST="";
 ############################################################
 ## IMG rendering (see vmaps_img script)
 IMG_DIR=OUT;   # img files (if empty, img files are not rendered)
-TYPSRC=/usr/share/mapsoft2/typ.txt;  # typ file source for garmin IMG
+TYPSRC=@out@/share/mapsoft2/typ.txt;  # typ file source for garmin IMG
 GMT=gmt;                             # gmt program
 CGPSM=cgpsmapper-static              # cgpsmapper program
 IMGMAP="";                           # combined img file
diff --git i/vmap_data/scripts/vmaps_diff w/vmap_data/scripts/vmaps_diff
index 136155e..0aa1ed0 100755
--- i/vmap_data/scripts/vmaps_diff
+++ w/vmap_data/scripts/vmaps_diff
@@ -4,7 +4,7 @@
 # Render difference between two git revitions of a file
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_get_fig w/vmap_data/scripts/vmaps_get_fig
index 0ba7b7f..5048bcb 100755
--- i/vmap_data/scripts/vmaps_get_fig
+++ w/vmap_data/scripts/vmaps_get_fig
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_img w/vmap_data/scripts/vmaps_img
index 1e0599c..84413bb 100755
--- i/vmap_data/scripts/vmaps_img
+++ w/vmap_data/scripts/vmaps_img
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_in w/vmap_data/scripts/vmaps_in
index 48da7ff..70203bc 100755
--- i/vmap_data/scripts/vmaps_in
+++ w/vmap_data/scripts/vmaps_in
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_index w/vmap_data/scripts/vmaps_index
index ae225ec..e59c592 100755
--- i/vmap_data/scripts/vmaps_index
+++ w/vmap_data/scripts/vmaps_index
@@ -4,7 +4,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_out w/vmap_data/scripts/vmaps_out
index 8996a29..914546b 100755
--- i/vmap_data/scripts/vmaps_out
+++ w/vmap_data/scripts/vmaps_out
@@ -18,7 +18,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_pack_img w/vmap_data/scripts/vmaps_pack_img
index f8ebf4f..c101941 100755
--- i/vmap_data/scripts/vmaps_pack_img
+++ w/vmap_data/scripts/vmaps_pack_img
@@ -5,7 +5,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_pack_mbtiles w/vmap_data/scripts/vmaps_pack_mbtiles
index b73121e..da52332 100755
--- i/vmap_data/scripts/vmaps_pack_mbtiles
+++ w/vmap_data/scripts/vmaps_pack_mbtiles
@@ -7,7 +7,7 @@
 # see https://github.com/mapbox/mbtiles-spec/blob/master/1.3/spec.md
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_pack_sqlitedb w/vmap_data/scripts/vmaps_pack_sqlitedb
index 3dfd676..9b4887b 100755
--- i/vmap_data/scripts/vmaps_pack_sqlitedb
+++ w/vmap_data/scripts/vmaps_pack_sqlitedb
@@ -7,7 +7,7 @@
 # see: https://osmand.net/docs/technical/osmand-file-formats/osmand-sqlite/
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_png w/vmap_data/scripts/vmaps_png
index 5fbf71c..f726bdb 100755
--- i/vmap_data/scripts/vmaps_png
+++ w/vmap_data/scripts/vmaps_png
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_preview w/vmap_data/scripts/vmaps_preview
index 1ed94bc..be42a26 100755
--- i/vmap_data/scripts/vmaps_preview
+++ w/vmap_data/scripts/vmaps_preview
@@ -3,7 +3,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_rend_mbtiles w/vmap_data/scripts/vmaps_rend_mbtiles
index a8ffee0..4809371 100755
--- i/vmap_data/scripts/vmaps_rend_mbtiles
+++ w/vmap_data/scripts/vmaps_rend_mbtiles
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_tiles w/vmap_data/scripts/vmaps_tiles
index fc45eab..3a06f66 100755
--- i/vmap_data/scripts/vmaps_tiles
+++ w/vmap_data/scripts/vmaps_tiles
@@ -9,7 +9,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_tlist w/vmap_data/scripts/vmaps_tlist
index 6aff657..1b7df8c 100755
--- i/vmap_data/scripts/vmaps_tlist
+++ w/vmap_data/scripts/vmaps_tlist
@@ -3,7 +3,7 @@
 # Create list of tiles
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
diff --git i/vmap_data/scripts/vmaps_wp_update w/vmap_data/scripts/vmaps_wp_update
index 0c3ec89..e235642 100755
--- i/vmap_data/scripts/vmaps_wp_update
+++ w/vmap_data/scripts/vmaps_wp_update
@@ -26,7 +26,7 @@
 ##################################################
 
 # read global configuration and functions
-. vmaps.sh ||:
+. @out@/bin/vmaps.sh ||:
 
 # read local configuration
 . ./vmaps.conf ||:
