diff '--color=auto' -u -r open/OpenEXR.cpp jj/OpenEXR.cpp
--- open/OpenEXR.cpp	2022-08-26 18:17:29.000000000 +0800
+++ jj/OpenEXR.cpp	2023-07-05 09:33:48.715779275 +0800
@@ -33,41 +33,41 @@
   #define PyUTF8_AsSstring(x)   PyString_AsString(x)
 #endif
 
-#include <ImfIO.h>
-#include <Iex.h>
-#include <ImfArray.h>
-#include <ImfAttribute.h>
-#include <ImfBoxAttribute.h>
-#include <ImfChannelList.h>
-#include <ImfStandardAttributes.h>
-#include <ImfChannelListAttribute.h>
-#include <ImfChromaticitiesAttribute.h>
-#include <ImfCompressionAttribute.h>
-#include <ImfDoubleAttribute.h>
-#include <ImfEnvmapAttribute.h>
-#include <ImfFloatAttribute.h>
-#include <ImfFrameBuffer.h>
-#include <ImfHeader.h>
-#include <ImfInputFile.h>
-#include <ImfIntAttribute.h>
-#include <ImfKeyCodeAttribute.h>
-#include <ImfLineOrderAttribute.h>
-#include <ImfMatrixAttribute.h>
-#include <ImfOutputFile.h>
-#include <ImfPreviewImageAttribute.h>
-#include <ImfStringAttribute.h>
-#include <ImfTileDescriptionAttribute.h>
-#include <ImfTiledOutputFile.h>
-#include <ImfTimeCodeAttribute.h>
-#include <ImfVecAttribute.h>
-#include <ImfVersion.h>
+#include <OpenEXR/ImfIO.h>
+#include <OpenEXR/Iex.h>
+#include <OpenEXR/ImfArray.h>
+#include <OpenEXR/ImfAttribute.h>
+#include <OpenEXR/ImfBoxAttribute.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfStandardAttributes.h>
+#include <OpenEXR/ImfChannelListAttribute.h>
+#include <OpenEXR/ImfChromaticitiesAttribute.h>
+#include <OpenEXR/ImfCompressionAttribute.h>
+#include <OpenEXR/ImfDoubleAttribute.h>
+#include <OpenEXR/ImfEnvmapAttribute.h>
+#include <OpenEXR/ImfFloatAttribute.h>
+#include <OpenEXR/ImfFrameBuffer.h>
+#include <OpenEXR/ImfHeader.h>
+#include <OpenEXR/ImfInputFile.h>
+#include <OpenEXR/ImfIntAttribute.h>
+#include <OpenEXR/ImfKeyCodeAttribute.h>
+#include <OpenEXR/ImfLineOrderAttribute.h>
+#include <OpenEXR/ImfMatrixAttribute.h>
+#include <OpenEXR/ImfOutputFile.h>
+#include <OpenEXR/ImfPreviewImageAttribute.h>
+#include <OpenEXR/ImfStringAttribute.h>
+#include <OpenEXR/ImfTileDescriptionAttribute.h>
+#include <OpenEXR/ImfTiledOutputFile.h>
+#include <OpenEXR/ImfTimeCodeAttribute.h>
+#include <OpenEXR/ImfVecAttribute.h>
+#include <OpenEXR/ImfVersion.h>
 
-#include <ImfRationalAttribute.h>
-#include <ImfRational.h>
-#include <ImfKeyCodeAttribute.h>
-#include <ImfKeyCode.h>
-#include <ImfTimeCodeAttribute.h>
-#include <ImfTimeCode.h>
+#include <OpenEXR/ImfRationalAttribute.h>
+#include <OpenEXR/ImfRational.h>
+#include <OpenEXR/ImfKeyCodeAttribute.h>
+#include <OpenEXR/ImfKeyCode.h>
+#include <OpenEXR/ImfTimeCodeAttribute.h>
+#include <OpenEXR/ImfTimeCode.h>
 
 #include <string>
 #include <fstream>
diff '--color=auto' -u -r open/setup.py jj/setup.py
--- open/setup.py	2022-08-26 18:18:17.000000000 +0800
+++ jj/setup.py	2023-07-05 09:33:59.745767766 +0800
@@ -44,15 +44,6 @@
   ext_modules=[ 
     Extension('OpenEXR',
               ['OpenEXR.cpp'],
-              include_dirs=['/usr/include/OpenEXR',
-                            '/usr/local/include/OpenEXR',
-                            '/opt/local/include/OpenEXR',
-                            '/usr/include/Imath',
-                            '/usr/local/include/Imath',
-                            '/opt/local/include/Imath'],
-              library_dirs=['/usr/lib',
-                            '/usr/local/lib',
-                            '/opt/local/lib'],
               libraries=libraries,
               extra_compile_args=extra_compile_args)
   ],
