diff -ur a/install.sh b/install.sh
--- a/install.sh	2020-07-23 12:14:47.339088042 +0200
+++ b/install.sh	2020-07-23 12:30:46.374428540 +0200
@@ -25,8 +25,6 @@
 cp -r ./* /tmp/assistinstall
 cd /tmp/assistinstall || die "failed to create tmp file"
 
-git pull
-
 [ -e "${ASSISTPREFIX}"/usr/bin ] || mkdir -p "${ASSISTPREFIX}"/usr/bin
 [ -e "${ASSISTPREFIX}"/usr/share/instantassist ] || mkdir -p "${ASSISTPREFIX}"/usr/share/instantassist
 
@@ -42,17 +40,7 @@
 chmod -R 755 cache
 
 if ! [ -e "${ASSISTPREFIX}"/usr/share/instantassist/spotify-adblock.so ]; then
-    pushd .
-    mkdir /tmp/spotblock
-    cd /tmp/spotblock || die "no spotblock"
-    git clone --depth=1 "https://github.com/abba23/spotify-adblock-linux.git"
-    cd spotify-adblock-linux || die "can't clone spotfiy"
-    sed -i '/\};/i    "audio4-fa.spotifycdn.com", //audio' whitelist.h
-    make
-    mv spotify-adblock.so "${ASSISTPREFIX}"/usr/share/instantassist/ || die "error moving stuff"
-    cd ..
-    rm -rf spotify-adblock-linux
-    popd
+	ln -s path/to/spotify-adblock.so "${ASSISTPREFIX}/share/instantassist"
 fi
 
 echo "installing instantassist binary"
@@ -86,4 +74,4 @@
 
 [ -z "${ASSISTPREFIX}" ] && rm -rf /tmp/assistinstall
 
-popd
\ Pas de fin de ligne à la fin du fichier
+popd
