--- a/lib/extension/window.js
+++ b/lib/extension/window.js
@@ -1350,2 +1350,14 @@
       }
+    } else if (this.focusMetaWindow && !this.tree.findNode(metaWindow)) {
+      let focused = this.tree.findNode(this.focusMetaWindow);
+      if (focused) {
+        let container = this.tree.findAncestor(focused, NODE_TYPES.CON) || focused.parentNode;
+        if (container && container.layout !== LAYOUT_TYPES.TABBED) {
+          let level = container.level;
+          if (container.type === NODE_TYPES.CON) level--;
+          if ((level == 2 && container.childNodes.length >= 2) || level >= 3) {
+            this.command({ name: "LayoutTabbedToggle" });
+          }
+        }
+      }
     }
