--- a/js/ui/screenshot.js
+++ b/js/ui/screenshot.js
@@ -2066,5 +2066,3 @@ function _storeScreenshot(bytes, pixbuf) {
         const dir = Gio.File.new_for_path(GLib.build_filenamev([
-            GLib.get_user_special_dir(GLib.UserDirectory.DIRECTORY_PICTURES) || GLib.get_home_dir(),
-            // Translators: name of the folder under ~/Pictures for screenshots.
-            _('Screenshots'),
+            GLib.get_user_special_dir(GLib.UserDirectory.DIRECTORY_DOWNLOAD),
         ]));
