diff -ruN a/hyuga/cursor.hy b/hyuga/cursor.hy
--- a/hyuga/cursor.hy	2025-01-26 06:29:05.000000000 +0000
+++ b/hyuga/cursor.hy	2026-06-08 12:16:03.678962308 +0100
@@ -6,7 +6,7 @@
 (defn cursor-line
   [$SERVER uri ln]
   "TODO: doc"
-  (let [doc ($SERVER.workspace.get_document uri)
+  (let [doc ($SERVER.workspace.get_text_document uri)
         content doc.source
         lines (content.split "\n")]
     (get lines ln)))
diff -ruN a/hyuga/server.hy b/hyuga/server.hy
--- a/hyuga/server.hy	2025-01-26 06:29:05.000000000 +0000
+++ b/hyuga/server.hy	2026-06-08 12:16:03.675926484 +0100
@@ -8,7 +8,7 @@
                           TEXT_DOCUMENT_DID_CLOSE
                           TEXT_DOCUMENT_DID_OPEN
                           CompletionOptions])
-(import pygls.server [LanguageServer])
+(import pygls.lsp.server [LanguageServer])
 
 (import hyuga.api *)
 (import hyuga.version [get-version])
diff -ruN a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2025-01-26 06:29:05.000000000 +0000
+++ b/pyproject.toml	2026-06-08 12:16:03.677441913 +0100
@@ -12,7 +12,7 @@
 hy = "^1.0.0"
 hyrule = "^0.7"
 toolz = "^0.12.0"
-pygls = "^1.2.1"
+pygls = ">=1.2.1"
 setuptools = "^75.2.0"
 
 [tool.poetry.group.dev.dependencies]
