From 51bce23546c46626b5e829cffb1f91c3de1c068a Mon Sep 17 00:00:00 2001
From: Zhenbo Li <lizhenbo@yahoo.com>
Date: Tue, 12 Dec 2023 14:29:48 -0500
Subject: [PATCH] Fix git URL scheme

---
 Package.swift | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Package.swift b/Package.swift
index 0042360..27f7bf4 100644
--- a/Package.swift
+++ b/Package.swift
@@ -18,7 +18,7 @@ let package = Package(
         .package(url: "https://github.com/onevcat/Rainbow.git", from: "3.0.0"),
         .package(url: "https://github.com/tuist/XcodeProj.git", from: "8.13.0"),
         .package(url: "https://github.com/jakeheis/SwiftCLI.git", from: "6.0.3"),
-        .package(url: "https://github.com/mxcl/Version", from: "2.0.0"),
+        .package(url: "https://github.com/mxcl/Version.git", from: "2.0.0"),
         .package(url: "https://github.com/SwiftDocOrg/GraphViz.git", exact: "0.2.0"),
     ],
 
-- 
2.42.0

