diff --git a/mate-session/main.c b/mate-session/main.c
index 96c1092..32a696c 100644
--- a/mate-session/main.c
+++ b/mate-session/main.c
@@ -634,6 +634,7 @@ int main(int argc, char** argv)
 	sigaction(SIGPIPE, &sa, 0);
 
 	error = NULL;
+	gdk_set_allowed_backends("x11");
 	gtk_init_with_args(&argc, &argv, (char*) _(" - the MATE session manager"), entries, GETTEXT_PACKAGE, &error);
 
 	if (error != NULL)
diff --git a/tools/mate-session-check-accelerated-gles-helper.c b/tools/mate-session-check-accelerated-gles-helper.c
index e6312c6..50eb422 100644
--- a/tools/mate-session-check-accelerated-gles-helper.c
+++ b/tools/mate-session-check-accelerated-gles-helper.c
@@ -217,6 +217,7 @@ main (int argc,
         }
 
 #ifdef GDK_WINDOWING_X11
+        gdk_set_allowed_backends ("x11");
         char *renderer = get_gles_renderer ();
         if (renderer != NULL) {
                 if (print_renderer)
diff --git a/tools/mate-session-check-accelerated.c b/tools/mate-session-check-accelerated.c
index 6fe6ec4..b1c2374 100644
--- a/tools/mate-session-check-accelerated.c
+++ b/tools/mate-session-check-accelerated.c
@@ -146,6 +146,7 @@ main (int argc, char **argv)
         glong is_accelerated, is_software_rendering;
         GError *gl_error = NULL;
 
+        gdk_set_allowed_backends ("x11");
         gtk_init (NULL, NULL);
 
         /* mate-session-check-accelerated gets run before X is started in the wayland
