diff --git i/nob.c w/nob.c
index 37ebbe1..f4b22a9 100644
--- i/nob.c
+++ w/nob.c
@@ -129,17 +129,6 @@ int main(int argc, char **argv)
     if (!build_plug_c(force, &cmd, PLUGS_DIR"squares/plug.c", BUILD_DIR"libsquare.so")) return 1;
     if (!build_plug_c(force, &cmd, PLUGS_DIR"bezier/plug.c", BUILD_DIR"libbezier.so")) return 1;
     if (!build_plug_cxx(force, &cmd, PLUGS_DIR"cpp/plug.cpp", BUILD_DIR"libcpp.so")) return 1;
-    {
-        const char *output_path = BUILD_DIR"libc3";
-        const char *source_paths[] = {
-            PLUGS_DIR"c3/plug.c3",
-            PLUGS_DIR"c3/raylib.c3i",
-            PLUGS_DIR"c3/future.c3"
-        };
-        size_t source_paths_count = NOB_ARRAY_LEN(source_paths);
-
-        if (!build_plug_c3(force, &cmd, output_path, source_paths, source_paths_count)) return 1;
-    }
 
     {
         const char *output_path = BUILD_DIR"panim";
