diff --git a/contrib/imager/lib/util_resample.f90 b/contrib/imager/lib/util_resample.f90
index 7ec45dd..3497ca6 100644
--- a/contrib/imager/lib/util_resample.f90
+++ b/contrib/imager/lib/util_resample.f90
@@ -57,11 +57,7 @@ subroutine resample_uv (uvint,uvout,uvint_data,uvout_data,nv,nt,filtered)
   ! Local
   integer(kind=index_length) :: iv
   real(4), allocatable :: uvtmp(:)
-  real(4), allocatable :: wmin(:)
-  real(4), allocatable :: wmax(:)
-  integer(4), allocatable :: imin(:)
-  integer(4), allocatable :: imax(:)
-  integer :: nin, nou, k, kk, nch, ier
+  integer :: nin, nou, k, kk
   real(8) :: xref,xval,xinc
   real(8) :: yref,yval,yinc
   logical :: ok
@@ -75,12 +71,6 @@ subroutine resample_uv (uvint,uvout,uvint_data,uvout_data,nv,nt,filtered)
   !
   nou = uvout%gil%dim(1)
   nin = uvint%gil%dim(1)
-  nch = uvint%gil%nchan
-  !
-  allocate (imin(nch),imax(nch),wmin(nch),wmax(nch),stat=ier)
-  !
-  call interpolate_axis (uvout%gil%nchan, xinc,xref,xval, &
-      & uvint%gil%nchan,yinc,yref,yval, imin,imax,wmin,wmax)
   !
   ok = .false. !TEST! present(filtered)
   if (ok) then
