commit 1e4f743663ace0ca2452e02922109027b8c1bd23
diff --git a/src/address/address.go b/src/address/address.go
index 5c13257..560df59 100644
--- a/src/address/address.go
+++ b/src/address/address.go
@@ -13,7 +13,7 @@ type Subnet [8]byte
 // The 8th bit of the last byte is used to signal nodes (0) or /64 prefixes (1).
 // Nodes that configure this differently will be unable to communicate with eachother, though routing and the DHT machinery *should* still work.
 func GetPrefix() [1]byte {
-	return [...]byte{0x02}
+	return [...]byte{0x04}
 }
 
 // isValid returns true if an address falls within the range used by nodes in the network.
diff --git a/src/tuntap/icmpv6.go b/src/tuntap/icmpv6.go
index 52ca50c..f1ddd3e 100644
--- a/src/tuntap/icmpv6.go
+++ b/src/tuntap/icmpv6.go
@@ -315,7 +315,7 @@ func (i *icmpv6) handle_ndp(in []byte) ([]byte, error) {
 	case source.IsValid():
 	case snet.IsValid():
 	default:
-		return nil, errors.New("Not an NDP for 0200::/7")
+		return nil, errors.New("Not an NDP for 0400::/7")
 	}
 
 	// Create our NDP message body response
