--- ./share/extensions/dxf_outlines.py	2017-10-08 17:28:45.553368917 +0200
+++ ./share/extensions/dxf_outlines.py.new	2017-10-08 17:29:20.172554152 +0200
@@ -341,7 +341,7 @@
         if not scale:
             scale = 25.4/96     # if no scale is specified, assume inch as baseunit
         scale /= self.unittouu('1px')
-        h = self.unittouu(self.document.getroot().xpath('@height', namespaces=inkex.NSS)[0])
+        h = self.unittouu(self.documentHeight())
         self.groupmat = [[[scale, 0.0, 0.0], [0.0, -scale, h*scale]]]
         doc = self.document.getroot()
         self.process_group(doc)

