diff --git a/telegram_send/telegram_send.py b/telegram_send/telegram_send.py
index 6b34a0c..1d9a661 100644
--- a/telegram_send/telegram_send.py
+++ b/telegram_send/telegram_send.py
@@ -119,7 +119,7 @@ def main():
         args.message = [message]
 
     try:
-        delete(args.delete)
+        delete(args.delete, conf=conf[0])
         message_ids = []
         for c in conf:
             message_ids += send(
