Apply viridis color palette

--- a/app/javascript/controllers/maps_controller.js
+++ b/app/javascript/controllers/maps_controller.js
@@ -224 +224,14 @@
-    this.heatmapLayer = L.heatLayer(this.heatmapMarkers, { radius: 20 }).addTo(
+    const gradient = {
+      0.50: '#440154', // minOpacity (misnamed; see mourner/simpleheat#31)
+      0.55: '#482475',
+      0.60: '#414487',
+      0.65: '#355f8d',
+      0.70: '#2a788e',
+      0.75: '#21918c',
+      0.80: '#22a884',
+      0.85: '#44bf70',
+      0.90: '#7ad151',
+      0.95: '#bddf26',
+      1.00: '#fde725',
+    };
+    this.heatmapLayer = L.heatLayer(this.heatmapMarkers, { radius: 20, gradient }).addTo(
