--- a/settings.gradle.kts	2026-09-24 13:42:01
+++ b/settings.gradle.kts	2026-09-24 13:44:00
@@ -1,77 +1,19 @@
-@file:Suppress("UnstableApiUsage")
 
 pluginManagement {
     repositories {
-        google {
-            content {
-                includeGroupAndSubgroups("com.android")
-                includeGroupAndSubgroups("com.google")
-                includeGroupAndSubgroups("androidx")
-            }
-        }
-        mavenCentral()
         gradlePluginPortal()
+        google()
+        mavenCentral()
     }
 }
-
 dependencyResolutionManagement {
-    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
-
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
     repositories {
         google()
-        mavenCentral {
-            mavenContent {
-                releasesOnly()
-            }
-        }
-        exclusiveContent {
-            forRepository {
-                maven {
-                    name = "JitPack"
-                    setUrl("https://jitpack.io")
-                }
-            }
-            filter {
-                includeGroup("com.github.therealbush")
-                includeGroup("com.github.TeamNewPipe")
-            }
-        }
+        mavenCentral()
+        maven("https://jitpack.io")
     }
 }
-
-// F-Droid doesn't support foojay-resolver plugin
-// plugins {
-//     id("org.gradle.toolchains.foojay-resolver-convention") version("1.0.0")
-// }
-
-rootProject.name = "ArchiveTune"
-include(":app")
 include(":core")
-include(":lyrics:kugou")
-include(":lyrics:lrclib")
-include(":lyrics:simpmusic")
-include(":lyrics:paxsenix")
-include(":lyrics:betterlyrics")
-include(":lyrics:unison")
-include(":lyrics:youlyplus")
-include(":lastfm")
-include(":canvas")
-include(":shazamkit")
-include(":spotifycore")
-include(":morideobfuscator")
-
-// Use a local copy of NewPipe Extractor by uncommenting the lines below.
-// We assume, that ArchiveTune and NewPipe Extractor have the same parent directory.
-// If this is not the case, please change the path in includeBuild().
-//
-// For this to work you also need to change the implementation in core/build.gradle.kts
-// to one which does not specify a version.
-// From:
-//      implementation(libs.newpipe.extractor)
-// To:
-//      implementation("com.github.TeamNewPipe:NewPipeExtractor")
-// includeBuild("../NewPipeExtractor") {
-//    dependencySubstitution {
-//        substitute(module("com.github.TeamNewPipe:NewPipeExtractor")).using(project(":extractor"))
-//    }
-// }
+include(":app")
+include(":android-stubs")
