diff -ur a/config.def.h b/config.def.h
--- a/config.def.h	2020-05-27 12:07:30.000000000 +0200
+++ b/config.def.h	2020-07-03 15:11:09.762736457 +0200
@@ -121,7 +121,7 @@
 static const char *instantmenucmd[] = {"instantmenu_run", NULL};
 static const char *roficmd[] = {"rofi", "-show", "run", NULL};
 static const char *instantmenustcmd[] = {"instantmenu_run_st", NULL};
-static const char *termcmd[] = {"st", NULL};
+static const char *termcmd[] = {"konsole", NULL};
 static const char *quickmenucmd[] = {"quickmenu", NULL};
 static const char *instantassistcmd[] = {"instantassist", NULL};
 static const char *nautiluscmd[] = {"nautilus", NULL};
@@ -132,7 +132,7 @@
 static const char *instantswitchcmd[] = {"rofi", "-show", "window", "-kb-row-down", "Alt+Tab,Down", "-kb-row-up", "Alt+Ctrl+Tab,Up", "-kb-accept-entry", "!Alt_L,!Alt+Tab,Return", "-me-select-entry", "", "-me-accept-entry", "MousePrimary", NULL};
 static const char *caretinstantswitchcmd[] = {"rofi", "-show", "window", "-kb-row-down", "Alt+Tab,Down", "-kb-row-up", "Alt+Ctrl+Tab,Up", "-kb-accept-entry", "!Alt_L,!Alt+Tab,Return", "-me-select-entry", "", "-me-accept-entry", "MousePrimary", "-theme", "/usr/share/instantdotfiles/rofi/appmenu.rasi", NULL};
 static const char *onboardcmd[] = {"onboard", NULL};
-static const char *instantshutdowncmd[] = {"instantshutdown", NULL};
+static const char *instantshutdowncmd[] = {"killall", "instantwm", NULL};
 static const char *notifycmd[] = {"instantnotify", NULL};
 static const char *rangercmd[] = { "urxvt", "-e", "ranger", NULL };
 static const char *panther[] = { "appmenu", NULL};
