--- a/make/bsd/bsd.cc
+++ b/make/bsd/bsd.cc
@@ -50,13 +50,13 @@
 extern SIG_PF
 bsdsignal (int Signal, SIG_PF Handler)
 {
-  auto SIG_PF                   previous_handler;
+  SIG_PF                   previous_handler;
 #ifdef SUN5_0
 #ifdef sun
   previous_handler = sigset (Signal, Handler);
 #else
-  auto struct sigaction         new_action;
-  auto struct sigaction         old_action;
+  struct sigaction         new_action;
+  struct sigaction         old_action;
 
   new_action.sa_flags = SA_SIGINFO;
   new_action.sa_handler = (void (*) ()) Handler;
