--- a/resources/OgreCore/StdQuad_vp.glsl
+++ b/resources/OgreCore/StdQuad_vp.glsl
@@ -1,11 +1,12 @@
+uniform mat4 worldViewProj;
+
 varying vec2 uv;
 
-void main()                    
+void main()
 {
-	gl_Position = ftransform();
-	
-	vec2 inPos = sign(gl_Vertex.xy);
-	
-	uv = (vec2(inPos.x, -inPos.y) + 1.0)/2.0;
-}
+    gl_Position = worldViewProj * gl_Vertex;
 
+    vec2 inPos = sign(gl_Vertex.xy);
+
+    uv = (vec2(inPos.x, -inPos.y) + 1.0) / 2.0;
+}
--- a/resources/OgreCore/StdQuad_vp.program
+++ b/resources/OgreCore/StdQuad_vp.program
@@ -61,6 +61,46 @@
 vertex_program Ogre/Compositor/StdQuad_GLSL_vp glsl
 {
 	source StdQuad_vp.glsl
+	default_params
+	{
+		param_named_auto worldViewProj worldviewproj_matrix
+	}
+}
+
+vertex_program Ogre/Compositor/StdQuad_Tex2_GLSL_vp glsl
+{
+	source StdQuad_Tex2_vp.glsl
+	default_params
+	{
+		param_named_auto worldViewProj worldviewproj_matrix
+	}
+}
+
+vertex_program Ogre/Compositor/StdQuad_Tex2a_GLSL_vp glsl
+{
+	source StdQuad_Tex2a_vp.glsl
+	default_params
+	{
+		param_named_auto worldViewProj worldviewproj_matrix
+	}
+}
+
+vertex_program Ogre/Compositor/StdQuad_Tex3_GLSL_vp glsl
+{
+	source StdQuad_Tex3_vp.glsl
+	default_params
+	{
+		param_named_auto worldViewProj worldviewproj_matrix
+	}
+}
+
+vertex_program Ogre/Compositor/StdQuad_Tex4_GLSL_vp glsl
+{
+	source StdQuad_Tex4_vp.glsl
+	default_params
+	{
+		param_named_auto worldViewProj worldviewproj_matrix
+	}
 }
 
 vertex_program Ogre/Compositor/StdQuad_GLSLES_vp glsles
@@ -125,8 +165,8 @@
 // Unified definitions
 vertex_program Ogre/Compositor/StdQuad_vp unified
 {
+	delegate Ogre/Compositor/StdQuad_GLSL_vp
 	delegate Ogre/Compositor/StdQuad_GLSLES_vp
-	delegate Ogre/Compositor/StdQuad_Cg_vp
 }
 
 vertex_program Ogre/Compositor/StdQuad_NoCG_vp unified
@@ -138,25 +178,25 @@
 
 vertex_program Ogre/Compositor/StdQuad_Tex2_vp unified
 {
+	delegate Ogre/Compositor/StdQuad_Tex2_GLSL_vp
 	delegate Ogre/Compositor/StdQuad_Tex2_GLSLES_vp
-	delegate Ogre/Compositor/StdQuad_Tex2_Cg_vp
 }
 
 vertex_program Ogre/Compositor/StdQuad_Tex2a_vp unified
 {
+	delegate Ogre/Compositor/StdQuad_Tex2a_GLSL_vp
 	delegate Ogre/Compositor/StdQuad_Tex2a_GLSLES_vp
-	delegate Ogre/Compositor/StdQuad_Tex2a_Cg_vp
 }
 
 vertex_program Ogre/Compositor/StdQuad_Tex3_vp unified
 {
+	delegate Ogre/Compositor/StdQuad_Tex3_GLSL_vp
 	delegate Ogre/Compositor/StdQuad_Tex3_GLSLES_vp
-    delegate Ogre/Compositor/StdQuad_Tex3_Cg_vp
 }
 
 vertex_program Ogre/Compositor/StdQuad_Tex4_vp unified
 {
+	delegate Ogre/Compositor/StdQuad_Tex4_GLSL_vp
 	delegate Ogre/Compositor/StdQuad_Tex4_GLSLES_vp
-	delegate Ogre/Compositor/StdQuad_Tex4_Cg_vp
 }
 
--- a/resources/OgreCore/StdQuad_Tex2_vp.glsl
+++ b/resources/OgreCore/StdQuad_Tex2_vp.glsl
@@ -0,0 +1,12 @@
+uniform mat4 worldViewProj;
+
+void main()
+{
+    gl_Position = worldViewProj * gl_Vertex;
+
+    vec2 inPos = sign(gl_Vertex.xy);
+
+    vec2 uv0 = (vec2(inPos.x, -inPos.y) + 1.0) * 0.5;
+    gl_TexCoord[0].xy = uv0;
+    gl_TexCoord[1].xy = uv0;
+}
--- a/resources/OgreCore/StdQuad_Tex2a_vp.glsl
+++ b/resources/OgreCore/StdQuad_Tex2a_vp.glsl
@@ -0,0 +1,12 @@
+uniform mat4 worldViewProj;
+
+void main()
+{
+    gl_Position = worldViewProj * gl_Vertex;
+
+    vec2 inPos = sign(gl_Vertex.xy);
+
+    vec2 uv0 = (vec2(inPos.x, -inPos.y) + 1.0) * 0.5;
+    gl_TexCoord[0].xy = uv0;
+    gl_TexCoord[1].xy = inPos;
+}
--- a/resources/OgreCore/StdQuad_Tex3_vp.glsl
+++ b/resources/OgreCore/StdQuad_Tex3_vp.glsl
@@ -0,0 +1,13 @@
+uniform mat4 worldViewProj;
+
+void main()
+{
+    gl_Position = worldViewProj * gl_Vertex;
+
+    vec2 inPos = sign(gl_Vertex.xy);
+
+    vec2 uv0 = (vec2(inPos.x, -inPos.y) + 1.0) * 0.5;
+    gl_TexCoord[0].xy = uv0;
+    gl_TexCoord[1].xy = uv0;
+    gl_TexCoord[2].xy = uv0;
+}
--- a/resources/OgreCore/StdQuad_Tex4_vp.glsl
+++ b/resources/OgreCore/StdQuad_Tex4_vp.glsl
@@ -0,0 +1,14 @@
+uniform mat4 worldViewProj;
+
+void main()
+{
+    gl_Position = worldViewProj * gl_Vertex;
+
+    vec2 inPos = sign(gl_Vertex.xy);
+
+    vec2 uv0 = (vec2(inPos.x, -inPos.y) + 1.0) * 0.5;
+    gl_TexCoord[0].xy = uv0;
+    gl_TexCoord[1].xy = uv0;
+    gl_TexCoord[2].xy = uv0;
+    gl_TexCoord[3].xy = uv0;
+}
