--- a/source/main/gui/GUIUtils.cpp
+++ b/source/main/gui/GUIUtils.cpp
@@ -541,7 +541,7 @@
 
         ImGui::BeginTooltip();
         std::string text = _L("Hold to confirm");
-        ImGui::TextDisabled(text.c_str());
+        ImGui::TextDisabled("%s", text.c_str());
         ImGui::ProgressBar(active_time_left/time_limit, ImVec2(ImGui::CalcTextSize(text.c_str()).x, 8.f), "");
         ImGui::EndTooltip();
     }
@@ -596,7 +596,7 @@
     const ImVec4 LINKCOLOR = ImVec4(0.3, 0.5, 0.9, 1.0);
     ImGui::PushStyleColor(0, LINKCOLOR);  //Text
     ImVec2 cursorBefore = ImGui::GetCursorScreenPos();
-    ImGui::Text(caption.c_str());
+    ImGui::Text("%s", caption.c_str());
     ImVec2 textSize = ImGui::CalcTextSize(caption.c_str());
     ImGui::GetWindowDrawList()->AddLine(
         cursorBefore + ImVec2(0, textSize.y), cursorBefore + textSize,  //from-to
