diff --git a/KikoPlay.pro b/KikoPlay.pro
index e3f57f5..813e42a 100644
--- a/KikoPlay.pro	2026-07-22 23:06:43.475285607 -0700
+++ b/KikoPlay.pro	2026-07-22 23:08:39.545720164 -0700
@@ -809,14 +809,14 @@
 # Linux install layout (not applicable to the macOS .app bundle)
 unix:!macx {
     # Install settings
-    target.path += /usr/bin
-    desktop.path = /usr/share/applications
+    target.path += OUTPATH/bin
+    desktop.path = OUTPATH/share/applications
     desktop.files = io.github.KikoPlayProject.KikoPlay.desktop
-    icons.path = /usr/share/icons/hicolor/128x128/apps
+    icons.path = OUTPATH/share/icons/hicolor/128x128/apps
     icons.files = io.github.KikoPlayProject.KikoPlay.png
-    metainfo.path = /usr/share/metainfo
+    metainfo.path = OUTPATH/share/metainfo
     metainfo.files = io.github.KikoPlayProject.KikoPlay.metainfo.xml
-    web.path = /usr/share/kikoplay/web
+    web.path = OUTPATH/share/kikoplay/web
     web.files = web/*
 
     INSTALLS += target desktop icons metainfo web
