=== modified file 'src/gui/editorwidget/debuggers/inodebugger.cpp'
--- old/src/gui/editorwidget/debuggers/inodebugger.cpp	2023-02-02 00:09:34 +0000
+++ new/src/gui/editorwidget/debuggers/inodebugger.cpp	2023-06-03 10:54:42 +0000
@@ -39,7 +39,7 @@
     m_version = 0;
     m_Ardboard = Uno;
     m_ArdboardList << "uno" << "megaADK" << "nano" << "diecimila" << "leonardo" << "custom";
-    m_buildPath = MainWindow::self()->getFilePath("data/codeeditor/buildIno");
+    m_buildPath = MainWindow::self()->getConfigPath("data/codeeditor/buildIno");
 }
 InoDebugger::~InoDebugger() {}

