diff --git i/lib/attribution.py w/lib/attribution.py
index 1422bb6..1e07064 100644
--- i/lib/attribution.py
+++ w/lib/attribution.py
@@ -1,8 +1,11 @@
 # coding: utf-8
 from PIL import Image
 import cairo
-import pango
-import pangocairo
+import gi
+gi.require_version('Pango', '1.0')
+gi.require_version('PangoCairo', '1.0')
+from gi.repository import Pango as pango
+from gi.repository import PangoCairo as pangocairo
 import math
 
 
@@ -27,7 +30,7 @@ def check_font_installed():
 
 
 def create_path(ctx, text, font_size_px, rotate):
-    check_font_installed()
+    #check_font_installed()
     ctx.set_line_width(font_size_px / 5.)
     ctx.set_line_join(cairo.LINE_JOIN_ROUND)
     ctx.save()
