diff --git i/tiles_update.py w/tiles_update.py
index 2d73b9a..88be9da 100755
--- i/tiles_update.py
+++ w/tiles_update.py
@@ -124,10 +124,10 @@ class JobManager(object):
         fingerprint = maprecord.fingerprint
         attrib_filename = attribution.get_attrib_path(maprecord.image_path)
         if os.path.exists(attrib_filename):
-            fingerprint = hashlib.sha1(fingerprint)
-            fingerprint.update(':~:' + open(attrib_filename).read())
+            fingerprint = hashlib.sha1(str(fingerprint).encode('utf-8'))
+            fingerprint.update(str(':~:' + open(attrib_filename).read()).encode('utf-8'))
             info_filename = attribution.get_info_path(maprecord.image_path)
-            fingerprint.update(':~:' + open(info_filename).read())
+            fingerprint.update(str(':~:' + open(info_filename).read()).encode('utf-8'))
             fingerprint = fingerprint.hexdigest()
         return fingerprint
 
@@ -225,8 +225,8 @@ def get_reprojected_image(tile_x, tile_y, level, map_reference):
         src_has_alpha = True
     cell_size = 64
     mesh = []
-    for cell_x in xrange(tile_size / cell_size):
-        for cell_y in xrange(tile_size / cell_size):
+    for cell_x in xrange(int(tile_size / cell_size)):
+        for cell_y in xrange(int(tile_size / cell_size)):
             x1 = cell_x * cell_size
             y1 = cell_y * cell_size
             x2 = x1 + cell_size
@@ -370,7 +370,7 @@ def build_overviews(altered_tiles):
 
 
 def filename_arg_type(s):
-    return s.decode(sys.getfilesystemencoding())
+    return s
 
 
 def parse_image_format(s):
