diff --git a/htdocs/changesshkey.php b/htdocs/changesshkey.php
index e42da07..87f2f77 100644
--- a/htdocs/changesshkey.php
+++ b/htdocs/changesshkey.php
@@ -29,6 +29,7 @@ $result = "";
 $login = $presetLogin;
 $password = "";
 $sshkey = "";
+$replacekey = isset($_POST["replacekey"]) and $_POST["replacekey"];
 $ldap = "";
 $userdn = "";
 $mail = "";
@@ -143,7 +144,7 @@ if ( $result === "" ) {
 # Change sshPublicKey
 #==============================================================================
 if ( $result === "" ) {
-    $result = change_sshkey($ldap, $userdn, $change_sshkey_attribute, $sshkey);
+    $result = change_sshkey($ldap, $userdn, $change_sshkey_attribute, $sshkey, $replacekey);
 }
 
 #==============================================================================
diff --git a/lang/de.inc.php b/lang/de.inc.php
index 146158c..fe9b44f 100644
--- a/lang/de.inc.php
+++ b/lang/de.inc.php
@@ -148,8 +148,9 @@ $messages['policyforbiddenldapfields'] = "Ihr Passwort darf keine Werte aus folg
 $messages['ldap_cn'] = "Name";
 $messages['ldap_givenName'] = "Vorname";
 $messages['ldap_sn'] = "Nachname";
-$messages['ldap_mail'] = "Email-Adresse";
-$messages["questionspopulatehint"] = "Enter only your login to retrieve the questions you've registered.";
-$messages['badquality'] = "Password quality is too low";
-$messages['tooyoung'] = "Password was changed too recently";
-$messages['inhistory'] = "Password is in history of old passwords";
+$messages['ldap_mail'] = "E-Mail-Adresse";
+$messages["questionspopulatehint"] = "Geben Sie Ihren Benutzernamen ein, um die hinterlegten Fragen anzuzeigen.";
+$messages['badquality'] = "Geringe Passwortqualität";
+$messages['tooyoung'] = "Das Passwort wurde zu häufig geändert";
+$messages['inhistory'] = "Das Passwort wurde früher bereits verwendet";
+$messages['replacekey'] = "Vorhandene Schlüssel entfernen";
diff --git a/lang/en.inc.php b/lang/en.inc.php
index 342d646..1715932 100644
--- a/lang/en.inc.php
+++ b/lang/en.inc.php
@@ -151,3 +151,4 @@ $messages["questionspopulatehint"] = "Enter only your login to retrieve the ques
 $messages['badquality'] = "Password quality is too low";
 $messages['tooyoung'] = "Password was changed too recently";
 $messages['inhistory'] = "Password is in history of old passwords";
+$messages['replacekey'] = "Remove existing keys";
diff --git a/lib/functions.inc.php b/lib/functions.inc.php
index e9fb225..d61a55e 100644
--- a/lib/functions.inc.php
+++ b/lib/functions.inc.php
@@ -521,14 +521,19 @@ function change_password( $ldap, $dn, $password, $ad_mode, $ad_options, $samba_m
 
 # Change sshPublicKey attribute
 # @return result code
-function change_sshkey( $ldap, $dn, $attribute, $sshkey ) {
+function change_sshkey( $ldap, $dn, $attribute, $sshkey, $replacekey) {
 
     $result = "";
 
     $userdata[$attribute] = $sshkey;
 
     # Commit modification on directory
-    $replace = ldap_mod_replace($ldap, $dn, $userdata);
+    if($replacekey)
+    {
+        $replace = ldap_mod_replace($ldap, $dn, $userdata);
+    } else {
+        $replace = ldap_mod_add($ldap, $dn, $userdata);
+    }
 
     $errno = ldap_errno($ldap);
 
diff --git a/templates/changesshkey.tpl b/templates/changesshkey.tpl
index 48c2cb7..17b608a 100644
--- a/templates/changesshkey.tpl
+++ b/templates/changesshkey.tpl
@@ -33,6 +33,14 @@
                 </div>
             </div>
         </div>
+	<div class="form-group">
+            <label for="replacekey" class="col-sm-4 control-label">{$msg_replacekey}</label>
+            <div class="col-sm-8">
+                <div class="input-group">
+                   <input type="checkbox" name="replacekey" id="replacekey" value="replacekey">
+                </div>
+            </div>
+        </div>
         {if ($use_captcha)}
              {include file="captcha.tpl"}
         {/if}
