diff --git a/bin/tests.py b/bin/tests.py
index 4b61a53..9537e55 100755
--- a/bin/tests.py
+++ b/bin/tests.py
@@ -285,7 +285,10 @@ def run_tests(INFOS):
     if os.path.isdir(newpath):
       sys.stdout.write('%s *** OVERWRITTEN ***\n' % (newpath))
       shutil.rmtree(newpath)
     shutil.copytree(path,newpath)
+    os.system("mkdir -p " + newpath)
+    os.system("cp -r " + path + "/* " + newpath)
+    os.system("chmod -R +w " + newpath)
     os.chdir(newpath)

     starttime=datetime.datetime.now()
