From 52cde70b785398d794214a21e63e8a5c02d13254 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 28 Oct 2019 05:58:45 -0500
Subject: [PATCH 2/2] app-header: tooltips

---
 src/app-header.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/app-header.c b/src/app-header.c
index 53cd9b0..bbc2245 100644
--- a/src/app-header.c
+++ b/src/app-header.c
@@ -134,6 +134,9 @@ void app_header_init(AppHeader* ah)
 	g_signal_connect_swapped(prev, "clicked", (GCallback) &app_header_nav_prev, ah);
 	g_signal_connect_swapped(ah->btn_current, "clicked", (GCallback) &app_header_nav_current, ah);
 	g_signal_connect_swapped(ah->btn_next, "clicked", (GCallback) &app_header_nav_next, ah);
+	gtk_widget_set_tooltip_text(prev, "Previous week");
+	gtk_widget_set_tooltip_text(ah->btn_current, "Show today");
+	gtk_widget_set_tooltip_text(ah->btn_next, "Next week");
 
 	ah->btn_menu = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(ah->btn_menu), gtk_image_new_from_icon_name("open-menu-symbolic", GTK_ICON_SIZE_MENU));
@@ -149,16 +152,19 @@ void app_header_init(AppHeader* ah)
 	gtk_button_set_image(GTK_BUTTON(ah->btn_sync), ah->icon_sync);
 	g_signal_connect_swapped(ah->btn_sync, "clicked", (GCallback) &app_header_sync, ah);
 	gtk_header_bar_pack_end(GTK_HEADER_BAR(ah), ah->btn_sync);
+	gtk_widget_set_tooltip_text(ah->btn_sync, "Synchronize");
 
 	ah->btn_delete = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(ah->btn_delete), gtk_image_new_from_icon_name("edit-delete-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect_swapped(ah->btn_delete, "clicked", (GCallback) &app_header_delete, ah);
 	gtk_header_bar_pack_end(GTK_HEADER_BAR(ah), ah->btn_delete);
+	gtk_widget_set_tooltip_text(ah->btn_delete, "Delete");
 
 	ah->btn_save = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(ah->btn_save), gtk_image_new_from_icon_name("document-save-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect_swapped(ah->btn_save, "clicked", (GCallback) &app_header_save, ah);
 	gtk_header_bar_pack_end(GTK_HEADER_BAR(ah), ah->btn_save);
+	gtk_widget_set_tooltip_text(ah->btn_save, "Save");
 
 	gtk_header_bar_pack_start(GTK_HEADER_BAR(ah), nav);
 }
-- 
2.23.0

