From c16a1b8d0ba07792b0c8a801177c9ea73514c7c5 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 28 Oct 2019 05:49:11 -0500
Subject: [PATCH] event-popups: tooltips for buttons

---
 src/event-popup.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/event-popup.c b/src/event-popup.c
index 3712491..4dad056 100644
--- a/src/event-popup.c
+++ b/src/event-popup.c
@@ -117,31 +117,37 @@ static void event_popup_init(EventPopup* e)
 	btn = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(btn), gtk_image_new_from_icon_name("emblem-ok-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(btn, "clicked", (GCallback) &rsvp_yes_clicked, e);
+	gtk_widget_set_tooltip_text(btn, "RSVP: Yes");
 	gtk_action_bar_pack_start(GTK_ACTION_BAR(actions), btn);
 
 	btn = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(btn), gtk_image_new_from_icon_name("dialog-question-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(btn, "clicked", (GCallback) &rsvp_maybe_clicked, e);
+	gtk_widget_set_tooltip_text(btn, "RSVP: Maybe");
 	gtk_action_bar_pack_start(GTK_ACTION_BAR(actions), btn);
 
 	btn = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(btn), gtk_image_new_from_icon_name("dialog-error-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(btn, "clicked", (GCallback) &rsvp_no_clicked, e);
+	gtk_widget_set_tooltip_text(btn, "RSVP: No");
 	gtk_action_bar_pack_start(GTK_ACTION_BAR(actions), btn);
 
 	e->btn_delete = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(e->btn_delete), gtk_image_new_from_icon_name("edit-delete-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(e->btn_delete, "clicked", (GCallback) &delete_clicked, e);
+	gtk_widget_set_tooltip_text(e->btn_delete, "Delete");
 	gtk_action_bar_pack_end(GTK_ACTION_BAR(actions), e->btn_delete);
 
 	e->btn_save = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(e->btn_save), gtk_image_new_from_icon_name("document-save-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(e->btn_save, "clicked", (GCallback) &save_clicked, e);
+	gtk_widget_set_tooltip_text(e->btn_save, "Save");
 	gtk_action_bar_pack_end(GTK_ACTION_BAR(actions), e->btn_save);
 
 	btn = gtk_button_new();
 	gtk_button_set_image(GTK_BUTTON(btn), gtk_image_new_from_icon_name("view-more-horizontal-symbolic", GTK_ICON_SIZE_SMALL_TOOLBAR));
 	g_signal_connect(btn, "clicked", (GCallback) &open_details, e);
+	gtk_widget_set_tooltip_text(btn, "Details...");
 	gtk_action_bar_pack_end(GTK_ACTION_BAR(actions), btn);
 
 	gtk_box_pack_start(GTK_BOX(box), bar, FALSE, FALSE, 0);
-- 
2.24.0-rc1

