diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08b2770..2eb2e98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,16 +44,7 @@ else()
 	message(FATAL_ERROR "Compiler is not supported")
 endif()
 
-FetchContent_Declare(qsp-custom
-	GIT_REPOSITORY https://github.com/QSPFoundation/qsp.git
-	GIT_TAG master
-	#GIT_SHALLOW true # it works only with branch names and tags
-	GIT_PROGRESS true
-	SUBBUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/qsp-download"
-	SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/qsp-src"
-	BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/qsp-build"
-)
-FetchContent_MakeAvailable(qsp-custom)
+find_package(Qsp REQUIRED)
 
 if (USE_INSTALLED_WX)
 	find_package(wxWidgets REQUIRED base core adv aui html net)
@@ -148,7 +139,9 @@ else()
 endif()
 
 target_compile_definitions(qspgui PRIVATE _UNICODE)
-target_link_libraries(qspgui PRIVATE qsp wxbase wxcore wxadv wxaui wxnet wxhtml)
+target_link_libraries(qspgui PRIVATE qsp ${wxWidgets_LIBRARIES})
+target_include_directories(qspgui PRIVATE ${qsp_INCLUDE_DIRS})
+target_link_directories(qspgui PRIVATE ${qsp_LIBRARY_DIRS})
 
 # Installation
 
@@ -180,8 +173,8 @@ else()
 	message(FATAL_ERROR "Platform is not supported")
 endif()
 
-set_target_properties(qspgui qsp PROPERTIES INSTALL_RPATH "${INSTALL_RPATH}")
-install(TARGETS qspgui qsp
+set_target_properties(qspgui PROPERTIES INSTALL_RPATH "${INSTALL_RPATH}")
+install(TARGETS qspgui
 		COMPONENT Main
 		RUNTIME DESTINATION ${BINDIR}
 		LIBRARY DESTINATION ${RLIBDIR}
