diff --git a/htdocs/login.php b/htdocs/login.php
index 48f9e2e..298b07d 100644
--- a/htdocs/login.php
+++ b/htdocs/login.php
@@ -13,7 +13,7 @@ require './common.php';
 
 $user = array();
 $user['login'] = get_request('login');
-$user['password'] = get_request('login_pass');
+$user['password'] = get_request('login_pass','POST',false,null,false);
 
 if ($user['login'] && ! strlen($user['password']))
        system_message(array(
