diff --git i/radicale/auth/htpasswd.py w/radicale/auth/htpasswd.py
index dd66dfbd..23922f91 100644
--- i/radicale/auth/htpasswd.py
+++ w/radicale/auth/htpasswd.py
@@ -68,7 +68,7 @@ class Auth(auth.BaseAuth):
 
     _filename: str
     _encoding: str
-    _htpasswd: dict             # login -> digest
+    _htpasswd: dict[str, list[str]]
     _htpasswd_mtime_ns: int
     _htpasswd_size: int
     _htpasswd_ok: bool
@@ -217,7 +217,7 @@ class Auth(auth.BaseAuth):
         else:
             return self._plain(hash_value, password)
 
-    def _read_htpasswd(self, init: bool, suppress: bool) -> Tuple[bool, int, int, dict, int, int]:
+    def _read_htpasswd(self, init: bool, suppress: bool) -> Tuple[bool, int, int, dict[str, list[str]], int, int]:
         """Read htpasswd file
 
         init == True: stop on error
@@ -237,7 +237,7 @@ class Auth(auth.BaseAuth):
             logger.info("%s content of htpasswd file start: %r", info, self._filename)
         else:
             logger.debug("%s content of htpasswd file start: %r", info, self._filename)
-        htpasswd: dict[str, str] = dict()
+        htpasswd: dict[str, list[str]] = dict()
         entries = 0
         duplicates = 0
         errors = 0
@@ -260,7 +260,7 @@ class Auth(auth.BaseAuth):
                                     htpasswd_ok = False
                                     skip = True
                             else:
-                                if htpasswd.get(login):
+                                if False:
                                     duplicates += 1
                                     if init is True:
                                         raise ValueError("htpasswd file contains duplicate login: '%s'", login, line_num)
@@ -286,7 +286,9 @@ class Auth(auth.BaseAuth):
                                                 skip = True
                                                 htpasswd_ok = False
                             if skip is False:
-                                htpasswd[login] = digest
+                                if login not in htpasswd:
+                                    htpasswd[login] = []
+                                htpasswd[login].append(digest)
                                 entries += 1
                         except ValueError as e:
                             if init is True:
@@ -321,7 +323,7 @@ class Auth(auth.BaseAuth):
 
         """
         login_ok = False
-        digest: str
+        digests: list[str] = []
         if self._htpasswd_cache is True:
             # check and re-read file if required
             with self._lock:
@@ -342,18 +344,23 @@ class Auth(auth.BaseAuth):
                     self._htpasswd_not_ok_time = current_time
 
             if self._htpasswd.get(login):
-                digest = self._htpasswd[login]
+                digests = self._htpasswd[login]
                 login_ok = True
         else:
             # read file on every request
             (htpasswd_ok, htpasswd_bcrypt_use, htpasswd_argon2_use, htpasswd, htpasswd_size, htpasswd_mtime_ns) = self._read_htpasswd(False, True)
             if htpasswd.get(login):
-                digest = htpasswd[login]
+                digests = htpasswd[login]
                 login_ok = True
 
         if login_ok is True:
             try:
-                (method, password_ok) = self._verify(digest, password)
+                method = None
+                password_ok = False
+                for digest in digests:
+                    (method, password_ok) = self._verify(digest, password)
+                    if password_ok:
+                        break;
             except ValueError as e:
                 logger.error("Login verification failed for user: '%s' (htpasswd/%s) with error '%s'", login, self._encryption, e)
                 return ""
