diff --git a/data/dconf/make-dconf-override-db.sh b/data/dconf/make-dconf-override-db.sh
index 32cb1530..4ce36bc1 100755
--- a/data/dconf/make-dconf-override-db.sh
+++ b/data/dconf/make-dconf-override-db.sh
@@ -12,10 +12,6 @@ export XDG_CACHE_HOME="$TMPDIR/cache"
 export GSETTINGS_SCHEMA_DIR="$TMPDIR/schemas"
 mkdir -p $XDG_CONFIG_HOME $XDG_CACHE_HOME $GSETTINGS_SCHEMA_DIR

-eval `dbus-launch --sh-syntax`
-
-trap cleanup EXIT
-
 cleanup() {
   test $? -eq 0 && exit
   rm -rf $TMPDIR; kill $DBUS_SESSION_BUS_PID
@@ -57,5 +53,3 @@ if [ -d $TMPDIR/cache/gvfs ] ; then
     umount $TMPDIR/cache/gvfs
 fi
 rm -rf $TMPDIR
-
-kill $DBUS_SESSION_BUS_PID
