diff --git a/configure.ac b/configure.ac
index 392cc581..6d5750dc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -345,11 +345,7 @@ fi
 AC_SUBST(X11_PREFIX)

 # Check locale dir for Compose files.
-AC_CHECK_FILE($X11_PREFIX/share/X11/locale/locale.dir,
-              X11_LOCALEDATADIR="$X11_PREFIX/share/X11/locale",
-              [AC_CHECK_FILE($X11_PREFIX/lib/X11/locale/locale.dir,
-                             X11_LOCALEDATADIR="$X11_PREFIX/lib/X11/locale",
-                             X11_LOCALEDATADIR="$(datadir)/X11/locale")])
+X11_LOCALEDATADIR="$X11_PREFIX/share/X11/locale"
 AC_SUBST(X11_LOCALEDATADIR)

 if test x"$enable_wayland" = x"yes"; then

