diff --git a/src/utils/mod.rs b/src/utils/mod.rs
index 2ac37721..dd07d04d 100644
--- a/src/utils/mod.rs
+++ b/src/utils/mod.rs
@@ -11,7 +11,6 @@ use std::time::Duration;
 use anyhow::{ensure, Context};
 use bitflags::bitflags;
 use directories::UserDirs;
-use git_version::git_version;
 use niri_config::{Config, OutputName};
 use smithay::backend::renderer::utils::with_renderer_surface_state;
 use smithay::input::pointer::CursorIcon;
@@ -83,22 +82,7 @@ impl ResizeEdge {
 }
 
 pub fn version() -> String {
-    if let Some(v) = option_env!("NIRI_BUILD_VERSION_STRING") {
-        return String::from(v);
-    }
-
-    const MAJOR: &str = env!("CARGO_PKG_VERSION_MAJOR");
-    const MINOR: &str = env!("CARGO_PKG_VERSION_MINOR");
-    const PATCH: &str = env!("CARGO_PKG_VERSION_PATCH");
-
-    let commit =
-        option_env!("NIRI_BUILD_COMMIT").unwrap_or(git_version!(fallback = "unknown commit"));
-
-    if PATCH == "0" {
-        format!("{MAJOR}.{MINOR:0>2} ({commit})")
-    } else {
-        format!("{MAJOR}.{MINOR:0>2}.{PATCH} ({commit})")
-    }
+    return "@version@".to_string();
 }
 
 pub fn get_monotonic_time() -> Duration {
