diff --git a/src/cli.rs b/src/cli.rs
index cb3cc0ab..8bd3317b 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -26,6 +26,10 @@ pub struct Cli {
     /// on a TTY as your non-main compositor instance, to avoid messing up the global environment.
     #[arg(long)]
     pub session: bool,
+
+    #[arg(long)]
+    pub no_import_environment: bool,
+
     /// Command to run upon compositor startup.
     #[arg(last = true)]
     pub command: Vec<OsString>,
diff --git a/src/main.rs b/src/main.rs
index 401db23a..618158f6 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -201,8 +201,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     }
 
     if cli.session {
-        // We're starting as a session. Import our variables.
-        import_environment();
+        if (!cli.no_import_environment) {
+            // We're starting as a session. Import our variables.
+            import_environment();
+        }
 
         // Inhibit power key handling so we can suspend on it.
         #[cfg(feature = "dbus")]
