diff --git i/eyefi-linux.c w/eyefi-linux.c
index 091157d..1622244 100644
--- i/eyefi-linux.c
+++ w/eyefi-linux.c
@@ -128,11 +128,12 @@ int zero_file(enum eyefi_file file, char *mnt)
 	char *zerobuf[EYEFI_BUF_SIZE];
 	int fd;
 	int ret;
+	mode_t perms = 0777;
 	
 	memset(&zerobuf[0], 0, EYEFI_BUF_SIZE);
 	fname = eyefi_file_on(file, mnt);
 	debug_printf(1, "creating control file: '%s'\n", fname);
-	fd = open(fname, O_WRONLY|O_CREAT);
+	fd = open(fname, O_WRONLY|O_CREAT, perms);
 	ret = fd;
 	if (ret < 0)
 		goto out;
