diff --git a/src/greeter.c b/src/greeter.c
index c94b1a1..8f82817 100644
--- a/src/greeter.c
+++ b/src/greeter.c
@@ -336,7 +336,7 @@ main(int argc, char **argv) {
 	GdkWindow *root_window;
 	GdkRectangle geometry;
 	GKeyFile *keyfile;
-	gchar *theme;
+	gchar *theme, *theme_uri;
 	GError *err = NULL;
 	GdkRGBA bg_color;
 	WebKitWebContext *context;
@@ -389,6 +389,13 @@ main(int argc, char **argv) {
 	}
 
 	theme = rtrim_comments(theme);
+
+	if (theme[0] == '/') {
+		theme_uri = g_strdup_printf("file://%s/index.html", theme);
+	} else {
+		theme_uri = g_strdup_printf("file://%s/%s/index.html", THEME_DIR, theme);
+	}
+
 	config_timeout = g_key_file_get_integer(keyfile, "greeter", "screensaver_timeout", &err);
 
 	if ( NULL != err) {
@@ -487,8 +494,7 @@ main(int argc, char **argv) {
 
 	/* There's no turning back now, let's go! */
 	gtk_container_add(GTK_CONTAINER(window), web_view);
-	webkit_web_view_load_uri(WEBKIT_WEB_VIEW(web_view),
-							 g_strdup_printf("file://%s/%s/index.html", THEME_DIR, theme));
+	webkit_web_view_load_uri(WEBKIT_WEB_VIEW(web_view), theme_uri);
 
 	gtk_widget_show_all(window);
 	gtk_widget_set_can_focus(GTK_WIDGET(web_view), TRUE);
diff --git a/src/webkit2-extension.c b/src/webkit2-extension.c
index 94de995..1045408 100644
--- a/src/webkit2-extension.c
+++ b/src/webkit2-extension.c
@@ -103,6 +103,7 @@ static gboolean
 	SESSION_STARTING;
 
 static gchar
+	*theme,
 	*background_images_dir,
 	*user_image,
 	*logo;
@@ -2001,6 +2002,11 @@ webkit_web_extension_initialize(WebKitWebExtension *extension) {
 
 	paths = g_slist_prepend(paths, THEME_DIR);
 
+	theme = get_config_option_as_string("greeter", "webkit_theme");
+	if (theme[0] == '/') {
+		paths = g_slist_prepend(paths, theme);
+	}
+
 	background_images_dir = get_config_option_as_string("branding", "background_images");
 	paths = g_slist_prepend(paths, background_images_dir);
 
