diff --git a/src/greeter.c b/src/greeter.c
index c94b1a1..d654ab5 100644
--- a/src/greeter.c
+++ b/src/greeter.c
@@ -396,15 +396,15 @@ main(int argc, char **argv) {
 		config_timeout = g_key_file_get_integer(keyfile, "greeter", "screensaver-timeout", &err);
 
 		if ( NULL != err) {
-			g_error_free(err);
+			g_clear_error(&err);
 			config_timeout = 300;
 		}
 	}
 
-	debug_mode = g_key_file_get_boolean(keyfile, "greeter", "debug_mode", NULL);
+	debug_mode = g_key_file_get_boolean(keyfile, "greeter", "debug_mode", &err);
 
 	if ( NULL != err) {
-		g_clear_error(&err);
+		g_error_free(err);
 		debug_mode = FALSE;
 	}
 	/* END Greeter Config File */
