diff --git a/src_build/nob_linux.c b/src_build/nob_linux.c
index b707737f7b..7097babd0c 100644
--- a/src_build/nob_linux.c
+++ b/src_build/nob_linux.c
@@ -40,11 +40,12 @@
     cmd_append(&cmd, "cc",
         "-Wall", "-Wextra", "-ggdb",
         "-I.",
-        "-I"RAYLIB_SRC_FOLDER,
+        "-I@RAYLIB_SRC@/src",
+        "-I@RAYLIB@/include",
         "-I@NOB_H@/include",
         "-o", "./build/musializer",
         "./src/plug.c", "./src/ffmpeg_posix.c", "./src/musializer.c", "./thirdparty/tinyfiledialogs.c",
-        temp_sprintf("-L./build/raylib/%s", MUSIALIZER_TARGET_NAME), "-l:libraylib.a",
+        "-lraylib",
         "-O3", "-march=native", "-ffast-math",
         "-lm", "-ldl", "-flto=auto", "-lpthread");
     if (!cmd_run(&cmd)) return_defer(false);
diff --git a/src_build/nob_stage2.c b/src_build/nob_stage2.c
index 3553cb8a14..c6f2153869 100644
--- a/src_build/nob_stage2.c
+++ b/src_build/nob_stage2.c
@@ -147,7 +147,6 @@
     }

     if (strcmp(subcommand, "build") == 0) {
-        if (!build_raylib()) return 1;
 #ifndef MUSIALIZER_UNBUNDLE
         if (!generate_resource_bundle()) return 1;
 #endif // MUSIALIZER_UNBUNDLE
