diff --git a/setup.py b/setup.py
index c63b92d8..f2b425ec 100644
--- a/setup.py
+++ b/setup.py
@@ -9,8 +9,9 @@ import copy
 import distutils
 from pkg_resources import get_distribution
 from setuptools import setup, Extension
-from setuptools.command.build_ext import build_ext, copy_file
+from setuptools.command.build_ext import build_ext
 from distutils import log
+from shutil import copy2
 
 from distutils.version import LooseVersion
 
@@ -323,9 +324,8 @@ class BuildLibrary(build_ext):
             os.makedirs(self.build_lib)
 
         log.info("Copying {} into {}".format(pylief_path, pylief_dst))
-        copy_file(
-                pylief_path, pylief_dst, verbose=self.verbose,
-                dry_run=self.dry_run)
+        if not self.dry_run:
+            copy2(pylief_path, pylief_dst)
 
 
         # SDK
@@ -339,9 +339,8 @@ class BuildLibrary(build_ext):
             sdk_path = str(sdk_path.pop())
             sdk_output = str(pathlib.Path(CURRENT_DIR) / "build")
 
-            copy_file(
-                sdk_path, sdk_output, verbose=self.verbose,
-                dry_run=self.dry_run)
+            if not self.dry_run:
+                copy2(sdk_path, sdk_output)
 
 def get_platform():
     out = get_platform_backup()
