diff --color -ur a/Makefile b/Makefile
--- a/Makefile	2021-04-09 17:32:41.238187485 +0200
+++ b/Makefile	2021-04-09 17:36:14.428305548 +0200
@@ -8,14 +8,12 @@
 
 install:
 	install -Dm755 create_ap $(DESTDIR)$(BINDIR)/create_ap
-	install -Dm644 create_ap.conf $(DESTDIR)/etc/create_ap.conf
-	[ ! -d /lib/systemd/system ] || install -Dm644 create_ap.service $(DESTDIR)$(PREFIX)/lib/systemd/system/create_ap.service
+	install -Dm644 create_ap.conf $(DESTDIR)$(PREFIX)/etc/create_ap.conf
 	install -Dm644 bash_completion $(DESTDIR)$(PREFIX)/share/bash-completion/completions/create_ap
 	install -Dm644 README.md $(DESTDIR)$(PREFIX)/share/doc/create_ap/README.md
 
 uninstall:
 	rm -f $(DESTDIR)$(BINDIR)/create_ap
 	rm -f $(DESTDIR)/etc/create_ap.conf
-	[ ! -f /lib/systemd/system/create_ap.service ] || rm -f $(DESTDIR)$(PREFIX)/lib/systemd/system/create_ap.service
 	rm -f $(DESTDIR)$(PREFIX)/share/bash-completion/completions/create_ap
 	rm -f $(DESTDIR)$(PREFIX)/share/doc/create_ap/README.md
