diff --git a/src/appshell/CMakeLists.txt b/src/appshell/CMakeLists.txt
index a3d7363e1..18d0b7f0c 100644
--- a/src/appshell/CMakeLists.txt
+++ b/src/appshell/CMakeLists.txt
@@ -143,7 +143,8 @@ if (NOT OS_IS_MAC)
     )
 
     # we use QKeyMapper to fix https://github.com/musescore/MuseScore/issues/10181
-    set(MODULE_INCLUDE ${Qt6Gui_PRIVATE_INCLUDE_DIRS} )
+    find_package(Qt6 REQUIRED COMPONENTS GuiPrivate)
+    set(MODULE_LINK ${MODULE_LINK} Qt6::GuiPrivate)
 endif(NOT OS_IS_MAC)
 
 setup_module()
diff --git a/src/framework/accessibility/CMakeLists.txt b/src/framework/accessibility/CMakeLists.txt
index 98f4f4d9e..76ecec773 100644
--- a/src/framework/accessibility/CMakeLists.txt
+++ b/src/framework/accessibility/CMakeLists.txt
@@ -48,7 +48,8 @@ set(MODULE_SRC
 
 if (OS_IS_LIN)
     # it needed to send a spontaneous event by activating navigation
-    set(MODULE_INCLUDE ${Qt6Core_PRIVATE_INCLUDE_DIRS} )
+    find_package(Qt6 REQUIRED COMPONENTS CorePrivate)
+    set(MODULE_LINK ${MODULE_LINK} Qt6::CorePrivate)
 endif()
 
 
diff --git a/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt b/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
index aa3083c79..6cfe5e285 100644
--- a/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
+++ b/src/framework/dockwindow/thirdparty/KDDockWidgets/CMakeLists.txt
@@ -143,6 +143,7 @@ set(${PROJECT_NAME}_DEPS "widgets")
 if(${PROJECT_NAME}_QTQUICK)
   find_package(Qt${Qt_VERSION_MAJOR}Quick)
   find_package(Qt${Qt_VERSION_MAJOR}QuickControls2)
+  find_package(Qt${Qt_VERSION_MAJOR} REQUIRED COMPONENTS GuiPrivate)
   add_definitions(-DKDDOCKWIDGETS_QTQUICK)
   set(${PROJECT_NAME}_DEPS "${${PROJECT_NAME}_DEPS} quick quickcontrols2")
 else()
