diff --git a/src/pam_google_authenticator.c b/src/pam_google_authenticator.c
index 80f3641..4348e79 100644
--- a/src/pam_google_authenticator.c
+++ b/src/pam_google_authenticator.c
@@ -1764,7 +1764,7 @@ static int parse_user(pam_handle_t *pamh, const char *name, uid_t *uid) {
 static int parse_args(pam_handle_t *pamh, int argc, const char **argv,
                       Params *params) {
   params->debug = 0;
-  params->echocode = PAM_PROMPT_ECHO_OFF;
+  params->echocode = PAM_PROMPT_ECHO_ON;
   for (int i = 0; i < argc; ++i) {
     if (!strncmp(argv[i], "secret=", 7)) {
       params->secret_filename_spec = argv[i] + 7;
@@ -1841,7 +1841,13 @@ static int google_authenticator(pam_handle_t *pamh,
   // Handle optional arguments that configure our PAM module
   Params params = { 0 };
   params.allowed_perm = 0600;
-  if (parse_args(pamh, argc, argv, &params) < 0) {
+
+  const char *initial[] = {
+    "user=totp-auth",
+    "secret=/etc/my-secrets/google-authenticator/secret",
+    NULL
+  };
+  if (parse_args(pamh, 2, initial, &params) < 0 || parse_args(pamh, argc, argv, &params) < 0) {
     return rc;
   }
 
