diff --git a/src/config.c b/src/config.c
index bf7f20e..107ea06 100644
--- a/src/config.c
+++ b/src/config.c
@@ -10,13 +10,8 @@
 #include <stdio.h>
 #include <unistd.h>
 
-#ifndef DEBUG
-	#define INI_LANG DATADIR "/lang/%s.ini"
-	#define INI_CONFIG "/etc/ly/config.ini"
-#else
-	#define INI_LANG "../res/lang/%s.ini"
-	#define INI_CONFIG "../res/config.ini"
-#endif
+#define INI_LANG DATADIR "/lang/%s.ini"
+#define INI_CONFIG "/etc/ly/config.ini"
 
 static void lang_handle(void* data, char** pars, const int pars_count)
 {
@@ -283,23 +278,23 @@ void config_defaults()
 	config.max_desktop_len = 100;
 	config.max_login_len = 255;
 	config.max_password_len = 255;
-	config.mcookie_cmd = strdup("/usr/bin/mcookie");
+	config.mcookie_cmd = strdup("@util-linux@/bin/mcookie");
 	config.min_refresh_delta = 5;
-	config.path = strdup("/sbin:/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin");
-	config.restart_cmd = strdup("/sbin/shutdown -r now");
+	config.path = strdup("");
+	config.restart_cmd = strdup("@systemd@/bin/shutdown -r now");
 	config.save = true;
 	config.save_file = strdup("/etc/ly/save");
 	config.service_name = strdup("ly");
-	config.shutdown_cmd = strdup("/sbin/shutdown -a now");
-	config.term_reset_cmd = strdup("/usr/bin/tput reset");
+	config.shutdown_cmd = strdup("@systemd@/bin/shutdown -a now");
+	config.term_reset_cmd = strdup("@ncurses@/bin/tput reset");
 	config.tty = 2;
 	config.wayland_cmd = strdup(DATADIR "/wsetup.sh");
 	config.wayland_specifier = false;
 	config.waylandsessions = strdup("/usr/share/wayland-sessions");
-	config.x_cmd = strdup("/usr/bin/X");
+	config.x_cmd = strdup("@xorgserver@/bin/X");
 	config.xinitrc = strdup("~/.xinitrc");
 	config.x_cmd_setup = strdup(DATADIR "/xsetup.sh");
-	config.xauth_cmd = strdup("/usr/bin/xauth");
+	config.xauth_cmd = strdup("@xauth@/bin/xauth");
 	config.xsessions = strdup("/usr/share/xsessions");
 }
 
diff --git a/src/main.c b/src/main.c
index 8aea32f..8a5547f 100644
--- a/src/main.c
+++ b/src/main.c
@@ -292,7 +292,7 @@ int main(int argc, char** argv)
 				}
 
 				load(&desktop, &login);
-				system("tput cnorm");
+				system("@ncurses@/bin/tput cnorm");
 				break;
 			default:
 				(*input_handles[active_input])(
