From 65bca1a46b98daeaf280bad4ea77f461e2b85c02 Mon Sep 17 00:00:00 2001
From: Christian Kampka <christian@kampka.net>
Date: Fri, 30 Nov 2018 14:53:21 +0100
Subject: [PATCH 2/2] Fix filter with multiple options

---
 misc/zsh/keybind.zsh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/misc/zsh/keybind.zsh b/misc/zsh/keybind.zsh
index 154d1ff..f453ec6 100644
--- a/misc/zsh/keybind.zsh
+++ b/misc/zsh/keybind.zsh
@@ -4,7 +4,12 @@ __history::keybind::get()
 {
     local buf opt
     # by default, equals to __history::keybind::get_by_dir behavior
-    buf="$(command history search $ZSH_HISTORY_FILTER_OPTIONS --query "$LBUFFER")"
+
+    cmd="command history search $ZSH_HISTORY_FILTER_OPTIONS"
+    if [[ -n "$LBUFFER" ]]; then
+        cmd="$cmd --query "$LBUFFER""
+    fi
+    buf="$(eval $cmd)"
     if [[ -n $buf ]]; then
         BUFFER="$buf"
         CURSOR=$#BUFFER
-- 
2.18.1

