diff --git a/packaging/libcouchbase.pc.in b/packaging/libcouchbase.pc.in
index 94820a65..f2243de8 100644
--- a/packaging/libcouchbase.pc.in
+++ b/packaging/libcouchbase.pc.in
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}/@CMAKE_INSTALL_BINDIR@
-includedir=${prefix}/include
-libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+includedir==@CMAKE_INSTALL_PREFIX@/include
+libdir=@CMAKE_INSTALL_LIBDIR@

 Name: libcouchbase @LCB_VERSION@
 Description: Couchbase client library
