From 12516cb2f952940b3f931806c5c04608ba50ccd2 Mon Sep 17 00:00:00 2001
From: Millian Poquet <millian.poquet@irit.fr>
Date: Fri, 3 Feb 2023 17:12:35 +0100
Subject: [PATCH] bs: loosen pyzmq version constraint

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 513821d..37a394f 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -58,7 +58,7 @@ alembic = ">1.7.0"
 Flask = ">2.0.0"
 tabulate = "^0.8.9"
 Click = ">=8.0.0"
-pyzmq = "^22.0.3"
+pyzmq = ">=22.0.3"
 requests = ">=2.24.0"
 procset = "^1.0"
 simplejson = "^3.17.2"
-- 
2.38.1

