diff --git a/website/server/libs/cron.js b/website/server/libs/cron.js
index 6ac166a2ec..86c2c06157 100644
--- a/website/server/libs/cron.js
+++ b/website/server/libs/cron.js
@@ -470,20 +470,6 @@ export async function cronWrapper (req, res) {
       return null;
     }
 
-    // Clear old completed todos - 30 days for free users, 90 for subscribers
-    // Do not delete challenges completed todos TODO unless the task is broken?
-    // Do not delete group completed todos
-    await Tasks.Task.deleteMany({
-      userId: user._id,
-      type: 'todo',
-      completed: true,
-      dateCompleted: {
-        $lt: moment(now).subtract(user.isSubscribed() ? 90 : 30, 'days').toDate(),
-      },
-      'challenge.id': { $exists: false },
-      'group.id': { $exists: false },
-    }).exec();
-
     const tasks = await Tasks.Task.find({
       userId: user._id,
       $or: [ // Exclude completed todos
