diff --git a/install.sh b/install.sh
index 039922f..1657480 100755
--- a/install.sh
+++ b/install.sh
@@ -15,6 +15,7 @@
 # OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 set -e
+shopt -s extglob
 
 if [ $# -lt 1 ]; then
     echo "$0 {vc.zip sdk.zip target|target}"
@@ -183,21 +184,20 @@ cat "$ORIG"/wrappers/msvcenv.sh \
 | sed 's/SDKVER=.*/SDKVER='$SDKVER/ \
 | sed s/x64/$host/ \
 | sed s/amd64/$dotnet_host/ \
-> msvcenv.sh
+> "$TMPDIR"/msvcenv.sh
 
 for arch in x86 x64 arm arm64; do
     if [ ! -d "vc/tools/msvc/$MSVCVER/bin/Host$host/$arch" ]; then
         continue
     fi
     mkdir -p bin/$arch
-    cp -a "$ORIG"/wrappers/* bin/$arch
-    cat msvcenv.sh | sed 's/ARCH=.*/ARCH='$arch/ > bin/$arch/msvcenv.sh
+    cp -a "$ORIG"/wrappers/!(msvcenv.sh) bin/$arch
+    cat "$TMPDIR"/msvcenv.sh | sed 's/ARCH=.*/ARCH='$arch/ > bin/$arch/msvcenv.sh
 done
-rm msvcenv.sh
 
 if [ -d "$DEST/bin/$host" ]; then
-    if WINE="$(command -v wine64 || command -v wine)"; then
-        WINEDEBUG=-all "${WINE}" wineboot &>/dev/null
+    if WINE="@WINE@"; then
+        WINEDEBUG=-all "${WINE}" wineboot
         echo "Build msvctricks ..."
         "$DEST/bin/$host/cl" /EHsc /O2 "$ORIG/msvctricks.cpp"
         if [ $? -eq 0 ]; then
