diff --git a/nvidia_vbios_vfio_patcher.py b/nvidia_vbios_vfio_patcher.py
index 80087e8..b5aa9ab 100755
--- a/nvidia_vbios_vfio_patcher.py
+++ b/nvidia_vbios_vfio_patcher.py
@@ -70,14 +70,20 @@ class VBIOSROM(object):
             # Search for the footer, which are shortly followed by
             # 'NPDS' and 'NPDE' strings. 'NPDS' and 'NPDE' markers are separated by
             # 28 ASCII characters
-            FOOTER_REGEX = (
-                b'564e(([a-z]|[0-9]){348})(4e504453)(([a-z]|[0-9]){56})(4e504445)'
-            )
-            result = re.compile(FOOTER_REGEX).search(self.content)
-            if not result or len(result.groups()) != 6:
-                raise CheckException("Couldn't find the ROM footer!")
-
-            self.offsets["footer"] = result.start(0)
+            FOOTER_DETECTORS = [
+                (b'564e(([a-z]|[0-9]){476})(4e504453)(([a-z]|[0-9]){56})(4e504445)', 'RTX 20XX / GTX 16XX'),
+                (b'564e(([a-z]|[0-9]){348})(4e504453)(([a-z]|[0-9]){56})(4e504445)', 'GTX 10XX'),
+                (b'564e(([a-z]|[0-9]){124})(4e504453)(([a-z]|[0-9]){56})(4e504445)', 'GTX XXX (400 - 900 Series)'),
+            ]
+            for FOOTER_REGEX, SERIES in FOOTER_DETECTORS:
+                result = re.compile(FOOTER_REGEX).search(self.content)
+                if result and len(result.groups()) == 6:
+                    self.offsets["footer"] = result.start(0)
+                    print("ROM footer for " + SERIES + " found!")
+                    return
+
+            raise CheckException("Couldn't find the ROM footer!")
+
 
     def run_sanity_tests(self, ignore_check=False):
         """
