From cd7f28a8f7c9f13d2a114374e4eafbe57a667ad4 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Sun, 12 Oct 2025 17:38:13 +1100
Subject: [PATCH] wine-path-patch

---
 plugins/VstBase/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/plugins/VstBase/CMakeLists.txt b/plugins/VstBase/CMakeLists.txt
index 53c7c32e8..8f73b008d 100644
--- a/plugins/VstBase/CMakeLists.txt
+++ b/plugins/VstBase/CMakeLists.txt
@@ -52,10 +52,8 @@ endforeach()
 # See upstream bug report: https://bugs.winehq.org/show_bug.cgi?id=58480
 if(LMMS_BUILD_LINUX AND (LMMS_HAVE_VST_32 OR LMMS_HAVE_VST_64))
 	ExternalProject_Add(wine
-		GIT_REPOSITORY https://github.com/tresf/wine
-		GIT_TAG 1f8bb63e75baa5c9f901c8f50b4ea9dd69e0baa0
-		GIT_SHALLOW ON
-		SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/wine"
+		DOWNLOAD_COMMAND ""
+		SOURCE_DIR "${CMAKE_SOURCE_DIR}/plugins/VstBase/wine"
 		SOURCE_SUBDIR tools/winebuild
 		BUILD_ALWAYS ON
 		BUILD_IN_SOURCE ON
-- 
2.51.0

