--- a/src/api-impl-jni/widgets/WrapperWidget.c
+++ b/src/api-impl-jni/widgets/WrapperWidget.c
@@ -129,6 +129,13 @@
 	} else if (orientation == GTK_ORIENTATION_VERTICAL && (wrapper->layout_height > 0)) {
 		*minimum = *natural = wrapper->layout_height;
 	}
+
+	if (minimum_baseline && natural_baseline) {
+		if (*minimum_baseline > *minimum || *natural_baseline > *natural || *minimum_baseline == -1 || *natural_baseline == -1) {
+			*minimum_baseline = -1;
+			*natural_baseline = -1;
+		}
+	}
 }
 
 void wrapper_widget_allocate(GtkWidget *widget, int width, int height, int baseline)
