diff -urN a/src/api-impl/com/google/android/gms/common/ConnectionResult.java b/src/api-impl/com/google/android/gms/common/ConnectionResult.java
--- a/src/api-impl/com/google/android/gms/common/ConnectionResult.java	1970-01-01 10:00:00.000000000 +1000
+++ b/src/api-impl/com/google/android/gms/common/ConnectionResult.java	2026-08-07 19:48:56.685998692 +1000
@@ -0,0 +1,15 @@
+package com.google.android.gms.common;
+import android.os.Parcel;
+import android.os.Parcelable;
+public final class ConnectionResult implements Parcelable {
+    public static final int SUCCESS = 0;
+    public ConnectionResult(int statusCode) {}
+    public int getErrorCode() { return SUCCESS; }
+    public boolean isSuccess() { return true; }
+    @Override public int describeContents() { return 0; }
+    @Override public void writeToParcel(Parcel dest, int flags) {}
+    public static final Parcelable.Creator<ConnectionResult> CREATOR = new Parcelable.Creator<ConnectionResult>() {
+        public ConnectionResult createFromParcel(Parcel in) { return new ConnectionResult(0); }
+        public ConnectionResult[] newArray(int size) { return new ConnectionResult[size]; }
+    };
+}
diff -urN a/src/api-impl/com/google/android/gms/common/GoogleApiAvailability.java b/src/api-impl/com/google/android/gms/common/GoogleApiAvailability.java
--- a/src/api-impl/com/google/android/gms/common/GoogleApiAvailability.java	1970-01-01 10:00:00.000000000 +1000
+++ b/src/api-impl/com/google/android/gms/common/GoogleApiAvailability.java	2026-08-07 19:48:56.688998668 +1000
@@ -0,0 +1,6 @@
+package com.google.android.gms.common;
+import android.content.Context;
+public class GoogleApiAvailability extends GoogleApiAvailabilityLight {
+    private static final GoogleApiAvailability INSTANCE = new GoogleApiAvailability();
+    public static GoogleApiAvailability getInstance() { return INSTANCE; }
+}
diff -urN a/src/api-impl/com/google/android/gms/common/GoogleApiAvailabilityLight.java b/src/api-impl/com/google/android/gms/common/GoogleApiAvailabilityLight.java
--- a/src/api-impl/com/google/android/gms/common/GoogleApiAvailabilityLight.java	1970-01-01 10:00:00.000000000 +1000
+++ b/src/api-impl/com/google/android/gms/common/GoogleApiAvailabilityLight.java	2026-08-07 19:48:56.687998676 +1000
@@ -0,0 +1,7 @@
+package com.google.android.gms.common;
+import android.content.Context;
+public class GoogleApiAvailabilityLight {
+    private static final GoogleApiAvailabilityLight INSTANCE = new GoogleApiAvailabilityLight();
+    public static GoogleApiAvailabilityLight getInstance() { return INSTANCE; }
+    public int isGooglePlayServicesAvailable(Context context) { return ConnectionResult.SUCCESS; }
+}
