--- a/src/api-impl-jni/graphics/android_graphics_BitmapFactory.c	2026-08-08 20:06:57.097480149 +1000
+++ b/src/api-impl-jni/graphics/android_graphics_BitmapFactory.c	2026-08-08 20:08:48.037806533 +1000
@@ -60,6 +60,9 @@
 
 	GdkPixbuf *pixbuf = gdk_pixbuf_new_from_stream(stream, NULL, NULL);
 	g_object_unref(stream);
+
+	if (!pixbuf) return 0;
+
 	GdkTexture *texture = gdk_texture_new_for_pixbuf(pixbuf);
 	g_object_unref(pixbuf);
 	return _INTPTR(texture);
