--- a/src/api-impl-jni/graphics/android_graphics_BitmapFactory.c
+++ b/src/api-impl-jni/graphics/android_graphics_BitmapFactory.c
@@ -28,6 +28,8 @@
 	if (count == -1) { // end of stream
+        printf("ATL_DEBUG: stream EOF\n");
 		return 0;
 	}
 	jbyte *storage_buf = (*env)->GetByteArrayElements(env, stream->storage, NULL);
 	memcpy(buffer, storage_buf, count);
 	(*env)->ReleaseByteArrayElements(env, stream->storage, storage_buf, 0);
+    printf("ATL_DEBUG: read %ld bytes, first byte: %02x\n", (long)count, (unsigned char)((char*)buffer)[0]);
 
 	return count;
 }
