--- /dev/null
+++ b/src/api-impl/android/media/MediaDataSource.java
@@ -0,0 +1,13 @@
+package android.media;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+public abstract class MediaDataSource implements Closeable {
+    public MediaDataSource() {
+    }
+
+    public abstract int readAt(long position, byte[] buffer, int offset, int size) throws IOException;
+
+    public abstract long getSize() throws IOException;
+}
--- a/src/api-impl/meson.build
+++ b/src/api-impl/meson.build
@@ -340,6 +340,7 @@
 	'android/media/MediaRecorder.java',
 	'android/media/MediaRouter.java',
 	'android/media/MediaScannerConnection.java',
+	'android/media/MediaDataSource.java',
 	'android/media/Ringtone.java',
 	'android/media/RingtoneManager.java',
 	'android/media/SoundPool.java',
--- a/src/api-impl/android/media/MediaMetadataRetriever.java.bak	2026-08-09 16:07:00.584648826 +1000
+++ b/src/api-impl/android/media/MediaMetadataRetriever.java	2026-08-09 16:19:39.534453515 +1000
@@ -23,6 +23,18 @@
 		mediaPlayer.setDataSource(path);
 	}
 
+	public void setDataSource(MediaDataSource dataSource) {
+		mediaPlayer = new MediaPlayer();
+	}
+
+	public void setDataSource(java.io.FileDescriptor fd) {
+		mediaPlayer = new MediaPlayer();
+	}
+
+	public void setDataSource(java.io.FileDescriptor fd, long offset, long length) {
+		mediaPlayer = new MediaPlayer();
+	}
+
 	public byte[] getEmbeddedPicture() {
 		return null;
 	}
@@ -36,6 +48,10 @@
 		}
 	}
 
+	public Bitmap getFrameAtTime(long timeUs, int option) {
+		return null;
+	}
+
 	public Bitmap getFrameAtTime(long time) {
 		return null;
 	}
