--- /dev/null
+++ b/src/api-impl/android/media/MediaDataSource.java
@@ -0,0 +1,11 @@
+package android.media;
+
+import java.io.Closeable;
+
+public abstract class MediaDataSource implements Closeable {
+    public MediaDataSource() {}
+    public abstract int readAt(long position, byte[] buffer, int offset, int size) throws java.io.IOException;
+    public abstract long getSize() throws java.io.IOException;
+}
