--- /dev/null
+++ b/src/api-impl/android/media/MediaDataSource.java
@@ -0,0 +1,13 @@
+package android.media;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+public abstract class MediaDataSource implements Closeable {
+    public MediaDataSource() {
+    }
+
+    public abstract int readAt(long position, byte[] buffer, int offset, int size) throws IOException;
+
+    public abstract long getSize() throws IOException;
+}
--- a/src/api-impl/meson.build
+++ b/src/api-impl/meson.build
@@ -340,6 +340,7 @@
 	'android/media/MediaRecorder.java',
 	'android/media/MediaRouter.java',
 	'android/media/MediaScannerConnection.java',
+	'android/media/MediaDataSource.java',
 	'android/media/Ringtone.java',
 	'android/media/RingtoneManager.java',
 	'android/media/SoundPool.java',
