diff --git a/make/bsd/makefiles/rules.make b/make/bsd/makefiles/rules.make
index d40b88d5ed..4983e0061f 100644
--- a/make/bsd/makefiles/rules.make
+++ b/make/bsd/makefiles/rules.make
@@ -32,7 +32,7 @@ DEMANGLE        = $(DEMANGLER) < $@ > .$@ && mv -f .$@ $@
 
 # $(CC) is the c compiler (cc/gcc), $(CXX) is the c++ compiler (CC/g++).
 CC_COMPILE       = $(CC) $(CXXFLAGS) $(CFLAGS)
-CXX_COMPILE      = $(CXX) $(CXXFLAGS) $(CFLAGS)
+CXX_COMPILE      = $(CXX) $(CXXFLAGS) $(CFLAGS) -std=gnu++98
 
 AS.S            = $(AS) $(ASFLAGS)
 
