diff --git a/src/gresource/js/GreeterConfig.js b/src/gresource/js/GreeterConfig.js
index 63be03f..bb02bfd 100644
--- a/src/gresource/js/GreeterConfig.js
+++ b/src/gresource/js/GreeterConfig.js
@@ -35,7 +35,7 @@ function set_values( defaults, target_obj, method ) {
 
 	keys.forEach( prop => {
 		try {
-			target_obj[prop] = method( 'greeter', prop );
+			target_obj[prop] = method( prop );
 		} catch(err) {
 			target_obj[prop] = defaults[prop];
 		}
@@ -74,7 +74,7 @@ class GreeterConfig  {
 
 			_branding = {};
 
-			set_values( props, _branding, this.get_str );
+			set_values( props, _branding, this.get_str.bind( this, 'branding' ) );
 		}
 
 		return _branding;
@@ -104,9 +104,9 @@ class GreeterConfig  {
 
 			_greeter = {};
 
-			set_values( bools, _greeter, this.get_bool );
-			set_values( strings, _greeter, this.get_str );
-			set_values( numbers, _greeter, this.get_num );
+			set_values( bools, _greeter, this.get_bool.bind( this, 'greeter' ) );
+			set_values( strings, _greeter, this.get_str.bind( this, 'greeter' ) );
+			set_values( numbers, _greeter, this.get_num.bind( this, 'greeter' ) );
 		}
 
 		return _greeter;
