From bd3aa2261c9b6453faefd73dd4829ffbb73c0b41 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ren=C3=A9=20Werner?= <mail@renewerner.net>
Date: Fri, 17 Jul 2020 18:39:31 +0200
Subject: [PATCH] copy TOTP when using auto_type all

---
 bwmenu | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/bwmenu b/bwmenu
index be2fa80..a2c5f21 100755
--- a/bwmenu
+++ b/bwmenu
@@ -82,8 +82,9 @@ load_items() {
 exit_error() {
   local code="$1"
   local message="$2"
-
-  rofi -e "$message"
+  if [ $message ]; then
+    rofi -e "$message"
+  fi
   exit "$code"
 }

@@ -225,6 +226,7 @@ auto_type() {
         type_word "$(echo "$2" | jq -r '.[0].login.username')"
         type_tab
         type_word "$(echo "$2" | jq -r '.[0].login.password')"
+        copy_totp "$2"
         ;;
       username)
         type_word "$(echo "$2" | jq -r '.[0].login.username')"
